{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "### In this notebook I will try to repair hyperex amplicon extraction tool\n",
    "\n",
    "hyperex keeps crushing when there is forward primer found upstream revese primer:\n",
    "\n",
    "thread 'main' panicked at /home/mg/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/hyperex-0.1.1/src/utils.rs:420:45: slice index starts at 256 but ends at 65 note: run with \\`RUST_BACKTRACE=1` environment variable to display a backtrace Aborted (core dumped)   \n",
    "\n",
    "I will  load the fasta seq with Biopython and then use hyperex on each read"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mg/ChimeraProject/scripts\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "from Bio import SeqIO\n",
    "print(os.getcwd())\n",
    "from pprint import pprint"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### I can use uchime or unite datasets for Simera, but I will use Unite for reference kmers, so SILVA would be better\n",
    "\n",
    "#### UPDATE: SILVA aslo sucks because there are no primers in the sequences - it it only SSU and LSU database - no ITS I guess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10302\n"
     ]
    }
   ],
   "source": [
    "# only fungi\n",
    "pathh = \"../datasets/silva_data/SILVA_138.2_SSURef_NR99_tax_silva.fasta\"\n",
    "fasta_seq = list(SeqIO.parse(pathh, \"fasta\"))\n",
    "print(f\"{len(fasta_seq):,}\".replace(\",\", \".\"))\n",
    "only_fungi = [i for i in fasta_seq if \"fungi\" in i.description.lower()]\n",
    "print(len(only_fungi))\n",
    "fasta_seq = only_fungi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### lets look at the unite and uchime sets and if the uchime is only subset of unite"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2.030.075 161.335\n"
     ]
    }
   ],
   "source": [
    "pathh_uchime_untrim = \"../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/untrimmed_sequences/uchime_reference_dataset_untrimmed_16_10_2022.fasta\"\n",
    "pathh_unite_its2 = \"../datasets/unite_latest/itsx_out/UNITE_public_21.04.2024_ITSx.ITS2.fasta\"\n",
    "pathh_unite_whole = \"../datasets/unite_latest/UNITE_public_21.04.2024.fasta\"\n",
    "\n",
    "unite, uchime = list(SeqIO.parse(pathh_unite_whole, \"fasta\")), list(SeqIO.parse(pathh_uchime_untrim, \"fasta\"))\n",
    "#fasta_seq = list(SeqIO.parse(pathh, \"fasta\"))\n",
    "#fasta_seq = fasta_seq[:300]\n",
    "print(f\"{len(unite):,}\".replace(\",\", \".\"),f\"{len(uchime):,}\".replace(\",\", \".\"))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2030075 161335\n",
      "['UDB016649', 'UDB016650', 'UDB016651', 'UDB016653', 'UDB016654']\n",
      "['UDB03390183', 'UDB02869914', 'UDB02885146', 'UDB02757411', 'UDB02750185']\n"
     ]
    }
   ],
   "source": [
    "from pprint import pprint\n",
    "unite_ids = [i.id.split(\"|\")[0] for i in unite]\n",
    "uchime_ids = [i.id.split(\"|\")[1] for i in uchime]\n",
    "print(len(unite_ids), len(uchime_ids))\n",
    "pprint(unite_ids[:5])\n",
    "pprint(uchime_ids[:5])\n",
    "unite_ids_set = set(unite_ids)\n",
    "uchime_ids_set = set(uchime_ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "60609\n",
      "100726\n"
     ]
    }
   ],
   "source": [
    "intersection = uchime_ids_set & unite_ids_set\n",
    "print(len(intersection))\n",
    "diff = uchime_ids_set - unite_ids_set\n",
    "print(len(diff))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### there is one milion read that are unique to uchime_untrimmed dataset when compared to UNITE so I will use that"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "fasta_seq = [i for i in uchime if i.id.split(\"|\")[1] in diff]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>homogenized primer names</th>\n",
       "      <th>primers sequences</th>\n",
       "      <th>target marker</th>\n",
       "      <th>Notes</th>\n",
       "      <th>better_names</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>18S-F/5.8S-R</td>\n",
       "      <td>GTAAAAGTCGTAACAAGGTTTC/GTTCAAAGAYTCGATGATTCAC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S-F___5.8S-R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5.8S_Fun/ITS4_Fun</td>\n",
       "      <td>AACTTTYRRCAAYGGATCWCT/AGCCTCCGCTTATTGATATGCTTAART</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.8S_Fun___ITS4_Fun</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>58A2F/ITS4</td>\n",
       "      <td>ATCGATGAAGAACGCAG/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>58A2F___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>fITS7/ITS4</td>\n",
       "      <td>GTGARTCATCGAATCTTTG/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fITS7___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>fITS9/ITS4</td>\n",
       "      <td>GAACGCAGCRAAIIGYGA/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>fITS9=ITSf9=ITS9F</td>\n",
       "      <td>fITS9___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>FSeq/RSeq</td>\n",
       "      <td>ATGCCTGTTTGAGCGTC/CCTACCTGATTTGAGGTC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FSeq___RSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>gITS7/ITS4</td>\n",
       "      <td>GTGARTCATCGARTCTTTG/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gITS7___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>gITS7/ITS4ngs</td>\n",
       "      <td>GTGARTCATCGARTCTTTG/TCCTSCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gITS7___ITS4ngs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>gITS7ngs/ITS4ngsUni</td>\n",
       "      <td>GTGARTCATCRARTYTTTG/CCTSCSCTTANTDATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gITS7ngs___ITS4ngsUni</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ITS1/ITS2</td>\n",
       "      <td>TCCGTAGGTGAACCTGCGG/GCTGCGTTCTTCATCGATGC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1___ITS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ITS1/ITS4</td>\n",
       "      <td>TCCGTAGGTGAACCTGCGG/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITSboth</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ITS1/qITS2*</td>\n",
       "      <td>CTCCGTAGGTGAACCTGCGG/TTYGCTGYGTTCTTCATCG</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1___qITS2_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ITS1-30F/ITS1-217R</td>\n",
       "      <td>GTCCCTGCCCTTTGTACACA/TTTCGCTGCGTTCTTCATCG</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1-30F___ITS1-217R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ITS1F/58A2R</td>\n",
       "      <td>CTTGGTCATTTAGAGGAAGTAA/CTGCGTTCTTCATCGAT</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1F___58A2R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ITS1F/ITS2</td>\n",
       "      <td>CTTGGTCATTTAGAGGAAGTAA/GCTGCGTTCTTCATCGATGC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1F___ITS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ITS1F/ITS3</td>\n",
       "      <td>CTTGGTCATTTAGAGGAAGTAA/GCATCGATGAAGAACGCAGC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1F___ITS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ITS1F/ITS4</td>\n",
       "      <td>CTTGGTCATTTAGAGGAAGTAA/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITSboth</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1F___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ITS1F/LR6</td>\n",
       "      <td>CTTGGTCATTTAGAGGAAGTAA/CGCCAGTTCTGCTTACC</td>\n",
       "      <td>ITSboth</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1F___LR6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ITS1F_KYO1/ITS2_KYO1</td>\n",
       "      <td>CTHGGTCATTTAGAGGAASTAA/CTRYGTTCTTCATCGDT</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1F_KYO1___ITS2_KYO1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ITS1F_KYO1/ITS2_KYO2</td>\n",
       "      <td>CTHGGTCATTTAGAGGAASTAA/TTYRCTRCGTTCTTCATC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1F_KYO1___ITS2_KYO2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ITS1F_KYO2/ITS2_KYO2</td>\n",
       "      <td>TAGAGGAAGTAAAAGTCGTAA/TTYRCTRCGTTCTTCATC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1F_KYO2___ITS2_KYO2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ITS1FI2/5.8S</td>\n",
       "      <td>GAACCWGCGGARGGATCA/CGCTGCGTTCTTCATCG</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1FI2___5.8S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ITS1FI2/ITS2</td>\n",
       "      <td>GAACCWGCGGARGGATCA/GCTGCGTTCTTCATCGATGC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1FI2___ITS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ITS1Fngs/ITS2</td>\n",
       "      <td>GGTCATTTAGAGGAAGTAA/GCTGCGTTCTTCATCGATGC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1Fngs___ITS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ITS1ngs/ITS2</td>\n",
       "      <td>TCCGTAGGTGAACCTGC/GCTGCGTTCTTCATCGATGC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1ngs___ITS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>ITS2F/ITS2R</td>\n",
       "      <td>GCATCGATGAAGAACGC/CCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>highly similar to ITS3/ITS4, and ITS2F/ITS2R s...</td>\n",
       "      <td>ITS2F___ITS2R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>ITS3/ITS4</td>\n",
       "      <td>GCATCGATGAAGAACGCAGC/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS3___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>ITS3_KYO2/ITS4</td>\n",
       "      <td>GATGAAGAACGYAGYRAA/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS3_KYO2___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>ITS3_KYO2/ITS4_KYO3</td>\n",
       "      <td>GATGAAGAACGYAGYRAA/CTBTTVCCKCTTCACTCG</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS3_KYO2___ITS4_KYO3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>ITS5/5.8S_fungi</td>\n",
       "      <td>GGAAGTAAAAGTCGTAACAAGG/CAAGAGATCCGTTGTTGAAAGTT</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS5___5.8S_fungi</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>ITS5/ITS2</td>\n",
       "      <td>GGAAGTAAAAGTCGTAACAAGG/GCTGCGTTCTTCATCGATGC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS5___ITS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>ITS5/ITS2_KYO2</td>\n",
       "      <td>GGAAGTAAAAGTCGTAACAAGG/TTYRCTRCGTTCTTCATC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS5___ITS2_KYO2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>ITS5/ITS4</td>\n",
       "      <td>GGAAGTAAAAGTCGTAACAAGG/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITSboth</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS5___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>ITS7o/ITS4</td>\n",
       "      <td>GTGAATCATCRAATYTTTG/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS7o___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>ITS86F/ITS4</td>\n",
       "      <td>GTGAATCATCGAATCTTTGAA/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS86F___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>ITS86F/ITS4-Tul</td>\n",
       "      <td>GTGAATCATCGAATCTTTGAA/CCGCCAGATTCACACATTGA</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS86F___ITS4-Tul</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>ITS9/ITS4</td>\n",
       "      <td>GCATTAGAAACTGCTCGTAATG/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS9___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>ITS9MUNngs/ITS4ngsUni</td>\n",
       "      <td>TACACACCGCCCGTCG/CCTSCSCTTANTDATATGC</td>\n",
       "      <td>ITSboth</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS9MUNngs___ITS4ngsUni</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   homogenized primer names  \\\n",
       "0              18S-F/5.8S-R   \n",
       "1         5.8S_Fun/ITS4_Fun   \n",
       "2                58A2F/ITS4   \n",
       "3                fITS7/ITS4   \n",
       "4                fITS9/ITS4   \n",
       "5                 FSeq/RSeq   \n",
       "6                gITS7/ITS4   \n",
       "10            gITS7/ITS4ngs   \n",
       "11      gITS7ngs/ITS4ngsUni   \n",
       "12                ITS1/ITS2   \n",
       "13                ITS1/ITS4   \n",
       "14              ITS1/qITS2*   \n",
       "15       ITS1-30F/ITS1-217R   \n",
       "16              ITS1F/58A2R   \n",
       "17               ITS1F/ITS2   \n",
       "18               ITS1F/ITS3   \n",
       "19               ITS1F/ITS4   \n",
       "20                ITS1F/LR6   \n",
       "21     ITS1F_KYO1/ITS2_KYO1   \n",
       "22     ITS1F_KYO1/ITS2_KYO2   \n",
       "23     ITS1F_KYO2/ITS2_KYO2   \n",
       "25             ITS1FI2/5.8S   \n",
       "26             ITS1FI2/ITS2   \n",
       "27            ITS1Fngs/ITS2   \n",
       "28             ITS1ngs/ITS2   \n",
       "31              ITS2F/ITS2R   \n",
       "32                ITS3/ITS4   \n",
       "35           ITS3_KYO2/ITS4   \n",
       "36      ITS3_KYO2/ITS4_KYO3   \n",
       "47          ITS5/5.8S_fungi   \n",
       "48                ITS5/ITS2   \n",
       "49           ITS5/ITS2_KYO2   \n",
       "50                ITS5/ITS4   \n",
       "51               ITS7o/ITS4   \n",
       "52              ITS86F/ITS4   \n",
       "53          ITS86F/ITS4-Tul   \n",
       "54                ITS9/ITS4   \n",
       "55    ITS9MUNngs/ITS4ngsUni   \n",
       "\n",
       "                                    primers sequences target marker  \\\n",
       "0       GTAAAAGTCGTAACAAGGTTTC/GTTCAAAGAYTCGATGATTCAC          ITS1   \n",
       "1   AACTTTYRRCAAYGGATCWCT/AGCCTCCGCTTATTGATATGCTTAART          ITS2   \n",
       "2              ATCGATGAAGAACGCAG/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "3            GTGARTCATCGAATCTTTG/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "4             GAACGCAGCRAAIIGYGA/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "5                ATGCCTGTTTGAGCGTC/CCTACCTGATTTGAGGTC          ITS2   \n",
       "6            GTGARTCATCGARTCTTTG/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "10           GTGARTCATCGARTCTTTG/TCCTSCGCTTATTGATATGC          ITS2   \n",
       "11            GTGARTCATCRARTYTTTG/CCTSCSCTTANTDATATGC          ITS2   \n",
       "12           TCCGTAGGTGAACCTGCGG/GCTGCGTTCTTCATCGATGC          ITS1   \n",
       "13           TCCGTAGGTGAACCTGCGG/TCCTCCGCTTATTGATATGC       ITSboth   \n",
       "14           CTCCGTAGGTGAACCTGCGG/TTYGCTGYGTTCTTCATCG          ITS1   \n",
       "15          GTCCCTGCCCTTTGTACACA/TTTCGCTGCGTTCTTCATCG          ITS1   \n",
       "16           CTTGGTCATTTAGAGGAAGTAA/CTGCGTTCTTCATCGAT          ITS1   \n",
       "17        CTTGGTCATTTAGAGGAAGTAA/GCTGCGTTCTTCATCGATGC          ITS1   \n",
       "18        CTTGGTCATTTAGAGGAAGTAA/GCATCGATGAAGAACGCAGC          ITS1   \n",
       "19        CTTGGTCATTTAGAGGAAGTAA/TCCTCCGCTTATTGATATGC       ITSboth   \n",
       "20           CTTGGTCATTTAGAGGAAGTAA/CGCCAGTTCTGCTTACC       ITSboth   \n",
       "21           CTHGGTCATTTAGAGGAASTAA/CTRYGTTCTTCATCGDT          ITS1   \n",
       "22          CTHGGTCATTTAGAGGAASTAA/TTYRCTRCGTTCTTCATC          ITS1   \n",
       "23           TAGAGGAAGTAAAAGTCGTAA/TTYRCTRCGTTCTTCATC          ITS1   \n",
       "25               GAACCWGCGGARGGATCA/CGCTGCGTTCTTCATCG          ITS1   \n",
       "26            GAACCWGCGGARGGATCA/GCTGCGTTCTTCATCGATGC          ITS1   \n",
       "27           GGTCATTTAGAGGAAGTAA/GCTGCGTTCTTCATCGATGC          ITS1   \n",
       "28             TCCGTAGGTGAACCTGC/GCTGCGTTCTTCATCGATGC          ITS1   \n",
       "31              GCATCGATGAAGAACGC/CCTCCGCTTATTGATATGC          ITS2   \n",
       "32          GCATCGATGAAGAACGCAGC/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "35            GATGAAGAACGYAGYRAA/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "36              GATGAAGAACGYAGYRAA/CTBTTVCCKCTTCACTCG          ITS2   \n",
       "47     GGAAGTAAAAGTCGTAACAAGG/CAAGAGATCCGTTGTTGAAAGTT          ITS1   \n",
       "48        GGAAGTAAAAGTCGTAACAAGG/GCTGCGTTCTTCATCGATGC          ITS1   \n",
       "49          GGAAGTAAAAGTCGTAACAAGG/TTYRCTRCGTTCTTCATC          ITS1   \n",
       "50        GGAAGTAAAAGTCGTAACAAGG/TCCTCCGCTTATTGATATGC       ITSboth   \n",
       "51           GTGAATCATCRAATYTTTG/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "52         GTGAATCATCGAATCTTTGAA/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "53         GTGAATCATCGAATCTTTGAA/CCGCCAGATTCACACATTGA          ITS2   \n",
       "54        GCATTAGAAACTGCTCGTAATG/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "55               TACACACCGCCCGTCG/CCTSCSCTTANTDATATGC       ITSboth   \n",
       "\n",
       "                                                Notes             better_names  \n",
       "0                                                 NaN           18S-F___5.8S-R  \n",
       "1                                                 NaN      5.8S_Fun___ITS4_Fun  \n",
       "2                                                 NaN             58A2F___ITS4  \n",
       "3                                                 NaN             fITS7___ITS4  \n",
       "4                                   fITS9=ITSf9=ITS9F             fITS9___ITS4  \n",
       "5                                                 NaN              FSeq___RSeq  \n",
       "6                                                 NaN             gITS7___ITS4  \n",
       "10                                                NaN          gITS7___ITS4ngs  \n",
       "11                                                NaN    gITS7ngs___ITS4ngsUni  \n",
       "12                                                NaN              ITS1___ITS2  \n",
       "13                                                NaN              ITS1___ITS4  \n",
       "14                                                NaN            ITS1___qITS2_  \n",
       "15                                                NaN     ITS1-30F___ITS1-217R  \n",
       "16                                                NaN            ITS1F___58A2R  \n",
       "17                                                NaN             ITS1F___ITS2  \n",
       "18                                                NaN             ITS1F___ITS3  \n",
       "19                                                NaN             ITS1F___ITS4  \n",
       "20                                                NaN              ITS1F___LR6  \n",
       "21                                                NaN   ITS1F_KYO1___ITS2_KYO1  \n",
       "22                                                NaN   ITS1F_KYO1___ITS2_KYO2  \n",
       "23                                                NaN   ITS1F_KYO2___ITS2_KYO2  \n",
       "25                                                NaN           ITS1FI2___5.8S  \n",
       "26                                                NaN           ITS1FI2___ITS2  \n",
       "27                                                NaN          ITS1Fngs___ITS2  \n",
       "28                                                NaN           ITS1ngs___ITS2  \n",
       "31  highly similar to ITS3/ITS4, and ITS2F/ITS2R s...            ITS2F___ITS2R  \n",
       "32                                                NaN              ITS3___ITS4  \n",
       "35                                                NaN         ITS3_KYO2___ITS4  \n",
       "36                                                NaN    ITS3_KYO2___ITS4_KYO3  \n",
       "47                                                NaN        ITS5___5.8S_fungi  \n",
       "48                                                NaN              ITS5___ITS2  \n",
       "49                                                NaN         ITS5___ITS2_KYO2  \n",
       "50                                                NaN              ITS5___ITS4  \n",
       "51                                                NaN             ITS7o___ITS4  \n",
       "52                                                NaN            ITS86F___ITS4  \n",
       "53                                                NaN        ITS86F___ITS4-Tul  \n",
       "54                                                NaN              ITS9___ITS4  \n",
       "55                                                NaN  ITS9MUNngs___ITS4ngsUni  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import re\n",
    "primers_df = pd.read_excel(\"../my_data/primers.xlsx\",header=1)\n",
    "primers_df[\"better_names\"] = primers_df[\"homogenized primer names\"].apply(lambda x: re.sub(r'[^a-zA-Z0-9\\-\\.\\_]', '_', str(x).replace(\"/\",\"___\")))\n",
    "primers_df_filtered = primers_df[~primers_df[\"primers sequences\"].str.contains(\"[\\\\(a-z\\\\+]\")]\n",
    "primers_df_filtered\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "primers_df_filtered.better_names.to_clipboard()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "from hyperex_repaired import hyperex_repaired"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TEST hyperex_repaired function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNINGS:\n",
      "25-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "1-times  Region  not found because primer TCCGTAGGTGAACCTGCGG was not found in the sequence\n",
      "1-times  Region  not found because primer GCTGCGTTCTTCATCGATGC was not found in the sequence\n",
      "number of real results: 22           ;  ERROR COUNT:  2;  SKIPPED READS:  3;  Execution time: 0.06654691696166992\n",
      "All  reads:\n",
      "seq_1, seq_2, seq_3_bad_miss_f, seq_4, seq_5, seq_6_bad_miss_r, seq_7, seq_8, seq_9_bad_rev_prim_first, seq_10, seq_11, seq_12, seq_13, seq_14, seq_15, seq_16, seq_17, seq_18, seq_19, seq_20, seq_21, seq_22, seq_23, seq_24, seq_25\n",
      "Good reads:\n",
      "seq_1, seq_2, seq_4, seq_5, seq_7, seq_8, seq_10, seq_11, seq_12, seq_13, seq_14, seq_15, seq_16, seq_17, seq_18, seq_19, seq_20, seq_21, seq_22, seq_23, seq_24, seq_25\n",
      "Bad reads:\n",
      "seq_9_bad_rev_prim_first, seq_3_bad_miss_f, seq_6_bad_miss_r\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_1', name='seq_1', description='seq_1 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_2', name='seq_2', description='seq_2 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_4', name='seq_4', description='seq_4 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_5', name='seq_5', description='seq_5 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_7', name='seq_7', description='seq_7 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_8', name='seq_8', description='seq_8 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_10', name='seq_10', description='seq_10 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_11', name='seq_11', description='seq_11 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_12', name='seq_12', description='seq_12 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_13', name='seq_13', description='seq_13 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_14', name='seq_14', description='seq_14 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_15', name='seq_15', description='seq_15 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_16', name='seq_16', description='seq_16 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_17', name='seq_17', description='seq_17 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_18', name='seq_18', description='seq_18 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_19', name='seq_19', description='seq_19 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_20', name='seq_20', description='seq_20 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_21', name='seq_21', description='seq_21 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_22', name='seq_22', description='seq_22 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_23', name='seq_23', description='seq_23 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_24', name='seq_24', description='seq_24 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_25', name='seq_25', description='seq_25 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[])]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#test hyperex_repaired on test_hyperex_repaired.fa\n",
    "test_fasta_seq = list(SeqIO.parse(\"test_hyperex_repaired.fa\", \"fasta\"))\n",
    "test_res_seq = hyperex_repaired(test_fasta_seq,\"TCCGTAGGTGAACCTGCGG\",\"GCTGCGTTCTTCATCGATGC\")\n",
    "print(\"All  reads:\\n\",\", \".join([i.id for i in test_fasta_seq]),sep=\"\")\n",
    "print(\"Good reads:\\n\",\", \".join([i.id for i in test_res_seq]),sep=\"\")\n",
    "print(\"Bad reads:\\n\",\", \".join(set([i.id for i in test_fasta_seq])-set([i.id for i in test_res_seq])),sep=\"\")\n",
    "test_res_seq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract amplicons with hyperex repaired"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18S-F___5.8S-R\n",
      "WARNINGS:726\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "30955-times  Region  not found because primers GTAAAAGTCGTAACAAGGTTTC, GTTCAAAGAYTCGATGATTCAC was not found in the sequence\n",
      "69585-times  Region  not found because primer GTAAAAGTCGTAACAAGGTTTC was not found in the sequence\n",
      "46-times  Region  not found because primer GTTCAAAGAYTCGATGATTCAC was not found in the sequence\n",
      "number of real results: 139           ;  ERROR COUNT:  2;  SKIPPED READS:  100587;  Execution time: 10.63282036781311\n",
      "0\n",
      "5.8S_Fun___ITS4_Fun\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "84325-times  Region  not found because primer AGCCTCCGCTTATTGATATGCTTAART was not found in the sequence\n",
      "16401-times  Region  not found because primers AACTTTYRRCAAYGGATCWCT, AGCCTCCGCTTATTGATATGCTTAART was not found in the sequence\n",
      "number of real results: 0           ;  ERROR COUNT:  1;  SKIPPED READS:  100726;  Execution time: 10.016794204711914\n",
      "1\n",
      "58A2F___ITS4\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "76864-times  Region  not found because primer TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "23584-times  Region  not found because primers ATCGATGAAGAACGCAG, TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "36-times  Region  not found because primer ATCGATGAAGAACGCAG was not found in the sequence\n",
      "number of real results: 242           ;  ERROR COUNT:  1;  SKIPPED READS:  100484;  Execution time: 9.839215517044067\n",
      "2\n",
      "fITS7___ITS4\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "73258-times  Region  not found because primer TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "27190-times  Region  not found because primers GTGARTCATCGAATCTTTG, TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "62-times  Region  not found because primer GTGARTCATCGAATCTTTG was not found in the sequence\n",
      "number of real results: 216           ;  ERROR COUNT:  1;  SKIPPED READS:  100510;  Execution time: 9.708998441696167\n",
      "3\n",
      "fITS9___ITS4\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "100448-times  Region  not found because primers GAACGCAGCRAAIIGYGA, TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "278-times  Region  not found because primer GAACGCAGCRAAIIGYGA was not found in the sequence\n",
      "number of real results: 0           ;  ERROR COUNT:  1;  SKIPPED READS:  100726;  Execution time: 9.912495136260986\n",
      "4\n",
      "FSeq___RSeq\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "99754-times  Region  not found because primers ATGCCTGTTTGAGCGTC, CCTACCTGATTTGAGGTC was not found in the sequence\n",
      "184-times  Region  not found because primer ATGCCTGTTTGAGCGTC was not found in the sequence\n",
      "779-times  Region  not found because primer CCTACCTGATTTGAGGTC was not found in the sequence\n",
      "number of real results: 9           ;  ERROR COUNT:  1;  SKIPPED READS:  100717;  Execution time: 9.763304948806763\n",
      "5\n",
      "gITS7___ITS4\n",
      "WARNINGS:726\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "77692-times  Region  not found because primer TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "22756-times  Region  not found because primers GTGARTCATCGARTCTTTG, TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "50-times  Region  not found because primer GTGARTCATCGARTCTTTG was not found in the sequence\n",
      "number of real results: 227           ;  ERROR COUNT:  2;  SKIPPED READS:  100499;  Execution time: 10.186236381530762\n",
      "6\n",
      "gITS7___ITS4ngs\n",
      "WARNINGS:726\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "77691-times  Region  not found because primer TCCTSCGCTTATTGATATGC was not found in the sequence\n",
      "22756-times  Region  not found because primers GTGARTCATCGARTCTTTG, TCCTSCGCTTATTGATATGC was not found in the sequence\n",
      "50-times  Region  not found because primer GTGARTCATCGARTCTTTG was not found in the sequence\n",
      "number of real results: 228           ;  ERROR COUNT:  2;  SKIPPED READS:  100498;  Execution time: 10.295665979385376\n",
      "10\n",
      "gITS7ngs___ITS4ngsUni\n",
      "WARNINGS:726\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "84087-times  Region  not found because primer CCTSCSCTTANTDATATGC was not found in the sequence\n",
      "16323-times  Region  not found because primers GTGARTCATCRARTYTTTG, CCTSCSCTTANTDATATGC was not found in the sequence\n",
      "43-times  Region  not found because primer GTGARTCATCRARTYTTTG was not found in the sequence\n",
      "number of real results: 272           ;  ERROR COUNT:  2;  SKIPPED READS:  100454;  Execution time: 10.41567873954773\n",
      "11\n",
      "ITS1___ITS2\n",
      "WARNINGS:726\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "71742-times  Region  not found because primer TCCGTAGGTGAACCTGCGG was not found in the sequence\n",
      "28653-times  Region  not found because primers TCCGTAGGTGAACCTGCGG, GCTGCGTTCTTCATCGATGC was not found in the sequence\n",
      "37-times  Region  not found because primer GCTGCGTTCTTCATCGATGC was not found in the sequence\n",
      "number of real results: 293           ;  ERROR COUNT:  2;  SKIPPED READS:  100433;  Execution time: 10.518956899642944\n",
      "12\n",
      "ITS1___ITS4\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "100213-times  Region  not found because primers TCCGTAGGTGAACCTGCGG, TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "235-times  Region  not found because primer TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "182-times  Region  not found because primer TCCGTAGGTGAACCTGCGG was not found in the sequence\n",
      "number of real results: 95           ;  ERROR COUNT:  1;  SKIPPED READS:  100631;  Execution time: 10.501153469085693\n",
      "13\n",
      "ITS1___qITS2_\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "82271-times  Region  not found because primer CTCCGTAGGTGAACCTGCGG was not found in the sequence\n",
      "18425-times  Region  not found because primers CTCCGTAGGTGAACCTGCGG, TTYGCTGYGTTCTTCATCG was not found in the sequence\n",
      "4-times  Region  not found because primer TTYGCTGYGTTCTTCATCG was not found in the sequence\n",
      "number of real results: 26           ;  ERROR COUNT:  1;  SKIPPED READS:  100700;  Execution time: 10.073706865310669\n",
      "14\n",
      "ITS1-30F___ITS1-217R\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "73048-times  Region  not found because primer GTCCCTGCCCTTTGTACACA was not found in the sequence\n",
      "27629-times  Region  not found because primers GTCCCTGCCCTTTGTACACA, TTTCGCTGCGTTCTTCATCG was not found in the sequence\n",
      "11-times  Region  not found because primer TTTCGCTGCGTTCTTCATCG was not found in the sequence\n",
      "number of real results: 38           ;  ERROR COUNT:  1;  SKIPPED READS:  100688;  Execution time: 10.07176947593689\n",
      "15\n",
      "ITS1F___58A2R\n",
      "WARNINGS:726\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "77025-times  Region  not found because primer CTTGGTCATTTAGAGGAAGTAA was not found in the sequence\n",
      "23604-times  Region  not found because primers CTTGGTCATTTAGAGGAAGTAA, CTGCGTTCTTCATCGAT was not found in the sequence\n",
      "16-times  Region  not found because primer CTGCGTTCTTCATCGAT was not found in the sequence\n",
      "number of real results: 80           ;  ERROR COUNT:  2;  SKIPPED READS:  100646;  Execution time: 10.66434621810913\n",
      "16\n",
      "ITS1F___ITS2\n",
      "WARNINGS:726\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "71960-times  Region  not found because primer CTTGGTCATTTAGAGGAAGTAA was not found in the sequence\n",
      "28669-times  Region  not found because primers CTTGGTCATTTAGAGGAAGTAA, GCTGCGTTCTTCATCGATGC was not found in the sequence\n",
      "21-times  Region  not found because primer GCTGCGTTCTTCATCGATGC was not found in the sequence\n",
      "number of real results: 75           ;  ERROR COUNT:  2;  SKIPPED READS:  100651;  Execution time: 10.660815477371216\n",
      "17\n",
      "ITS1F___ITS3\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "100615-times  Region  not found because primers CTTGGTCATTTAGAGGAAGTAA, GCATCGATGAAGAACGCAGC was not found in the sequence\n",
      "97-times  Region  not found because primer GCATCGATGAAGAACGCAGC was not found in the sequence\n",
      "14-times  Region  not found because primer CTTGGTCATTTAGAGGAAGTAA was not found in the sequence\n",
      "number of real results: 0           ;  ERROR COUNT:  1;  SKIPPED READS:  100726;  Execution time: 10.291415452957153\n",
      "18\n",
      "ITS1F___ITS4\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "100411-times  Region  not found because primers CTTGGTCATTTAGAGGAAGTAA, TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "218-times  Region  not found because primer CTTGGTCATTTAGAGGAAGTAA was not found in the sequence\n",
      "37-times  Region  not found because primer TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "number of real results: 60           ;  ERROR COUNT:  1;  SKIPPED READS:  100666;  Execution time: 10.41471266746521\n",
      "19\n",
      "ITS1F___LR6\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "100607-times  Region  not found because primers CTTGGTCATTTAGAGGAAGTAA, CGCCAGTTCTGCTTACC was not found in the sequence\n",
      "68-times  Region  not found because primer CGCCAGTTCTGCTTACC was not found in the sequence\n",
      "22-times  Region  not found because primer CTTGGTCATTTAGAGGAAGTAA was not found in the sequence\n",
      "number of real results: 29           ;  ERROR COUNT:  1;  SKIPPED READS:  100697;  Execution time: 10.261650323867798\n",
      "20\n",
      "ITS1F_KYO1___ITS2_KYO1\n",
      "WARNINGS:726\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "91260-times  Region  not found because primer CTHGGTCATTTAGAGGAASTAA was not found in the sequence\n",
      "9335-times  Region  not found because primers CTHGGTCATTTAGAGGAASTAA, CTRYGTTCTTCATCGDT was not found in the sequence\n",
      "1-times  Region  not found because primer CTRYGTTCTTCATCGDT was not found in the sequence\n",
      "number of real results: 129           ;  ERROR COUNT:  2;  SKIPPED READS:  100597;  Execution time: 10.840009689331055\n",
      "21\n",
      "ITS1F_KYO1___ITS2_KYO2\n",
      "WARNINGS:726\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "88808-times  Region  not found because primer CTHGGTCATTTAGAGGAASTAA was not found in the sequence\n",
      "11787-times  Region  not found because primers CTHGGTCATTTAGAGGAASTAA, TTYRCTRCGTTCTTCATC was not found in the sequence\n",
      "2-times  Region  not found because primer TTYRCTRCGTTCTTCATC was not found in the sequence\n",
      "number of real results: 128           ;  ERROR COUNT:  2;  SKIPPED READS:  100598;  Execution time: 10.429483413696289\n",
      "22\n",
      "ITS1F_KYO2___ITS2_KYO2\n",
      "WARNINGS:726\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "88770-times  Region  not found because primer TAGAGGAAGTAAAAGTCGTAA was not found in the sequence\n",
      "11782-times  Region  not found because primers TAGAGGAAGTAAAAGTCGTAA, TTYRCTRCGTTCTTCATC was not found in the sequence\n",
      "7-times  Region  not found because primer TTYRCTRCGTTCTTCATC was not found in the sequence\n",
      "number of real results: 166           ;  ERROR COUNT:  2;  SKIPPED READS:  100560;  Execution time: 10.67323088645935\n",
      "23\n",
      "ITS1FI2___5.8S\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "80028-times  Region  not found because primer GAACCWGCGGARGGATCA was not found in the sequence\n",
      "20044-times  Region  not found because primers GAACCWGCGGARGGATCA, CGCTGCGTTCTTCATCG was not found in the sequence\n",
      "55-times  Region  not found because primer CGCTGCGTTCTTCATCG was not found in the sequence\n",
      "number of real results: 599           ;  ERROR COUNT:  1;  SKIPPED READS:  100127;  Execution time: 10.558773040771484\n",
      "25\n",
      "ITS1FI2___ITS2\n",
      "WARNINGS:726\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "71445-times  Region  not found because primer GAACCWGCGGARGGATCA was not found in the sequence\n",
      "28627-times  Region  not found because primers GAACCWGCGGARGGATCA, GCTGCGTTCTTCATCGATGC was not found in the sequence\n",
      "63-times  Region  not found because primer GCTGCGTTCTTCATCGATGC was not found in the sequence\n",
      "number of real results: 590           ;  ERROR COUNT:  2;  SKIPPED READS:  100136;  Execution time: 10.740139484405518\n",
      "26\n",
      "ITS1Fngs___ITS2\n",
      "WARNINGS:726\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "71930-times  Region  not found because primer GGTCATTTAGAGGAAGTAA was not found in the sequence\n",
      "28669-times  Region  not found because primers GGTCATTTAGAGGAAGTAA, GCTGCGTTCTTCATCGATGC was not found in the sequence\n",
      "21-times  Region  not found because primer GCTGCGTTCTTCATCGATGC was not found in the sequence\n",
      "number of real results: 105           ;  ERROR COUNT:  2;  SKIPPED READS:  100621;  Execution time: 10.766485452651978\n",
      "27\n",
      "ITS1ngs___ITS2\n",
      "WARNINGS:726\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "71742-times  Region  not found because primer TCCGTAGGTGAACCTGC was not found in the sequence\n",
      "28653-times  Region  not found because primers TCCGTAGGTGAACCTGC, GCTGCGTTCTTCATCGATGC was not found in the sequence\n",
      "37-times  Region  not found because primer GCTGCGTTCTTCATCGATGC was not found in the sequence\n",
      "number of real results: 293           ;  ERROR COUNT:  2;  SKIPPED READS:  100433;  Execution time: 10.917585372924805\n",
      "28\n",
      "ITS2F___ITS2R\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "75458-times  Region  not found because primer CCTCCGCTTATTGATATGC was not found in the sequence\n",
      "24987-times  Region  not found because primers GCATCGATGAAGAACGC, CCTCCGCTTATTGATATGC was not found in the sequence\n",
      "47-times  Region  not found because primer GCATCGATGAAGAACGC was not found in the sequence\n",
      "number of real results: 234           ;  ERROR COUNT:  1;  SKIPPED READS:  100492;  Execution time: 10.788444757461548\n",
      "31\n",
      "ITS3___ITS4\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "71807-times  Region  not found because primer TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "28641-times  Region  not found because primers GCATCGATGAAGAACGCAGC, TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "49-times  Region  not found because primer GCATCGATGAAGAACGCAGC was not found in the sequence\n",
      "number of real results: 229           ;  ERROR COUNT:  1;  SKIPPED READS:  100497;  Execution time: 10.942463397979736\n",
      "32\n",
      "ITS3_KYO2___ITS4\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "88673-times  Region  not found because primer TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "11775-times  Region  not found because primers GATGAAGAACGYAGYRAA, TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "14-times  Region  not found because primer GATGAAGAACGYAGYRAA was not found in the sequence\n",
      "number of real results: 264           ;  ERROR COUNT:  1;  SKIPPED READS:  100462;  Execution time: 10.73490285873413\n",
      "35\n",
      "ITS3_KYO2___ITS4_KYO3\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "88723-times  Region  not found because primer CTBTTVCCKCTTCACTCG was not found in the sequence\n",
      "11779-times  Region  not found because primers GATGAAGAACGYAGYRAA, CTBTTVCCKCTTCACTCG was not found in the sequence\n",
      "10-times  Region  not found because primer GATGAAGAACGYAGYRAA was not found in the sequence\n",
      "number of real results: 214           ;  ERROR COUNT:  1;  SKIPPED READS:  100512;  Execution time: 11.14015245437622\n",
      "36\n",
      "ITS5___5.8S_fungi\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "50442-times  Region  not found because primer GGAAGTAAAAGTCGTAACAAGG was not found in the sequence\n",
      "50088-times  Region  not found because primers GGAAGTAAAAGTCGTAACAAGG, CAAGAGATCCGTTGTTGAAAGTT was not found in the sequence\n",
      "85-times  Region  not found because primer CAAGAGATCCGTTGTTGAAAGTT was not found in the sequence\n",
      "number of real results: 111           ;  ERROR COUNT:  1;  SKIPPED READS:  100615;  Execution time: 10.783565521240234\n",
      "47\n",
      "ITS5___ITS2\n",
      "WARNINGS:726\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "71874-times  Region  not found because primer GGAAGTAAAAGTCGTAACAAGG was not found in the sequence\n",
      "28656-times  Region  not found because primers GGAAGTAAAAGTCGTAACAAGG, GCTGCGTTCTTCATCGATGC was not found in the sequence\n",
      "34-times  Region  not found because primer GCTGCGTTCTTCATCGATGC was not found in the sequence\n",
      "number of real results: 161           ;  ERROR COUNT:  2;  SKIPPED READS:  100565;  Execution time: 11.378705978393555\n",
      "48\n",
      "ITS5___ITS2_KYO2\n",
      "WARNINGS:726\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "88748-times  Region  not found because primer GGAAGTAAAAGTCGTAACAAGG was not found in the sequence\n",
      "11782-times  Region  not found because primers GGAAGTAAAAGTCGTAACAAGG, TTYRCTRCGTTCTTCATC was not found in the sequence\n",
      "7-times  Region  not found because primer TTYRCTRCGTTCTTCATC was not found in the sequence\n",
      "number of real results: 188           ;  ERROR COUNT:  2;  SKIPPED READS:  100538;  Execution time: 11.736497402191162\n",
      "49\n",
      "ITS5___ITS4\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "100347-times  Region  not found because primers GGAAGTAAAAGTCGTAACAAGG, TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "101-times  Region  not found because primer TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "183-times  Region  not found because primer GGAAGTAAAAGTCGTAACAAGG was not found in the sequence\n",
      "number of real results: 95           ;  ERROR COUNT:  1;  SKIPPED READS:  100631;  Execution time: 11.593524932861328\n",
      "50\n",
      "ITS7o___ITS4\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "29616-times  Region  not found because primers GTGAATCATCRAATYTTTG, TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "70832-times  Region  not found because primer TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "52-times  Region  not found because primer GTGAATCATCRAATYTTTG was not found in the sequence\n",
      "number of real results: 226           ;  ERROR COUNT:  1;  SKIPPED READS:  100500;  Execution time: 11.166181802749634\n",
      "51\n",
      "ITS86F___ITS4\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "34534-times  Region  not found because primers GTGAATCATCGAATCTTTGAA, TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "65914-times  Region  not found because primer TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "69-times  Region  not found because primer GTGAATCATCGAATCTTTGAA was not found in the sequence\n",
      "number of real results: 209           ;  ERROR COUNT:  1;  SKIPPED READS:  100517;  Execution time: 11.019603490829468\n",
      "52\n",
      "ITS86F___ITS4-Tul\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "34603-times  Region  not found because primers GTGAATCATCGAATCTTTGAA, CCGCCAGATTCACACATTGA was not found in the sequence\n",
      "66123-times  Region  not found because primer CCGCCAGATTCACACATTGA was not found in the sequence\n",
      "number of real results: 0           ;  ERROR COUNT:  1;  SKIPPED READS:  100726;  Execution time: 10.85806679725647\n",
      "53\n",
      "ITS9___ITS4\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "100448-times  Region  not found because primers GCATTAGAAACTGCTCGTAATG, TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "278-times  Region  not found because primer GCATTAGAAACTGCTCGTAATG was not found in the sequence\n",
      "number of real results: 0           ;  ERROR COUNT:  1;  SKIPPED READS:  100726;  Execution time: 11.103401899337769\n",
      "54\n",
      "ITS9MUNngs___ITS4ngsUni\n",
      "WARNINGS:\n",
      "100600-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "100390-times  Region  not found because primers TACACACCGCCCGTCG, CCTSCSCTTANTDATATGC was not found in the sequence\n",
      "274-times  Region  not found because primer TACACACCGCCCGTCG was not found in the sequence\n",
      "20-times  Region  not found because primer CCTSCSCTTANTDATATGC was not found in the sequence\n",
      "number of real results: 42           ;  ERROR COUNT:  1;  SKIPPED READS:  100684;  Execution time: 10.927330732345581\n",
      "55\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "import sys\n",
    "\n",
    "lenky = {}\n",
    "seqs_list = []\n",
    "for n,row in primers_df_filtered.iterrows():\n",
    "    _,primer_seq,region,_,better_name, = list(row)\n",
    "    forw = primer_seq.split(\"/\")[0].strip()\n",
    "    rev = primer_seq.split(\"/\")[1].strip()\n",
    "    print(better_name)\n",
    "    res_seq = hyperex_repaired(fasta_seq,forw,rev)\n",
    "    output_file = f\"../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_hyperex/{better_name}.fasta\"\n",
    "    output_dir = f\"../datasets/silva_data/amplicons_from_primers_hyperex/{better_name}.fasta\"\n",
    "    with open(output_file, \"w\") as handle:\n",
    "        SeqIO.write(res_seq, handle, \"fasta-2line\")\n",
    "    print(n)\n",
    "    lenky[better_name] = len(res_seq)\n",
    "    seqs_list.append(res_seq)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'seqs_list' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[27], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m ids \u001b[38;5;241m=\u001b[39m {}\n\u001b[0;32m----> 2\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m p,seq \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mzip\u001b[39m(primers_df_filtered[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mbetter_names\u001b[39m\u001b[38;5;124m\"\u001b[39m],seqs_list):\n\u001b[1;32m      3\u001b[0m     ids[p] \u001b[38;5;241m=\u001b[39m [i\u001b[38;5;241m.\u001b[39mid \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m seq]\n",
      "\u001b[0;31mNameError\u001b[0m: name 'seqs_list' is not defined"
     ]
    }
   ],
   "source": [
    "ids = {}\n",
    "for p,seq in zip(primers_df_filtered[\"better_names\"],seqs_list):\n",
    "    ids[p] = [i.id for i in seq]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'ids' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[12], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21;01mitertools\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mimport\u001b[39;00m chain\n\u001b[0;32m----> 2\u001b[0m ids_one_list \u001b[38;5;241m=\u001b[39m chain\u001b[38;5;241m.\u001b[39mfrom_iterable([v \u001b[38;5;28;01mfor\u001b[39;00m k, v \u001b[38;5;129;01min\u001b[39;00m \u001b[43mids\u001b[49m\u001b[38;5;241m.\u001b[39mitems()])\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21;01mcollections\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mimport\u001b[39;00m Counter\n\u001b[1;32m      4\u001b[0m counts \u001b[38;5;241m=\u001b[39m Counter(ids_one_list)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'ids' is not defined"
     ]
    }
   ],
   "source": [
    "from itertools import chain\n",
    "ids_one_list = chain.from_iterable([v for k, v in ids.items()])\n",
    "from collections import Counter\n",
    "counts = Counter(ids_one_list)\n",
    "len(counts),len(fasta_seq)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### use Simera on fastas from hyperex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "skipped = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18S-F___5.8S-R\n",
      "simera!!! 18S-F___5.8S-R GTAAAAGTCGTAACAAGGTTTC GTTCAAAGAYTCGATGATTCAC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed 18S-F___5.8S-R:\n",
      "1 / 38\n",
      "5.8S_Fun___ITS4_Fun\n",
      "Skipping 5.8S_Fun___ITS4_Fun as the file is empty.\n",
      "58A2F___ITS4\n",
      "simera!!! 58A2F___ITS4 ATCGATGAAGAACGCAG TCCTCCGCTTATTGATATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed 58A2F___ITS4:\n",
      "3 / 38\n",
      "fITS7___ITS4\n",
      "simera!!! fITS7___ITS4 GTGARTCATCGAATCTTTG TCCTCCGCTTATTGATATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed fITS7___ITS4:\n",
      "4 / 38\n",
      "fITS9___ITS4\n",
      "Skipping fITS9___ITS4 as the file is empty.\n",
      "FSeq___RSeq\n",
      "Skipping FSeq___RSeq as the file is empty.\n",
      "gITS7___ITS4\n",
      "simera!!! gITS7___ITS4 GTGARTCATCGARTCTTTG TCCTCCGCTTATTGATATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed gITS7___ITS4:\n",
      "7 / 38\n",
      "gITS7___ITS4ngs\n",
      "simera!!! gITS7___ITS4ngs GTGARTCATCGARTCTTTG TCCTSCGCTTATTGATATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed gITS7___ITS4ngs:\n",
      "8 / 38\n",
      "gITS7ngs___ITS4ngsUni\n",
      "simera!!! gITS7ngs___ITS4ngsUni GTGARTCATCRARTYTTTG CCTSCSCTTANTDATATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 74 potential chimeras...Simera timed out for gITS7ngs___ITS4ngsUni. Skipping...\n",
      "ITS1___ITS2\n",
      "simera!!! ITS1___ITS2 TCCGTAGGTGAACCTGCGG GCTGCGTTCTTCATCGATGC\n",
      "Simera timed out for ITS1___ITS2. Skipping...\n",
      "ITS1___ITS4\n",
      "simera!!! ITS1___ITS4 TCCGTAGGTGAACCTGCGG TCCTCCGCTTATTGATATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 40 potential chimeras...Simera timed out for ITS1___ITS4. Skipping...\n",
      "ITS1___qITS2_\n",
      "simera!!! ITS1___qITS2_ CTCCGTAGGTGAACCTGCGG TTYGCTGYGTTCTTCATCG\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS1___qITS2_:\n",
      "12 / 38\n",
      "ITS1-30F___ITS1-217R\n",
      "simera!!! ITS1-30F___ITS1-217R GTCCCTGCCCTTTGTACACA TTTCGCTGCGTTCTTCATCG\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS1-30F___ITS1-217R:\n",
      "13 / 38\n",
      "ITS1F___58A2R\n",
      "simera!!! ITS1F___58A2R CTTGGTCATTTAGAGGAAGTAA CTGCGTTCTTCATCGAT\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS1F___58A2R:\n",
      "14 / 38\n",
      "ITS1F___ITS2\n",
      "simera!!! ITS1F___ITS2 CTTGGTCATTTAGAGGAAGTAA GCTGCGTTCTTCATCGATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS1F___ITS2:\n",
      "15 / 38\n",
      "ITS1F___ITS3\n",
      "Skipping ITS1F___ITS3 as the file is empty.\n",
      "ITS1F___ITS4\n",
      "simera!!! ITS1F___ITS4 CTTGGTCATTTAGAGGAAGTAA TCCTCCGCTTATTGATATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS1F___ITS4:\n",
      "17 / 38\n",
      "ITS1F___LR6\n",
      "simera!!! ITS1F___LR6 CTTGGTCATTTAGAGGAAGTAA CGCCAGTTCTGCTTACC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS1F___LR6:\n",
      "18 / 38\n",
      "ITS1F_KYO1___ITS2_KYO1\n",
      "simera!!! ITS1F_KYO1___ITS2_KYO1 CTHGGTCATTTAGAGGAASTAA CTRYGTTCTTCATCGDT\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS1F_KYO1___ITS2_KYO1:\n",
      "19 / 38\n",
      "ITS1F_KYO1___ITS2_KYO2\n",
      "simera!!! ITS1F_KYO1___ITS2_KYO2 CTHGGTCATTTAGAGGAASTAA TTYRCTRCGTTCTTCATC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS1F_KYO1___ITS2_KYO2:\n",
      "20 / 38\n",
      "ITS1F_KYO2___ITS2_KYO2\n",
      "simera!!! ITS1F_KYO2___ITS2_KYO2 TAGAGGAAGTAAAAGTCGTAA TTYRCTRCGTTCTTCATC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS1F_KYO2___ITS2_KYO2:\n",
      "21 / 38\n",
      "ITS1FI2___5.8S\n",
      "simera!!! ITS1FI2___5.8S GAACCWGCGGARGGATCA CGCTGCGTTCTTCATCG\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 828 potential chimeras...Simera timed out for ITS1FI2___5.8S. Skipping...\n",
      "ITS1FI2___ITS2\n",
      "simera!!! ITS1FI2___ITS2 GAACCWGCGGARGGATCA GCTGCGTTCTTCATCGATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 1912 potential chimeras...Simera timed out for ITS1FI2___ITS2. Skipping...\n",
      "ITS1Fngs___ITS2\n",
      "simera!!! ITS1Fngs___ITS2 GGTCATTTAGAGGAAGTAA GCTGCGTTCTTCATCGATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS1Fngs___ITS2:\n",
      "24 / 38\n",
      "ITS1ngs___ITS2\n",
      "simera!!! ITS1ngs___ITS2 TCCGTAGGTGAACCTGC GCTGCGTTCTTCATCGATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 212 potential chimeras...Simera timed out for ITS1ngs___ITS2. Skipping...\n",
      "ITS2F___ITS2R\n",
      "simera!!! ITS2F___ITS2R GCATCGATGAAGAACGC CCTCCGCTTATTGATATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS2F___ITS2R:\n",
      "26 / 38\n",
      "ITS3___ITS4\n",
      "simera!!! ITS3___ITS4 GCATCGATGAAGAACGCAGC TCCTCCGCTTATTGATATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS3___ITS4:\n",
      "27 / 38\n",
      "ITS3_KYO2___ITS4\n",
      "simera!!! ITS3_KYO2___ITS4 GATGAAGAACGYAGYRAA TCCTCCGCTTATTGATATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS3_KYO2___ITS4:\n",
      "28 / 38\n",
      "ITS3_KYO2___ITS4_KYO3\n",
      "simera!!! ITS3_KYO2___ITS4_KYO3 GATGAAGAACGYAGYRAA CTBTTVCCKCTTCACTCG\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS3_KYO2___ITS4_KYO3:\n",
      "29 / 38\n",
      "ITS5___5.8S_fungi\n",
      "simera!!! ITS5___5.8S_fungi GGAAGTAAAAGTCGTAACAAGG CAAGAGATCCGTTGTTGAAAGTT\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS5___5.8S_fungi:\n",
      "30 / 38\n",
      "ITS5___ITS2\n",
      "simera!!! ITS5___ITS2 GGAAGTAAAAGTCGTAACAAGG GCTGCGTTCTTCATCGATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS5___ITS2:\n",
      "31 / 38\n",
      "ITS5___ITS2_KYO2\n",
      "simera!!! ITS5___ITS2_KYO2 GGAAGTAAAAGTCGTAACAAGG TTYRCTRCGTTCTTCATC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS5___ITS2_KYO2:\n",
      "32 / 38\n",
      "ITS5___ITS4\n",
      "simera!!! ITS5___ITS4 GGAAGTAAAAGTCGTAACAAGG TCCTCCGCTTATTGATATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS5___ITS4:\n",
      "33 / 38\n",
      "ITS7o___ITS4\n",
      "simera!!! ITS7o___ITS4 GTGAATCATCRAATYTTTG TCCTCCGCTTATTGATATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 786 potential chimeras...Simera timed out for ITS7o___ITS4. Skipping...\n",
      "ITS86F___ITS4\n",
      "simera!!! ITS86F___ITS4 GTGAATCATCGAATCTTTGAA TCCTCCGCTTATTGATATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS86F___ITS4:\n",
      "35 / 38\n",
      "ITS86F___ITS4-Tul\n",
      "Skipping ITS86F___ITS4-Tul as the file is empty.\n",
      "ITS9___ITS4\n",
      "Skipping ITS9___ITS4 as the file is empty.\n",
      "ITS9MUNngs___ITS4ngsUni\n",
      "simera!!! ITS9MUNngs___ITS4ngsUni TACACACCGCCCGTCG CCTSCSCTTANTDATATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 10000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed ITS9MUNngs___ITS4ngsUni:\n",
      "38 / 38\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq\n",
    "from time import sleep\n",
    "from subprocess import run\n",
    "import time\n",
    "from subprocess import TimeoutExpired\n",
    "\n",
    "# Function to append \"_1\" to each read name in the FASTA file\n",
    "def append_suffix_to_fasta(input_file, output_file, suffix=\"_1\"):\n",
    "    records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "    modified_records = []\n",
    "    for record in records:\n",
    "        record.id = f\"{record.id}{suffix}\"\n",
    "        record.description = \"\"  # Clear description to avoid duplication\n",
    "        modified_records.append(record)\n",
    "    with open(output_file, \"w\") as handle:\n",
    "        SeqIO.write(modified_records, handle, \"fasta-2line\")\n",
    "\n",
    "# Get the next available directory number for a primer\n",
    "def get_next_dir_number(base_dir):\n",
    "    if not os.path.exists(base_dir):\n",
    "        return 1\n",
    "    \n",
    "    existing_dirs = [d for d in os.listdir(base_dir) if os.path.isdir(os.path.join(base_dir, d)) and d.isdigit()]\n",
    "    if not existing_dirs:\n",
    "        return 1\n",
    "    \n",
    "    dir_numbers = [int(d) for d in existing_dirs]\n",
    "    return max(dir_numbers) + 1\n",
    "\n",
    "# Iterate over each primer and its corresponding file\n",
    "for numero, row in enumerate(primers_df_filtered.iterrows()):\n",
    "    if numero < 0: continue\n",
    "    row = row[1]\n",
    "    primer_name = row[\"better_names\"]\n",
    "    primer_seq = row[\"primers sequences\"]\n",
    "    forward_primer = primer_seq.split(\"/\")[0].strip()\n",
    "    reverse_primer = primer_seq.split(\"/\")[1].strip()\n",
    "        # Check if the FASTA file is empty\n",
    "\n",
    "    #if primer_name != \"gITS7___ITS4\": print(\"skipping....\") ; continue\n",
    "    print(primer_name)\n",
    "\n",
    "    input_file = f\"../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_hyperex/{primer_name}.fasta\"\n",
    "    modified_file = f\"../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_hyperex/{primer_name}_abund.fasta\"\n",
    "    \n",
    "    # Base directory for this primer\n",
    "    primer_base_dir = f\"../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/{primer_name}/\"\n",
    "    \n",
    "    # Get the next sequential directory number\n",
    "    next_dir_num = get_next_dir_number(primer_base_dir)\n",
    "    output_dir = f\"{primer_base_dir}{next_dir_num}/\"\n",
    "    \n",
    "    records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "\n",
    "    if len(records) < 10:  # If no records are found, continue to the next iteration\n",
    "        print(f\"Skipping {primer_name} as the file is empty.\")\n",
    "        continue\n",
    "        \n",
    "    # Append \"_1\" to each read name\n",
    "    append_suffix_to_fasta(input_file, modified_file)\n",
    "    \n",
    "    # Create output directory if it doesn't exist\n",
    "    os.makedirs(output_dir, exist_ok=True)\n",
    "    print(\"simera!!!\", primer_name, forward_primer, reverse_primer)\n",
    "    \n",
    "    # Run Simera\n",
    "    command = [\n",
    "        \"../Simera/Simera\",\n",
    "        \"-i\", modified_file,\n",
    "        \"-o\", output_dir,\n",
    "        \"-n\", \"25\",\n",
    "        \"-s\", \"10000\",\n",
    "        \"-l\", \"0.00005\",\n",
    "        \"-c\", \"10000\", # too large values for big fastas (6+Mb) will freeze Simera sometimes ??? dunno why, it fries my pc tho\n",
    "        \"-f\", forward_primer, # the limit is 20000 sequences in one fasta file                      \n",
    "        \"-r\", reverse_primer,\n",
    "        \"-x\"\n",
    "    ] # Simera is stuck when there is too little seqs in the fasta\n",
    "    \n",
    "    try:\n",
    "        result = run(command, capture_output=False, text=False, timeout=30)  # Timeout after 240 seconds\n",
    "    except TimeoutExpired:\n",
    "        print(f\"Simera timed out for {primer_name}. Skipping...\")\n",
    "        skipped.append((numero, primer_name))\n",
    "        continue  # Skip to the next iteration\n",
    " \n",
    "    # Print the result for debugging\n",
    "    print(f\"Processed {primer_name}:\")\n",
    "    print(numero+1, \"/\", len(primers_df_filtered))\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{(8, 'gITS7ngs___ITS4ngsUni'),\n",
       " (9, 'ITS1___ITS2'),\n",
       " (10, 'ITS1___ITS4'),\n",
       " (21, 'ITS1FI2___5.8S'),\n",
       " (22, 'ITS1FI2___ITS2'),\n",
       " (24, 'ITS1ngs___ITS2'),\n",
       " (33, 'ITS7o___ITS4')}"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(skipped)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cut primers with cutadapt - cut the Simera output - I should do it at the end"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     369\n",
      "Reads with adapters:                       369 (100.0%)\n",
      "Reads written (passing filters):           369 (100.0%)\n",
      "\n",
      "Total basepairs processed:       133,210 bp\n",
      "Total written (filtered):        125,092 bp (93.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 369 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t369\t0.0\t2\t369\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     397\n",
      "Reads with adapters:                       397 (100.0%)\n",
      "Reads written (passing filters):           397 (100.0%)\n",
      "\n",
      "Total basepairs processed:       148,416 bp\n",
      "Total written (filtered):        139,682 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 397 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t397\t0.0\t2\t397\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     376\n",
      "Reads with adapters:                       376 (100.0%)\n",
      "Reads written (passing filters):           376 (100.0%)\n",
      "\n",
      "Total basepairs processed:       139,105 bp\n",
      "Total written (filtered):        130,833 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 376 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t376\t0.0\t2\t376\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     392\n",
      "Reads with adapters:                       392 (100.0%)\n",
      "Reads written (passing filters):           392 (100.0%)\n",
      "\n",
      "Total basepairs processed:       143,655 bp\n",
      "Total written (filtered):        135,031 bp (94.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 392 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t392\t0.0\t2\t392\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/5/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/5/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     391\n",
      "Reads with adapters:                       391 (100.0%)\n",
      "Reads written (passing filters):           391 (100.0%)\n",
      "\n",
      "Total basepairs processed:       145,561 bp\n",
      "Total written (filtered):        136,959 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 391 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t391\t0.0\t2\t391\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/6/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/6/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     405\n",
      "Reads with adapters:                       405 (100.0%)\n",
      "Reads written (passing filters):           405 (100.0%)\n",
      "\n",
      "Total basepairs processed:       153,434 bp\n",
      "Total written (filtered):        144,524 bp (94.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 405 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t405\t0.0\t2\t405\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/7/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/7/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     394\n",
      "Reads with adapters:                       394 (100.0%)\n",
      "Reads written (passing filters):           394 (100.0%)\n",
      "\n",
      "Total basepairs processed:       142,093 bp\n",
      "Total written (filtered):        133,425 bp (93.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 394 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t394\t0.0\t2\t394\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/8/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/8/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     381\n",
      "Reads with adapters:                       381 (100.0%)\n",
      "Reads written (passing filters):           381 (100.0%)\n",
      "\n",
      "Total basepairs processed:       142,304 bp\n",
      "Total written (filtered):        133,922 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 381 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t381\t0.0\t2\t381\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/9/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/9/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     370\n",
      "Reads with adapters:                       370 (100.0%)\n",
      "Reads written (passing filters):           370 (100.0%)\n",
      "\n",
      "Total basepairs processed:       139,122 bp\n",
      "Total written (filtered):        130,982 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 370 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t370\t0.0\t2\t370\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "5.8S_Fun___ITS4_Fun error - not found\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     509\n",
      "Reads with adapters:                       509 (100.0%)\n",
      "Reads written (passing filters):           509 (100.0%)\n",
      "\n",
      "Total basepairs processed:       185,969 bp\n",
      "Total written (filtered):        177,316 bp (95.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 509 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t509\t0.0\t1\t509\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     485\n",
      "Reads with adapters:                       485 (100.0%)\n",
      "Reads written (passing filters):           485 (100.0%)\n",
      "\n",
      "Total basepairs processed:       176,152 bp\n",
      "Total written (filtered):        167,907 bp (95.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 485 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t485\t0.0\t1\t485\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     498\n",
      "Reads with adapters:                       498 (100.0%)\n",
      "Reads written (passing filters):           498 (100.0%)\n",
      "\n",
      "Total basepairs processed:       179,565 bp\n",
      "Total written (filtered):        171,099 bp (95.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 498 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t498\t0.0\t1\t498\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     526\n",
      "Reads with adapters:                       526 (100.0%)\n",
      "Reads written (passing filters):           526 (100.0%)\n",
      "\n",
      "Total basepairs processed:       190,425 bp\n",
      "Total written (filtered):        181,483 bp (95.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 526 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t526\t0.0\t1\t526\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/5/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/5/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     508\n",
      "Reads with adapters:                       508 (100.0%)\n",
      "Reads written (passing filters):           508 (100.0%)\n",
      "\n",
      "Total basepairs processed:       181,794 bp\n",
      "Total written (filtered):        173,158 bp (95.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 508 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t508\t0.0\t1\t508\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/6/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/6/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     496\n",
      "Reads with adapters:                       496 (100.0%)\n",
      "Reads written (passing filters):           496 (100.0%)\n",
      "\n",
      "Total basepairs processed:       180,016 bp\n",
      "Total written (filtered):        171,584 bp (95.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 496 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t496\t0.0\t1\t496\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/7/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/7/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     524\n",
      "Reads with adapters:                       524 (100.0%)\n",
      "Reads written (passing filters):           524 (100.0%)\n",
      "\n",
      "Total basepairs processed:       190,318 bp\n",
      "Total written (filtered):        181,410 bp (95.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 524 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t524\t0.0\t1\t524\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/8/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/8/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     478\n",
      "Reads with adapters:                       478 (100.0%)\n",
      "Reads written (passing filters):           478 (100.0%)\n",
      "\n",
      "Total basepairs processed:       172,917 bp\n",
      "Total written (filtered):        164,791 bp (95.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 478 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t478\t0.0\t1\t478\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/9/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/9/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     494\n",
      "Reads with adapters:                       494 (100.0%)\n",
      "Reads written (passing filters):           494 (100.0%)\n",
      "\n",
      "Total basepairs processed:       178,704 bp\n",
      "Total written (filtered):        170,306 bp (95.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 494 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t494\t0.0\t1\t494\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     361\n",
      "Reads with adapters:                       360 (99.7%)\n",
      "Reads written (passing filters):           361 (100.0%)\n",
      "\n",
      "Total basepairs processed:       111,927 bp\n",
      "Total written (filtered):        105,087 bp (93.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 360 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t360\t0.0\t1\t360\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     346\n",
      "Reads with adapters:                       346 (100.0%)\n",
      "Reads written (passing filters):           346 (100.0%)\n",
      "\n",
      "Total basepairs processed:       106,671 bp\n",
      "Total written (filtered):        100,097 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 346 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t346\t0.0\t1\t346\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     332\n",
      "Reads with adapters:                       332 (100.0%)\n",
      "Reads written (passing filters):           332 (100.0%)\n",
      "\n",
      "Total basepairs processed:       101,806 bp\n",
      "Total written (filtered):         95,498 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 332 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t332\t0.0\t1\t332\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     374\n",
      "Reads with adapters:                       374 (100.0%)\n",
      "Reads written (passing filters):           374 (100.0%)\n",
      "\n",
      "Total basepairs processed:       114,413 bp\n",
      "Total written (filtered):        107,307 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 374 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t374\t0.0\t1\t374\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/5/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/5/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     358\n",
      "Reads with adapters:                       358 (100.0%)\n",
      "Reads written (passing filters):           358 (100.0%)\n",
      "\n",
      "Total basepairs processed:       110,427 bp\n",
      "Total written (filtered):        103,625 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 358 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t358\t0.0\t1\t358\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/6/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/6/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     338\n",
      "Reads with adapters:                       338 (100.0%)\n",
      "Reads written (passing filters):           338 (100.0%)\n",
      "\n",
      "Total basepairs processed:       103,781 bp\n",
      "Total written (filtered):         97,359 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 338 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t338\t0.0\t1\t338\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/7/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/7/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     365\n",
      "Reads with adapters:                       365 (100.0%)\n",
      "Reads written (passing filters):           365 (100.0%)\n",
      "\n",
      "Total basepairs processed:       112,547 bp\n",
      "Total written (filtered):        105,612 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 365 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t365\t0.0\t1\t365\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/8/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/8/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     376\n",
      "Reads with adapters:                       376 (100.0%)\n",
      "Reads written (passing filters):           376 (100.0%)\n",
      "\n",
      "Total basepairs processed:       115,435 bp\n",
      "Total written (filtered):        108,291 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 376 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t376\t0.0\t1\t376\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "fITS9___ITS4 error - not found\n",
      "FSeq___RSeq error - not found\n",
      "Processed gITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     391\n",
      "Reads with adapters:                       391 (100.0%)\n",
      "Reads written (passing filters):           391 (100.0%)\n",
      "\n",
      "Total basepairs processed:       120,549 bp\n",
      "Total written (filtered):        113,120 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 391 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t391\t0.0\t1\t391\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     383\n",
      "Reads with adapters:                       383 (100.0%)\n",
      "Reads written (passing filters):           383 (100.0%)\n",
      "\n",
      "Total basepairs processed:       117,390 bp\n",
      "Total written (filtered):        110,113 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 383 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t383\t0.0\t1\t383\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     384\n",
      "Reads with adapters:                       384 (100.0%)\n",
      "Reads written (passing filters):           384 (100.0%)\n",
      "\n",
      "Total basepairs processed:       118,255 bp\n",
      "Total written (filtered):        110,959 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 384 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t384\t0.0\t1\t384\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     371\n",
      "Reads with adapters:                       371 (100.0%)\n",
      "Reads written (passing filters):           371 (100.0%)\n",
      "\n",
      "Total basepairs processed:       116,117 bp\n",
      "Total written (filtered):        109,068 bp (93.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 371 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t371\t0.0\t1\t371\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/5/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/5/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     377\n",
      "Reads with adapters:                       377 (100.0%)\n",
      "Reads written (passing filters):           377 (100.0%)\n",
      "\n",
      "Total basepairs processed:       116,231 bp\n",
      "Total written (filtered):        109,068 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 377 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t377\t0.0\t1\t377\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/6/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/6/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     396\n",
      "Reads with adapters:                       396 (100.0%)\n",
      "Reads written (passing filters):           396 (100.0%)\n",
      "\n",
      "Total basepairs processed:       122,847 bp\n",
      "Total written (filtered):        115,323 bp (93.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 396 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t396\t0.0\t1\t396\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/7/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/7/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     374\n",
      "Reads with adapters:                       374 (100.0%)\n",
      "Reads written (passing filters):           374 (100.0%)\n",
      "\n",
      "Total basepairs processed:       115,120 bp\n",
      "Total written (filtered):        108,014 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 374 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t374\t0.0\t1\t374\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/8/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/8/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     415\n",
      "Reads with adapters:                       415 (100.0%)\n",
      "Reads written (passing filters):           415 (100.0%)\n",
      "\n",
      "Total basepairs processed:       129,223 bp\n",
      "Total written (filtered):        121,338 bp (93.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 415 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t415\t0.0\t1\t415\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4ngs:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTSCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     460\n",
      "Reads with adapters:                       460 (100.0%)\n",
      "Reads written (passing filters):           460 (100.0%)\n",
      "\n",
      "Total basepairs processed:       142,155 bp\n",
      "Total written (filtered):        133,415 bp (93.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 460 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t460\t0.0\t1\t460\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTSCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4ngs:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTSCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     447\n",
      "Reads with adapters:                       447 (100.0%)\n",
      "Reads written (passing filters):           447 (100.0%)\n",
      "\n",
      "Total basepairs processed:       138,947 bp\n",
      "Total written (filtered):        130,454 bp (93.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 447 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t447\t0.0\t1\t447\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTSCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4ngs:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTSCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     428\n",
      "Reads with adapters:                       428 (100.0%)\n",
      "Reads written (passing filters):           428 (100.0%)\n",
      "\n",
      "Total basepairs processed:       131,961 bp\n",
      "Total written (filtered):        123,829 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 428 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t428\t0.0\t1\t428\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTSCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4ngs:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTSCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     451\n",
      "Reads with adapters:                       451 (100.0%)\n",
      "Reads written (passing filters):           451 (100.0%)\n",
      "\n",
      "Total basepairs processed:       139,806 bp\n",
      "Total written (filtered):        131,237 bp (93.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 451 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t451\t0.0\t1\t451\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTSCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4ngs:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTSCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/5/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/5/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     437\n",
      "Reads with adapters:                       437 (100.0%)\n",
      "Reads written (passing filters):           437 (100.0%)\n",
      "\n",
      "Total basepairs processed:       134,528 bp\n",
      "Total written (filtered):        126,225 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 437 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t437\t0.0\t1\t437\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTSCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4ngs:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTSCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/6/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/6/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     418\n",
      "Reads with adapters:                       418 (100.0%)\n",
      "Reads written (passing filters):           418 (100.0%)\n",
      "\n",
      "Total basepairs processed:       128,983 bp\n",
      "Total written (filtered):        121,041 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 418 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t418\t0.0\t1\t418\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTSCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4ngs:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTSCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/7/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/7/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     405\n",
      "Reads with adapters:                       405 (100.0%)\n",
      "Reads written (passing filters):           405 (100.0%)\n",
      "\n",
      "Total basepairs processed:       124,599 bp\n",
      "Total written (filtered):        116,904 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 405 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t405\t0.0\t1\t405\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTSCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4ngs:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTSCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/8/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/8/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     438\n",
      "Reads with adapters:                       438 (100.0%)\n",
      "Reads written (passing filters):           438 (100.0%)\n",
      "\n",
      "Total basepairs processed:       136,087 bp\n",
      "Total written (filtered):        127,765 bp (93.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 438 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t438\t0.0\t1\t438\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTSCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7ngs___ITS4ngsUni/1/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7ngs___ITS4ngsUni/2/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7ngs___ITS4ngsUni/3/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7ngs___ITS4ngsUni/4/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7ngs___ITS4ngsUni/5/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7ngs___ITS4ngsUni/6/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7ngs___ITS4ngsUni/7/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7ngs___ITS4ngsUni/8/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___ITS2/1/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___ITS2/2/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___ITS2/3/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___ITS2/4/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___ITS2/5/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___ITS2/6/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___ITS2/7/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___ITS2/8/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___ITS4/1/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___ITS4/2/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___ITS4/3/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___ITS4/4/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___ITS4/5/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___ITS4/6/samp_all_seqs.fa does not exist. Skipping...\n",
      "Processed ITS1___qITS2_:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTCCGTAGGTGAACCTGCGG -a TTYGCTGYGTTCTTCATCG -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___qITS2_/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___qITS2_/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                      98\n",
      "Reads with adapters:                        98 (100.0%)\n",
      "Reads written (passing filters):            98 (100.0%)\n",
      "\n",
      "Total basepairs processed:        22,076 bp\n",
      "Total written (filtered):         20,116 bp (91.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTCCGTAGGTGAACCTGCGG; Type: regular 5'; Length: 20; Trimmed: 98 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t98\t0.0\t2\t98\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTYGCTGYGTTCTTCATCG; Type: regular 3'; Length: 19; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1___qITS2_:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTCCGTAGGTGAACCTGCGG -a TTYGCTGYGTTCTTCATCG -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___qITS2_/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___qITS2_/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                      89\n",
      "Reads with adapters:                        89 (100.0%)\n",
      "Reads written (passing filters):            89 (100.0%)\n",
      "\n",
      "Total basepairs processed:        20,207 bp\n",
      "Total written (filtered):         18,427 bp (91.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTCCGTAGGTGAACCTGCGG; Type: regular 5'; Length: 20; Trimmed: 89 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t89\t0.0\t2\t89\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTYGCTGYGTTCTTCATCG; Type: regular 3'; Length: 19; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1___qITS2_:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTCCGTAGGTGAACCTGCGG -a TTYGCTGYGTTCTTCATCG -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___qITS2_/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___qITS2_/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                      94\n",
      "Reads with adapters:                        94 (100.0%)\n",
      "Reads written (passing filters):            94 (100.0%)\n",
      "\n",
      "Total basepairs processed:        20,748 bp\n",
      "Total written (filtered):         18,868 bp (90.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTCCGTAGGTGAACCTGCGG; Type: regular 5'; Length: 20; Trimmed: 94 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t94\t0.0\t2\t93 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTYGCTGYGTTCTTCATCG; Type: regular 3'; Length: 19; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1___qITS2_:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTCCGTAGGTGAACCTGCGG -a TTYGCTGYGTTCTTCATCG -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___qITS2_/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1___qITS2_/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     101\n",
      "Reads with adapters:                       101 (100.0%)\n",
      "Reads written (passing filters):           101 (100.0%)\n",
      "\n",
      "Total basepairs processed:        22,743 bp\n",
      "Total written (filtered):         20,723 bp (91.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTCCGTAGGTGAACCTGCGG; Type: regular 5'; Length: 20; Trimmed: 101 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t101\t0.0\t2\t101\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTYGCTGYGTTCTTCATCG; Type: regular 3'; Length: 19; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1-30F___ITS1-217R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTCCCTGCCCTTTGTACACA -a TTTCGCTGCGTTCTTCATCG -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1-30F___ITS1-217R/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1-30F___ITS1-217R/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     360\n",
      "Reads with adapters:                       360 (100.0%)\n",
      "Reads written (passing filters):           360 (100.0%)\n",
      "\n",
      "Total basepairs processed:       151,437 bp\n",
      "Total written (filtered):        144,237 bp (95.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTCCCTGCCCTTTGTACACA; Type: regular 5'; Length: 20; Trimmed: 360 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t360\t0.0\t2\t360\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTTCGCTGCGTTCTTCATCG; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1-30F___ITS1-217R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTCCCTGCCCTTTGTACACA -a TTTCGCTGCGTTCTTCATCG -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1-30F___ITS1-217R/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1-30F___ITS1-217R/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     394\n",
      "Reads with adapters:                       394 (100.0%)\n",
      "Reads written (passing filters):           394 (100.0%)\n",
      "\n",
      "Total basepairs processed:       163,976 bp\n",
      "Total written (filtered):        156,096 bp (95.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTCCCTGCCCTTTGTACACA; Type: regular 5'; Length: 20; Trimmed: 394 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t394\t0.0\t2\t394\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTTCGCTGCGTTCTTCATCG; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1-30F___ITS1-217R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTCCCTGCCCTTTGTACACA -a TTTCGCTGCGTTCTTCATCG -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1-30F___ITS1-217R/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1-30F___ITS1-217R/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     408\n",
      "Reads with adapters:                       408 (100.0%)\n",
      "Reads written (passing filters):           408 (100.0%)\n",
      "\n",
      "Total basepairs processed:       169,288 bp\n",
      "Total written (filtered):        161,128 bp (95.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTCCCTGCCCTTTGTACACA; Type: regular 5'; Length: 20; Trimmed: 408 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t408\t0.0\t2\t408\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTTCGCTGCGTTCTTCATCG; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1-30F___ITS1-217R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTCCCTGCCCTTTGTACACA -a TTTCGCTGCGTTCTTCATCG -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1-30F___ITS1-217R/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1-30F___ITS1-217R/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     381\n",
      "Reads with adapters:                       381 (100.0%)\n",
      "Reads written (passing filters):           381 (100.0%)\n",
      "\n",
      "Total basepairs processed:       161,063 bp\n",
      "Total written (filtered):        153,443 bp (95.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTCCCTGCCCTTTGTACACA; Type: regular 5'; Length: 20; Trimmed: 381 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t381\t0.0\t2\t381\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTTCGCTGCGTTCTTCATCG; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F___58A2R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a CTGCGTTCTTCATCGAT -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___58A2R/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___58A2R/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     238\n",
      "Reads with adapters:                       238 (100.0%)\n",
      "Reads written (passing filters):           238 (100.0%)\n",
      "\n",
      "Total basepairs processed:        78,366 bp\n",
      "Total written (filtered):         73,130 bp (93.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 238 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t238\t0.0\t2\t238\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CTGCGTTCTTCATCGAT; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F___58A2R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a CTGCGTTCTTCATCGAT -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___58A2R/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___58A2R/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     220\n",
      "Reads with adapters:                       220 (100.0%)\n",
      "Reads written (passing filters):           220 (100.0%)\n",
      "\n",
      "Total basepairs processed:        74,982 bp\n",
      "Total written (filtered):         70,142 bp (93.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 220 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t220\t0.0\t2\t220\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CTGCGTTCTTCATCGAT; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F___58A2R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a CTGCGTTCTTCATCGAT -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___58A2R/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___58A2R/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     221\n",
      "Reads with adapters:                       221 (100.0%)\n",
      "Reads written (passing filters):           221 (100.0%)\n",
      "\n",
      "Total basepairs processed:        78,201 bp\n",
      "Total written (filtered):         73,339 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 221 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t221\t0.0\t2\t221\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CTGCGTTCTTCATCGAT; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F___58A2R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a CTGCGTTCTTCATCGAT -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___58A2R/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___58A2R/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     223\n",
      "Reads with adapters:                       223 (100.0%)\n",
      "Reads written (passing filters):           223 (100.0%)\n",
      "\n",
      "Total basepairs processed:        77,133 bp\n",
      "Total written (filtered):         72,227 bp (93.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 223 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t223\t0.0\t2\t222 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CTGCGTTCTTCATCGAT; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F___ITS2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a GCTGCGTTCTTCATCGATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___ITS2/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___ITS2/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     211\n",
      "Reads with adapters:                       211 (100.0%)\n",
      "Reads written (passing filters):           211 (100.0%)\n",
      "\n",
      "Total basepairs processed:        71,677 bp\n",
      "Total written (filtered):         67,035 bp (93.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 211 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t211\t0.0\t2\t211\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GCTGCGTTCTTCATCGATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F___ITS2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a GCTGCGTTCTTCATCGATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___ITS2/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___ITS2/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     211\n",
      "Reads with adapters:                       211 (100.0%)\n",
      "Reads written (passing filters):           211 (100.0%)\n",
      "\n",
      "Total basepairs processed:        73,824 bp\n",
      "Total written (filtered):         69,182 bp (93.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 211 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t211\t0.0\t2\t211\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GCTGCGTTCTTCATCGATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F___ITS2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a GCTGCGTTCTTCATCGATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___ITS2/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___ITS2/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     183\n",
      "Reads with adapters:                       183 (100.0%)\n",
      "Reads written (passing filters):           183 (100.0%)\n",
      "\n",
      "Total basepairs processed:        64,679 bp\n",
      "Total written (filtered):         60,653 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 183 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t183\t0.0\t2\t183\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GCTGCGTTCTTCATCGATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F___ITS2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a GCTGCGTTCTTCATCGATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___ITS2/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___ITS2/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     207\n",
      "Reads with adapters:                       207 (100.0%)\n",
      "Reads written (passing filters):           207 (100.0%)\n",
      "\n",
      "Total basepairs processed:        69,255 bp\n",
      "Total written (filtered):         64,701 bp (93.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 207 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t207\t0.0\t2\t207\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GCTGCGTTCTTCATCGATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "ITS1F___ITS3 error - not found\n",
      "Processed ITS1F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___ITS4/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___ITS4/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     628\n",
      "Reads with adapters:                       628 (100.0%)\n",
      "Reads written (passing filters):           628 (100.0%)\n",
      "\n",
      "Total basepairs processed:       421,671 bp\n",
      "Total written (filtered):        407,855 bp (96.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 628 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t628\t0.0\t2\t628\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___ITS4/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___ITS4/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     550\n",
      "Reads with adapters:                       550 (100.0%)\n",
      "Reads written (passing filters):           550 (100.0%)\n",
      "\n",
      "Total basepairs processed:       353,894 bp\n",
      "Total written (filtered):        341,794 bp (96.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 550 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t550\t0.0\t2\t550\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___ITS4/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___ITS4/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     539\n",
      "Reads with adapters:                       539 (100.0%)\n",
      "Reads written (passing filters):           539 (100.0%)\n",
      "\n",
      "Total basepairs processed:       363,542 bp\n",
      "Total written (filtered):        351,684 bp (96.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 539 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t539\t0.0\t2\t539\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___ITS4/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___ITS4/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     603\n",
      "Reads with adapters:                       603 (100.0%)\n",
      "Reads written (passing filters):           603 (100.0%)\n",
      "\n",
      "Total basepairs processed:       393,662 bp\n",
      "Total written (filtered):        380,396 bp (96.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 603 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t603\t0.0\t2\t603\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F___LR6:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a CGCCAGTTCTGCTTACC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___LR6/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___LR6/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   1,652\n",
      "Reads with adapters:                     1,652 (100.0%)\n",
      "Reads written (passing filters):         1,652 (100.0%)\n",
      "\n",
      "Total basepairs processed:     2,872,109 bp\n",
      "Total written (filtered):      2,835,765 bp (98.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 1652 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t1652\t0.0\t2\t1652\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CGCCAGTTCTGCTTACC; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F___LR6:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a CGCCAGTTCTGCTTACC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___LR6/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___LR6/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   1,538\n",
      "Reads with adapters:                     1,538 (100.0%)\n",
      "Reads written (passing filters):         1,538 (100.0%)\n",
      "\n",
      "Total basepairs processed:     2,683,781 bp\n",
      "Total written (filtered):      2,649,945 bp (98.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 1538 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t1538\t0.0\t2\t1538\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CGCCAGTTCTGCTTACC; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F___LR6:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a CGCCAGTTCTGCTTACC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___LR6/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___LR6/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   1,547\n",
      "Reads with adapters:                     1,547 (100.0%)\n",
      "Reads written (passing filters):         1,547 (100.0%)\n",
      "\n",
      "Total basepairs processed:     2,714,946 bp\n",
      "Total written (filtered):      2,680,912 bp (98.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 1547 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t1547\t0.0\t2\t1547\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CGCCAGTTCTGCTTACC; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F___LR6:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTTGGTCATTTAGAGGAAGTAA -a CGCCAGTTCTGCTTACC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___LR6/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F___LR6/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   1,576\n",
      "Reads with adapters:                     1,576 (100.0%)\n",
      "Reads written (passing filters):         1,576 (100.0%)\n",
      "\n",
      "Total basepairs processed:     2,744,871 bp\n",
      "Total written (filtered):      2,710,199 bp (98.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 1576 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t1576\t0.0\t2\t1576\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CGCCAGTTCTGCTTACC; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F_KYO1___ITS2_KYO1:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTHGGTCATTTAGAGGAASTAA -a CTRYGTTCTTCATCGDT -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO1___ITS2_KYO1/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO1___ITS2_KYO1/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     554\n",
      "Reads with adapters:                       554 (100.0%)\n",
      "Reads written (passing filters):           554 (100.0%)\n",
      "\n",
      "Total basepairs processed:       185,965 bp\n",
      "Total written (filtered):        173,777 bp (93.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTHGGTCATTTAGAGGAASTAA; Type: regular 5'; Length: 22; Trimmed: 554 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t554\t0.0\t2\t554\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CTRYGTTCTTCATCGDT; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F_KYO1___ITS2_KYO1:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTHGGTCATTTAGAGGAASTAA -a CTRYGTTCTTCATCGDT -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO1___ITS2_KYO1/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO1___ITS2_KYO1/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     541\n",
      "Reads with adapters:                       540 (99.8%)\n",
      "Reads written (passing filters):           541 (100.0%)\n",
      "\n",
      "Total basepairs processed:       179,033 bp\n",
      "Total written (filtered):        167,152 bp (93.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTHGGTCATTTAGAGGAASTAA; Type: regular 5'; Length: 22; Trimmed: 540 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t539\t0.0\t2\t539\n",
      "23\t1\t0.0\t2\t0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CTRYGTTCTTCATCGDT; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F_KYO1___ITS2_KYO1:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTHGGTCATTTAGAGGAASTAA -a CTRYGTTCTTCATCGDT -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO1___ITS2_KYO1/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO1___ITS2_KYO1/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     514\n",
      "Reads with adapters:                       514 (100.0%)\n",
      "Reads written (passing filters):           514 (100.0%)\n",
      "\n",
      "Total basepairs processed:       177,520 bp\n",
      "Total written (filtered):        166,212 bp (93.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTHGGTCATTTAGAGGAASTAA; Type: regular 5'; Length: 22; Trimmed: 514 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t514\t0.0\t2\t514\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CTRYGTTCTTCATCGDT; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F_KYO1___ITS2_KYO1:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTHGGTCATTTAGAGGAASTAA -a CTRYGTTCTTCATCGDT -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO1___ITS2_KYO1/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO1___ITS2_KYO1/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     546\n",
      "Reads with adapters:                       546 (100.0%)\n",
      "Reads written (passing filters):           546 (100.0%)\n",
      "\n",
      "Total basepairs processed:       185,441 bp\n",
      "Total written (filtered):        173,429 bp (93.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTHGGTCATTTAGAGGAASTAA; Type: regular 5'; Length: 22; Trimmed: 546 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t546\t0.0\t2\t546\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CTRYGTTCTTCATCGDT; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F_KYO1___ITS2_KYO2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTHGGTCATTTAGAGGAASTAA -a TTYRCTRCGTTCTTCATC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO1___ITS2_KYO2/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO1___ITS2_KYO2/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     535\n",
      "Reads with adapters:                       535 (100.0%)\n",
      "Reads written (passing filters):           535 (100.0%)\n",
      "\n",
      "Total basepairs processed:       181,855 bp\n",
      "Total written (filtered):        170,085 bp (93.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTHGGTCATTTAGAGGAASTAA; Type: regular 5'; Length: 22; Trimmed: 535 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t535\t0.0\t2\t534 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTYRCTRCGTTCTTCATC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F_KYO1___ITS2_KYO2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTHGGTCATTTAGAGGAASTAA -a TTYRCTRCGTTCTTCATC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO1___ITS2_KYO2/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO1___ITS2_KYO2/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     538\n",
      "Reads with adapters:                       538 (100.0%)\n",
      "Reads written (passing filters):           538 (100.0%)\n",
      "\n",
      "Total basepairs processed:       189,990 bp\n",
      "Total written (filtered):        178,154 bp (93.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTHGGTCATTTAGAGGAASTAA; Type: regular 5'; Length: 22; Trimmed: 538 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t538\t0.0\t2\t538\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTYRCTRCGTTCTTCATC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F_KYO1___ITS2_KYO2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTHGGTCATTTAGAGGAASTAA -a TTYRCTRCGTTCTTCATC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO1___ITS2_KYO2/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO1___ITS2_KYO2/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     542\n",
      "Reads with adapters:                       542 (100.0%)\n",
      "Reads written (passing filters):           542 (100.0%)\n",
      "\n",
      "Total basepairs processed:       188,377 bp\n",
      "Total written (filtered):        176,453 bp (93.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTHGGTCATTTAGAGGAASTAA; Type: regular 5'; Length: 22; Trimmed: 542 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t542\t0.0\t2\t542\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTYRCTRCGTTCTTCATC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F_KYO1___ITS2_KYO2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g CTHGGTCATTTAGAGGAASTAA -a TTYRCTRCGTTCTTCATC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO1___ITS2_KYO2/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO1___ITS2_KYO2/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     543\n",
      "Reads with adapters:                       543 (100.0%)\n",
      "Reads written (passing filters):           543 (100.0%)\n",
      "\n",
      "Total basepairs processed:       187,845 bp\n",
      "Total written (filtered):        175,899 bp (93.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTHGGTCATTTAGAGGAASTAA; Type: regular 5'; Length: 22; Trimmed: 543 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t543\t0.0\t2\t543\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTYRCTRCGTTCTTCATC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F_KYO2___ITS2_KYO2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g TAGAGGAAGTAAAAGTCGTAA -a TTYRCTRCGTTCTTCATC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO2___ITS2_KYO2/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO2___ITS2_KYO2/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     394\n",
      "Reads with adapters:                       394 (100.0%)\n",
      "Reads written (passing filters):           394 (100.0%)\n",
      "\n",
      "Total basepairs processed:       123,994 bp\n",
      "Total written (filtered):        115,720 bp (93.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TAGAGGAAGTAAAAGTCGTAA; Type: regular 5'; Length: 21; Trimmed: 394 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "21\t394\t0.0\t2\t394\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTYRCTRCGTTCTTCATC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F_KYO2___ITS2_KYO2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g TAGAGGAAGTAAAAGTCGTAA -a TTYRCTRCGTTCTTCATC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO2___ITS2_KYO2/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO2___ITS2_KYO2/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     380\n",
      "Reads with adapters:                       380 (100.0%)\n",
      "Reads written (passing filters):           380 (100.0%)\n",
      "\n",
      "Total basepairs processed:       116,077 bp\n",
      "Total written (filtered):        108,097 bp (93.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TAGAGGAAGTAAAAGTCGTAA; Type: regular 5'; Length: 21; Trimmed: 380 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "21\t380\t0.0\t2\t380\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTYRCTRCGTTCTTCATC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F_KYO2___ITS2_KYO2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g TAGAGGAAGTAAAAGTCGTAA -a TTYRCTRCGTTCTTCATC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO2___ITS2_KYO2/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO2___ITS2_KYO2/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     383\n",
      "Reads with adapters:                       382 (99.7%)\n",
      "Reads written (passing filters):           383 (100.0%)\n",
      "\n",
      "Total basepairs processed:       125,767 bp\n",
      "Total written (filtered):        117,745 bp (93.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TAGAGGAAGTAAAAGTCGTAA; Type: regular 5'; Length: 21; Trimmed: 382 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "21\t382\t0.0\t2\t382\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTYRCTRCGTTCTTCATC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1F_KYO2___ITS2_KYO2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g TAGAGGAAGTAAAAGTCGTAA -a TTYRCTRCGTTCTTCATC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO2___ITS2_KYO2/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1F_KYO2___ITS2_KYO2/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     379\n",
      "Reads with adapters:                       379 (100.0%)\n",
      "Reads written (passing filters):           379 (100.0%)\n",
      "\n",
      "Total basepairs processed:       119,042 bp\n",
      "Total written (filtered):        111,083 bp (93.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TAGAGGAAGTAAAAGTCGTAA; Type: regular 5'; Length: 21; Trimmed: 379 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "21\t379\t0.0\t2\t379\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTYRCTRCGTTCTTCATC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1FI2___5.8S/1/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1FI2___5.8S/2/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1FI2___5.8S/3/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1FI2___5.8S/4/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1FI2___ITS2/1/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1FI2___ITS2/2/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1FI2___ITS2/3/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1FI2___ITS2/4/samp_all_seqs.fa does not exist. Skipping...\n",
      "Processed ITS1Fngs___ITS2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGTCATTTAGAGGAAGTAA -a GCTGCGTTCTTCATCGATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1Fngs___ITS2/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1Fngs___ITS2/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     256\n",
      "Reads with adapters:                       256 (100.0%)\n",
      "Reads written (passing filters):           256 (100.0%)\n",
      "\n",
      "Total basepairs processed:        85,598 bp\n",
      "Total written (filtered):         80,733 bp (94.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 19; Trimmed: 256 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t255\t0.0\t1\t255\n",
      "20\t1\t0.0\t1\t0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GCTGCGTTCTTCATCGATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1Fngs___ITS2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGTCATTTAGAGGAAGTAA -a GCTGCGTTCTTCATCGATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1Fngs___ITS2/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1Fngs___ITS2/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     273\n",
      "Reads with adapters:                       273 (100.0%)\n",
      "Reads written (passing filters):           273 (100.0%)\n",
      "\n",
      "Total basepairs processed:        94,149 bp\n",
      "Total written (filtered):         88,963 bp (94.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 19; Trimmed: 273 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t1\t0.0\t1\t0 1\n",
      "19\t272\t0.0\t1\t272\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GCTGCGTTCTTCATCGATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1Fngs___ITS2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGTCATTTAGAGGAAGTAA -a GCTGCGTTCTTCATCGATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1Fngs___ITS2/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1Fngs___ITS2/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     278\n",
      "Reads with adapters:                       278 (100.0%)\n",
      "Reads written (passing filters):           278 (100.0%)\n",
      "\n",
      "Total basepairs processed:        97,680 bp\n",
      "Total written (filtered):         92,398 bp (94.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 19; Trimmed: 278 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t278\t0.0\t1\t278\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GCTGCGTTCTTCATCGATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS1Fngs___ITS2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGTCATTTAGAGGAAGTAA -a GCTGCGTTCTTCATCGATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1Fngs___ITS2/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1Fngs___ITS2/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     252\n",
      "Reads with adapters:                       252 (100.0%)\n",
      "Reads written (passing filters):           252 (100.0%)\n",
      "\n",
      "Total basepairs processed:        84,644 bp\n",
      "Total written (filtered):         79,856 bp (94.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 19; Trimmed: 252 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t252\t0.0\t1\t252\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GCTGCGTTCTTCATCGATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1ngs___ITS2/1/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1ngs___ITS2/2/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1ngs___ITS2/3/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS1ngs___ITS2/4/samp_all_seqs.fa does not exist. Skipping...\n",
      "Processed ITS2F___ITS2R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GCATCGATGAAGAACGC -a CCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS2F___ITS2R/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS2F___ITS2R/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     521\n",
      "Reads with adapters:                       521 (100.0%)\n",
      "Reads written (passing filters):           521 (100.0%)\n",
      "\n",
      "Total basepairs processed:       187,764 bp\n",
      "Total written (filtered):        178,907 bp (95.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GCATCGATGAAGAACGC; Type: regular 5'; Length: 17; Trimmed: 521 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t521\t0.0\t1\t521\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTCCGCTTATTGATATGC; Type: regular 3'; Length: 19; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS2F___ITS2R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GCATCGATGAAGAACGC -a CCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS2F___ITS2R/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS2F___ITS2R/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     511\n",
      "Reads with adapters:                       511 (100.0%)\n",
      "Reads written (passing filters):           511 (100.0%)\n",
      "\n",
      "Total basepairs processed:       183,392 bp\n",
      "Total written (filtered):        174,705 bp (95.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GCATCGATGAAGAACGC; Type: regular 5'; Length: 17; Trimmed: 511 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t511\t0.0\t1\t511\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTCCGCTTATTGATATGC; Type: regular 3'; Length: 19; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS2F___ITS2R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GCATCGATGAAGAACGC -a CCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS2F___ITS2R/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS2F___ITS2R/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     460\n",
      "Reads with adapters:                       460 (100.0%)\n",
      "Reads written (passing filters):           460 (100.0%)\n",
      "\n",
      "Total basepairs processed:       164,281 bp\n",
      "Total written (filtered):        156,461 bp (95.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GCATCGATGAAGAACGC; Type: regular 5'; Length: 17; Trimmed: 460 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t460\t0.0\t1\t460\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTCCGCTTATTGATATGC; Type: regular 3'; Length: 19; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS2F___ITS2R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GCATCGATGAAGAACGC -a CCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS2F___ITS2R/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS2F___ITS2R/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     512\n",
      "Reads with adapters:                       512 (100.0%)\n",
      "Reads written (passing filters):           512 (100.0%)\n",
      "\n",
      "Total basepairs processed:       183,512 bp\n",
      "Total written (filtered):        174,808 bp (95.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GCATCGATGAAGAACGC; Type: regular 5'; Length: 17; Trimmed: 512 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t512\t0.0\t1\t512\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTCCGCTTATTGATATGC; Type: regular 3'; Length: 19; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS3___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GCATCGATGAAGAACGCAGC -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3___ITS4/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3___ITS4/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     436\n",
      "Reads with adapters:                       436 (100.0%)\n",
      "Reads written (passing filters):           436 (100.0%)\n",
      "\n",
      "Total basepairs processed:       157,962 bp\n",
      "Total written (filtered):        149,242 bp (94.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GCATCGATGAAGAACGCAGC; Type: regular 5'; Length: 20; Trimmed: 436 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t436\t0.0\t2\t436\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS3___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GCATCGATGAAGAACGCAGC -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3___ITS4/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3___ITS4/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     463\n",
      "Reads with adapters:                       463 (100.0%)\n",
      "Reads written (passing filters):           463 (100.0%)\n",
      "\n",
      "Total basepairs processed:       167,067 bp\n",
      "Total written (filtered):        157,807 bp (94.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GCATCGATGAAGAACGCAGC; Type: regular 5'; Length: 20; Trimmed: 463 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t463\t0.0\t2\t463\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS3___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GCATCGATGAAGAACGCAGC -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3___ITS4/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3___ITS4/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     473\n",
      "Reads with adapters:                       473 (100.0%)\n",
      "Reads written (passing filters):           473 (100.0%)\n",
      "\n",
      "Total basepairs processed:       170,040 bp\n",
      "Total written (filtered):        160,580 bp (94.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GCATCGATGAAGAACGCAGC; Type: regular 5'; Length: 20; Trimmed: 473 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t473\t0.0\t2\t473\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS3___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GCATCGATGAAGAACGCAGC -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3___ITS4/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3___ITS4/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     433\n",
      "Reads with adapters:                       433 (100.0%)\n",
      "Reads written (passing filters):           433 (100.0%)\n",
      "\n",
      "Total basepairs processed:       156,375 bp\n",
      "Total written (filtered):        147,715 bp (94.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GCATCGATGAAGAACGCAGC; Type: regular 5'; Length: 20; Trimmed: 433 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t433\t0.0\t2\t433\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS3_KYO2___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GATGAAGAACGYAGYRAA -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3_KYO2___ITS4/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3_KYO2___ITS4/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     567\n",
      "Reads with adapters:                       567 (100.0%)\n",
      "Reads written (passing filters):           567 (100.0%)\n",
      "\n",
      "Total basepairs processed:       203,932 bp\n",
      "Total written (filtered):        193,726 bp (95.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GATGAAGAACGYAGYRAA; Type: regular 5'; Length: 18; Trimmed: 567 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-18 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t567\t0.0\t1\t567\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS3_KYO2___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GATGAAGAACGYAGYRAA -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3_KYO2___ITS4/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3_KYO2___ITS4/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     579\n",
      "Reads with adapters:                       579 (100.0%)\n",
      "Reads written (passing filters):           579 (100.0%)\n",
      "\n",
      "Total basepairs processed:       208,757 bp\n",
      "Total written (filtered):        198,335 bp (95.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GATGAAGAACGYAGYRAA; Type: regular 5'; Length: 18; Trimmed: 579 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-18 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t579\t0.0\t1\t579\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS3_KYO2___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GATGAAGAACGYAGYRAA -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3_KYO2___ITS4/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3_KYO2___ITS4/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     557\n",
      "Reads with adapters:                       557 (100.0%)\n",
      "Reads written (passing filters):           557 (100.0%)\n",
      "\n",
      "Total basepairs processed:       201,700 bp\n",
      "Total written (filtered):        191,674 bp (95.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GATGAAGAACGYAGYRAA; Type: regular 5'; Length: 18; Trimmed: 557 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-18 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t557\t0.0\t1\t557\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS3_KYO2___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GATGAAGAACGYAGYRAA -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3_KYO2___ITS4/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3_KYO2___ITS4/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     546\n",
      "Reads with adapters:                       546 (100.0%)\n",
      "Reads written (passing filters):           546 (100.0%)\n",
      "\n",
      "Total basepairs processed:       196,400 bp\n",
      "Total written (filtered):        186,572 bp (95.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GATGAAGAACGYAGYRAA; Type: regular 5'; Length: 18; Trimmed: 546 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-18 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t546\t0.0\t1\t546\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS3_KYO2___ITS4_KYO3:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GATGAAGAACGYAGYRAA -a CTBTTVCCKCTTCACTCG -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3_KYO2___ITS4_KYO3/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3_KYO2___ITS4_KYO3/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   1,221\n",
      "Reads with adapters:                     1,221 (100.0%)\n",
      "Reads written (passing filters):         1,221 (100.0%)\n",
      "\n",
      "Total basepairs processed:       497,153 bp\n",
      "Total written (filtered):        475,175 bp (95.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GATGAAGAACGYAGYRAA; Type: regular 5'; Length: 18; Trimmed: 1221 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-18 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t1221\t0.0\t1\t1221\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CTBTTVCCKCTTCACTCG; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS3_KYO2___ITS4_KYO3:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GATGAAGAACGYAGYRAA -a CTBTTVCCKCTTCACTCG -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3_KYO2___ITS4_KYO3/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3_KYO2___ITS4_KYO3/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   1,161\n",
      "Reads with adapters:                     1,161 (100.0%)\n",
      "Reads written (passing filters):         1,161 (100.0%)\n",
      "\n",
      "Total basepairs processed:       471,782 bp\n",
      "Total written (filtered):        450,884 bp (95.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GATGAAGAACGYAGYRAA; Type: regular 5'; Length: 18; Trimmed: 1161 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-18 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t1161\t0.0\t1\t1161\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CTBTTVCCKCTTCACTCG; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS3_KYO2___ITS4_KYO3:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GATGAAGAACGYAGYRAA -a CTBTTVCCKCTTCACTCG -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3_KYO2___ITS4_KYO3/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3_KYO2___ITS4_KYO3/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   1,258\n",
      "Reads with adapters:                     1,258 (100.0%)\n",
      "Reads written (passing filters):         1,258 (100.0%)\n",
      "\n",
      "Total basepairs processed:       511,105 bp\n",
      "Total written (filtered):        488,461 bp (95.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GATGAAGAACGYAGYRAA; Type: regular 5'; Length: 18; Trimmed: 1258 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-18 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t1258\t0.0\t1\t1258\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CTBTTVCCKCTTCACTCG; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS3_KYO2___ITS4_KYO3:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GATGAAGAACGYAGYRAA -a CTBTTVCCKCTTCACTCG -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3_KYO2___ITS4_KYO3/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS3_KYO2___ITS4_KYO3/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   1,160\n",
      "Reads with adapters:                     1,160 (100.0%)\n",
      "Reads written (passing filters):         1,160 (100.0%)\n",
      "\n",
      "Total basepairs processed:       475,826 bp\n",
      "Total written (filtered):        454,946 bp (95.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GATGAAGAACGYAGYRAA; Type: regular 5'; Length: 18; Trimmed: 1160 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-18 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t1160\t0.0\t1\t1160\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CTBTTVCCKCTTCACTCG; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS5___5.8S_fungi:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGAAGTAAAAGTCGTAACAAGG -a CAAGAGATCCGTTGTTGAAAGTT -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___5.8S_fungi/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___5.8S_fungi/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     194\n",
      "Reads with adapters:                       194 (100.0%)\n",
      "Reads written (passing filters):           194 (100.0%)\n",
      "\n",
      "Total basepairs processed:        61,719 bp\n",
      "Total written (filtered):         57,451 bp (93.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGAAGTAAAAGTCGTAACAAGG; Type: regular 5'; Length: 22; Trimmed: 194 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t194\t0.0\t2\t193 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CAAGAGATCCGTTGTTGAAAGTT; Type: regular 3'; Length: 23; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS5___5.8S_fungi:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGAAGTAAAAGTCGTAACAAGG -a CAAGAGATCCGTTGTTGAAAGTT -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___5.8S_fungi/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___5.8S_fungi/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     185\n",
      "Reads with adapters:                       185 (100.0%)\n",
      "Reads written (passing filters):           185 (100.0%)\n",
      "\n",
      "Total basepairs processed:        56,979 bp\n",
      "Total written (filtered):         52,909 bp (92.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGAAGTAAAAGTCGTAACAAGG; Type: regular 5'; Length: 22; Trimmed: 185 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t185\t0.0\t2\t185\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CAAGAGATCCGTTGTTGAAAGTT; Type: regular 3'; Length: 23; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS5___5.8S_fungi:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGAAGTAAAAGTCGTAACAAGG -a CAAGAGATCCGTTGTTGAAAGTT -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___5.8S_fungi/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___5.8S_fungi/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     190\n",
      "Reads with adapters:                       190 (100.0%)\n",
      "Reads written (passing filters):           190 (100.0%)\n",
      "\n",
      "Total basepairs processed:        56,321 bp\n",
      "Total written (filtered):         52,141 bp (92.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGAAGTAAAAGTCGTAACAAGG; Type: regular 5'; Length: 22; Trimmed: 190 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t190\t0.0\t2\t190\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CAAGAGATCCGTTGTTGAAAGTT; Type: regular 3'; Length: 23; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS5___5.8S_fungi:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGAAGTAAAAGTCGTAACAAGG -a CAAGAGATCCGTTGTTGAAAGTT -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___5.8S_fungi/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___5.8S_fungi/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     190\n",
      "Reads with adapters:                       190 (100.0%)\n",
      "Reads written (passing filters):           190 (100.0%)\n",
      "\n",
      "Total basepairs processed:        58,044 bp\n",
      "Total written (filtered):         53,864 bp (92.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGAAGTAAAAGTCGTAACAAGG; Type: regular 5'; Length: 22; Trimmed: 190 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t190\t0.0\t2\t190\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CAAGAGATCCGTTGTTGAAAGTT; Type: regular 3'; Length: 23; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS5___ITS2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGAAGTAAAAGTCGTAACAAGG -a GCTGCGTTCTTCATCGATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS2/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS2/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     282\n",
      "Reads with adapters:                       282 (100.0%)\n",
      "Reads written (passing filters):           282 (100.0%)\n",
      "\n",
      "Total basepairs processed:        91,526 bp\n",
      "Total written (filtered):         85,322 bp (93.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGAAGTAAAAGTCGTAACAAGG; Type: regular 5'; Length: 22; Trimmed: 282 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t282\t0.0\t2\t282\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GCTGCGTTCTTCATCGATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS5___ITS2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGAAGTAAAAGTCGTAACAAGG -a GCTGCGTTCTTCATCGATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS2/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS2/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     290\n",
      "Reads with adapters:                       290 (100.0%)\n",
      "Reads written (passing filters):           290 (100.0%)\n",
      "\n",
      "Total basepairs processed:        94,726 bp\n",
      "Total written (filtered):         88,346 bp (93.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGAAGTAAAAGTCGTAACAAGG; Type: regular 5'; Length: 22; Trimmed: 290 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t290\t0.0\t2\t290\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GCTGCGTTCTTCATCGATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS5___ITS2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGAAGTAAAAGTCGTAACAAGG -a GCTGCGTTCTTCATCGATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS2/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS2/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     283\n",
      "Reads with adapters:                       283 (100.0%)\n",
      "Reads written (passing filters):           283 (100.0%)\n",
      "\n",
      "Total basepairs processed:        91,346 bp\n",
      "Total written (filtered):         85,120 bp (93.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGAAGTAAAAGTCGTAACAAGG; Type: regular 5'; Length: 22; Trimmed: 283 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t283\t0.0\t2\t283\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GCTGCGTTCTTCATCGATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS5___ITS2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGAAGTAAAAGTCGTAACAAGG -a GCTGCGTTCTTCATCGATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS2/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS2/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     300\n",
      "Reads with adapters:                       300 (100.0%)\n",
      "Reads written (passing filters):           300 (100.0%)\n",
      "\n",
      "Total basepairs processed:        95,057 bp\n",
      "Total written (filtered):         88,457 bp (93.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGAAGTAAAAGTCGTAACAAGG; Type: regular 5'; Length: 22; Trimmed: 300 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t300\t0.0\t2\t300\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GCTGCGTTCTTCATCGATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS5___ITS2_KYO2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGAAGTAAAAGTCGTAACAAGG -a TTYRCTRCGTTCTTCATC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS2_KYO2/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS2_KYO2/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     405\n",
      "Reads with adapters:                       405 (100.0%)\n",
      "Reads written (passing filters):           405 (100.0%)\n",
      "\n",
      "Total basepairs processed:       131,583 bp\n",
      "Total written (filtered):        122,673 bp (93.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGAAGTAAAAGTCGTAACAAGG; Type: regular 5'; Length: 22; Trimmed: 405 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t405\t0.0\t2\t405\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTYRCTRCGTTCTTCATC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS5___ITS2_KYO2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGAAGTAAAAGTCGTAACAAGG -a TTYRCTRCGTTCTTCATC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS2_KYO2/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS2_KYO2/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     396\n",
      "Reads with adapters:                       396 (100.0%)\n",
      "Reads written (passing filters):           396 (100.0%)\n",
      "\n",
      "Total basepairs processed:       122,852 bp\n",
      "Total written (filtered):        114,140 bp (92.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGAAGTAAAAGTCGTAACAAGG; Type: regular 5'; Length: 22; Trimmed: 396 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t396\t0.0\t2\t396\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTYRCTRCGTTCTTCATC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS5___ITS2_KYO2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGAAGTAAAAGTCGTAACAAGG -a TTYRCTRCGTTCTTCATC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS2_KYO2/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS2_KYO2/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     426\n",
      "Reads with adapters:                       425 (99.8%)\n",
      "Reads written (passing filters):           426 (100.0%)\n",
      "\n",
      "Total basepairs processed:       130,280 bp\n",
      "Total written (filtered):        120,930 bp (92.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGAAGTAAAAGTCGTAACAAGG; Type: regular 5'; Length: 22; Trimmed: 425 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t425\t0.0\t2\t424 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTYRCTRCGTTCTTCATC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS5___ITS2_KYO2:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGAAGTAAAAGTCGTAACAAGG -a TTYRCTRCGTTCTTCATC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS2_KYO2/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS2_KYO2/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     387\n",
      "Reads with adapters:                       387 (100.0%)\n",
      "Reads written (passing filters):           387 (100.0%)\n",
      "\n",
      "Total basepairs processed:       122,404 bp\n",
      "Total written (filtered):        113,891 bp (93.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGAAGTAAAAGTCGTAACAAGG; Type: regular 5'; Length: 22; Trimmed: 387 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "21\t1\t0.0\t2\t0 1\n",
      "22\t386\t0.0\t2\t386\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TTYRCTRCGTTCTTCATC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS5___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGAAGTAAAAGTCGTAACAAGG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS4/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS4/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     676\n",
      "Reads with adapters:                       676 (100.0%)\n",
      "Reads written (passing filters):           676 (100.0%)\n",
      "\n",
      "Total basepairs processed:       438,286 bp\n",
      "Total written (filtered):        423,414 bp (96.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGAAGTAAAAGTCGTAACAAGG; Type: regular 5'; Length: 22; Trimmed: 676 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t676\t0.0\t2\t676\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS5___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGAAGTAAAAGTCGTAACAAGG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS4/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS4/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     655\n",
      "Reads with adapters:                       655 (100.0%)\n",
      "Reads written (passing filters):           655 (100.0%)\n",
      "\n",
      "Total basepairs processed:       422,074 bp\n",
      "Total written (filtered):        407,664 bp (96.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGAAGTAAAAGTCGTAACAAGG; Type: regular 5'; Length: 22; Trimmed: 655 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t655\t0.0\t2\t655\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS5___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGAAGTAAAAGTCGTAACAAGG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS4/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS4/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     707\n",
      "Reads with adapters:                       707 (100.0%)\n",
      "Reads written (passing filters):           707 (100.0%)\n",
      "\n",
      "Total basepairs processed:       458,376 bp\n",
      "Total written (filtered):        442,822 bp (96.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGAAGTAAAAGTCGTAACAAGG; Type: regular 5'; Length: 22; Trimmed: 707 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t707\t0.0\t2\t707\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS5___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GGAAGTAAAAGTCGTAACAAGG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS4/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS5___ITS4/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     680\n",
      "Reads with adapters:                       680 (100.0%)\n",
      "Reads written (passing filters):           680 (100.0%)\n",
      "\n",
      "Total basepairs processed:       445,875 bp\n",
      "Total written (filtered):        430,915 bp (96.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GGAAGTAAAAGTCGTAACAAGG; Type: regular 5'; Length: 22; Trimmed: 680 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t680\t0.0\t2\t679 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS7o___ITS4/1/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS7o___ITS4/2/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS7o___ITS4/3/samp_all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS7o___ITS4/4/samp_all_seqs.fa does not exist. Skipping...\n",
      "Processed ITS86F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGAATCATCGAATCTTTGAA -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS86F___ITS4/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS86F___ITS4/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     326\n",
      "Reads with adapters:                       326 (100.0%)\n",
      "Reads written (passing filters):           326 (100.0%)\n",
      "\n",
      "Total basepairs processed:       101,111 bp\n",
      "Total written (filtered):         94,265 bp (93.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGAATCATCGAATCTTTGAA; Type: regular 5'; Length: 21; Trimmed: 326 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "21\t326\t0.0\t2\t326\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS86F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGAATCATCGAATCTTTGAA -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS86F___ITS4/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS86F___ITS4/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     329\n",
      "Reads with adapters:                       329 (100.0%)\n",
      "Reads written (passing filters):           329 (100.0%)\n",
      "\n",
      "Total basepairs processed:       101,917 bp\n",
      "Total written (filtered):         95,008 bp (93.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGAATCATCGAATCTTTGAA; Type: regular 5'; Length: 21; Trimmed: 329 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "21\t329\t0.0\t2\t329\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS86F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGAATCATCGAATCTTTGAA -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS86F___ITS4/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS86F___ITS4/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     304\n",
      "Reads with adapters:                       304 (100.0%)\n",
      "Reads written (passing filters):           304 (100.0%)\n",
      "\n",
      "Total basepairs processed:        94,774 bp\n",
      "Total written (filtered):         88,390 bp (93.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGAATCATCGAATCTTTGAA; Type: regular 5'; Length: 21; Trimmed: 304 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "21\t304\t0.0\t2\t304\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS86F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGAATCATCGAATCTTTGAA -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS86F___ITS4/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS86F___ITS4/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     314\n",
      "Reads with adapters:                       314 (100.0%)\n",
      "Reads written (passing filters):           314 (100.0%)\n",
      "\n",
      "Total basepairs processed:        96,730 bp\n",
      "Total written (filtered):         90,136 bp (93.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGAATCATCGAATCTTTGAA; Type: regular 5'; Length: 21; Trimmed: 314 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "21\t314\t0.0\t2\t314\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "ITS86F___ITS4-Tul error - not found\n",
      "ITS9___ITS4 error - not found\n",
      "Processed ITS9MUNngs___ITS4ngsUni:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g TACACACCGCCCGTCG -a CCTSCSCTTANTDATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS9MUNngs___ITS4ngsUni/1/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS9MUNngs___ITS4ngsUni/1/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     956\n",
      "Reads with adapters:                       956 (100.0%)\n",
      "Reads written (passing filters):           956 (100.0%)\n",
      "\n",
      "Total basepairs processed:       640,789 bp\n",
      "Total written (filtered):        625,493 bp (97.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TACACACCGCCCGTCG; Type: regular 5'; Length: 16; Trimmed: 956 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-16 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "16\t956\t0.0\t1\t956\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTSCSCTTANTDATATGC; Type: regular 3'; Length: 19; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS9MUNngs___ITS4ngsUni:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g TACACACCGCCCGTCG -a CCTSCSCTTANTDATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS9MUNngs___ITS4ngsUni/2/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS9MUNngs___ITS4ngsUni/2/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     935\n",
      "Reads with adapters:                       935 (100.0%)\n",
      "Reads written (passing filters):           935 (100.0%)\n",
      "\n",
      "Total basepairs processed:       638,173 bp\n",
      "Total written (filtered):        623,213 bp (97.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TACACACCGCCCGTCG; Type: regular 5'; Length: 16; Trimmed: 935 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-16 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "16\t935\t0.0\t1\t935\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTSCSCTTANTDATATGC; Type: regular 3'; Length: 19; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS9MUNngs___ITS4ngsUni:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g TACACACCGCCCGTCG -a CCTSCSCTTANTDATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS9MUNngs___ITS4ngsUni/3/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS9MUNngs___ITS4ngsUni/3/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     906\n",
      "Reads with adapters:                       906 (100.0%)\n",
      "Reads written (passing filters):           906 (100.0%)\n",
      "\n",
      "Total basepairs processed:       614,371 bp\n",
      "Total written (filtered):        599,875 bp (97.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TACACACCGCCCGTCG; Type: regular 5'; Length: 16; Trimmed: 906 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-16 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "16\t906\t0.0\t1\t906\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTSCSCTTANTDATATGC; Type: regular 3'; Length: 19; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed ITS9MUNngs___ITS4ngsUni:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g TACACACCGCCCGTCG -a CCTSCSCTTANTDATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS9MUNngs___ITS4ngsUni/4/samp_all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/ITS9MUNngs___ITS4ngsUni/4/samp_all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     862\n",
      "Reads with adapters:                       862 (100.0%)\n",
      "Reads written (passing filters):           862 (100.0%)\n",
      "\n",
      "Total basepairs processed:       591,651 bp\n",
      "Total written (filtered):        577,859 bp (97.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TACACACCGCCCGTCG; Type: regular 5'; Length: 16; Trimmed: 862 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-16 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "16\t862\t0.0\t1\t862\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTSCSCTTANTDATATGC; Type: regular 3'; Length: 19; Trimmed: 0 times\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq\n",
    "import os\n",
    "from subprocess import run\n",
    "\n",
    "# Function to sanitize sequence IDs and descriptions\n",
    "def sanitize_fasta_records(records):\n",
    "    sanitized_records = []\n",
    "    for record in records:\n",
    "        sanitized_id = ''.join(c if c.isascii() or c in \"-_.\" else \"_\" for c in record.id)\n",
    "        sanitized_description = ''.join(c if c.isascii() or c in \"-_.\" else \"_\" for c in record.description)\n",
    "        sanitized_records.append(SeqRecord(Seq(str(record.seq)), id=sanitized_id, description=sanitized_description))\n",
    "    return sanitized_records\n",
    "\n",
    "# Iterate over each primer and its corresponding file\n",
    "for _, row in primers_df_filtered.iterrows():\n",
    "    primer_name = row[\"better_names\"]\n",
    "    primer_seq = row[\"primers sequences\"]\n",
    "    #if primer_name != \"gITS7___ITS4\": continue #########!!!!!!!!!!!\n",
    "\n",
    "    simera_dir = f\"../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/{primer_name}/\"\n",
    "    try:\n",
    "        for time_dir in os.listdir(simera_dir):\n",
    "            time_dir = simera_dir + time_dir\n",
    "            \n",
    "            input_file = time_dir + \"/samp_all_seqs.fa\"\n",
    "            sanitized_file = time_dir + \"/samp_all_seqs_sanitized.fa\"\n",
    "            output_file = time_dir + \"/samp_all_seqs_cutadapt.fa\"\n",
    "            \n",
    "            if not os.path.exists(input_file):\n",
    "                print(f\"Input file {input_file} does not exist. Skipping...\")\n",
    "                continue\n",
    "            # Sanitize the input FASTA file\n",
    "            records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "            sanitized_records = sanitize_fasta_records(records)\n",
    "            with open(sanitized_file, \"w\") as handle:\n",
    "                SeqIO.write(sanitized_records, handle, \"fasta\")\n",
    "            \n",
    "            # Run cutadapt to cut primers\n",
    "            command = [\n",
    "                \"cutadapt\",\n",
    "                \"-g\", primer_seq.split(\"/\")[0].strip(),  # Forward primer\n",
    "                \"-a\", primer_seq.split(\"/\")[1].strip(),  # Reverse primer\n",
    "                \"-o\", output_file,\n",
    "                sanitized_file\n",
    "            ]\n",
    "            result = run(command, capture_output=True, text=True)\n",
    "            \n",
    "            # Print the result for debugging\n",
    "            print(f\"Processed {primer_name}:\")\n",
    "            print(result.stdout)\n",
    "            print(result.stderr)\n",
    "    except: print(primer_name, \"error - not found\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### RUN vsearch uchime on Simera data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "35456 len\n",
      "[('chimeraREV_253_1_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTATTGAATAAACCTGATGTGGTTGTTGC...AAC')),\n",
      " ('chimeraREV_287_1_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTATCGAAATCGAAGAGGTGGAGGGACTG...GAA')),\n",
      " ('chimeraFWD_150_5_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTATTGAATAAACCTGATGTGGTTGTAGC...AAC')),\n",
      " ('chimeraFWD_364_1_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTATTGAATAAACCTGGTGGGTTGTAGCT...AAC')),\n",
      " ('chimeraREV_164_2_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTACCAATAAGGCTACCTTGGACCAAAAG...AAC')),\n",
      " ('Agaricaceae_sp|OM972416|SH1182565.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Agaricales;f__Agaricaceae;g__unidentified;s__Agaricaceae_sp_1_12_4_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTATTGAATAAACCTGGTGGGTTGTCGCT...CAA')),\n",
      " ('chimeraREV_147_2_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTATCGATTTCAAAGGATGTGCTGGCTGC...AAC')),\n",
      " ('chimeraREV_372_1_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTATTGAATTAAAACTTGGTGGGTTGTTG...AAC')),\n",
      " ('chimeraFWD_351_1_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTACAGTTATTAGATGTTGCCAGCGCTTA...AAC')),\n",
      " ('Mycena_sp|OM987391|SH1126026.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Agaricales;f__Mycenaceae;g__Mycena;s__Mycena_sp_1_66_53_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTATTGAATACGATTGGAACTGATGCTGG...AAC')),\n",
      " ('chimeraREV_348_1_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTACCGAGCTAGGGTCCTAACATGGCCCA...AAC')),\n",
      " ('chimeraFWD_363_1_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTATTGAATTTATGGAGGGGGTTTTGCTG...AAC')),\n",
      " ('chimeraFWD_363_1_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTATTGTAACACAGGGGGTGCCGGGGCTG...AAC')),\n",
      " ('Fuscoporia_ferruginosa|OM972420|SH0939575.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Hymenochaetales;f__Hymenochaetaceae;g__Fuscoporia;s__Fuscoporia_ferruginosa_1_125_15_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTATCGAATTTTGAAGTCCGTCTTGAGCT...AAC')),\n",
      " ('Clavulinopsis_sp|OM972357|SH0895475.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Agaricales;f__Clavariaceae;g__Clavulinopsis;s__Clavulinopsis_sp_1_122_12_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTATCGAATCCAAAGGGATGTTGCTGGTC...AAC')),\n",
      " ('chimeraREV_290_1_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTAAAAGATACATGGCTACGCATCAAAGC...AAC')),\n",
      " ('chimeraFWD_271_2_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTATCGAATTTACAAGGGAGAAGGAAAGG...AAC')),\n",
      " ('chimeraFWD_138_7_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTACTGAGAGAGGAGCTTCGTGCTCCCGG...AAC')),\n",
      " ('chimeraFWD_223_2_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGAACCTGCGGAAGGATCATTAAAGTATGATTTGGATTCTTTTATGAG...AAC')),\n",
      " ('chimeraREV_353_1_-_18S-F___5.8S-R',\n",
      "  Seq('CGTAGGTGACCTGCGGAAGGATCATTATTGAATTCCAAGTAGAGTTGTTGCTGG...AAC'))]\n",
      "35456 len after filtering\n",
      "[('chimeraREV_253_1;size=1;18S-F___5.8S-R', Seq('CGTAGGTGAACCTGCGGAAGGATCATTATTGAATAAACCTGATGTGGTTGTTGC...AAC')), ('chimeraREV_287_1;size=1;18S-F___5.8S-R', Seq('CGTAGGTGAACCTGCGGAAGGATCATTATCGAAATCGAAGAGGTGGAGGGACTG...GAA'))]\n",
      "output_file: ../my_data/all_simera_for_uchime_for_uchime.fa\n"
     ]
    }
   ],
   "source": [
    "#simera_fasta_path = f\"../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/\"\n",
    "#simera_fasta_file = f\"gITS7___ITS4/20250325_182506/all_seqs_cutadapt.fa\"\n",
    "simera_fasta_path = \"\"\n",
    "simera_fasta_file = f\"../my_data/all_simera_for_uchime.fasta\"\n",
    "from pprint import pprint\n",
    "simera_seqs = [(i.id,i.seq) for i in SeqIO.parse(simera_fasta_path+simera_fasta_file, \"fasta\")]\n",
    "print(len(simera_seqs),\"len\")\n",
    "pprint(simera_seqs[:20])\n",
    "#simera_seqs = list(filter(lambda x: int(x[0].split(\"_\")[-1]) > 0, simera_seqs))\n",
    "simera_seqs = list(filter(lambda x: int(x[0].split(\"_-_\")[0].split(\"_\")[-1]) > 0, simera_seqs))\n",
    "\n",
    "print(len(simera_seqs),\"len after filtering\")\n",
    "\n",
    "\n",
    "simera_seqs_for_uchime=list(map(lambda x: (\"\".join(x[0].split(\"_-_\")[0])+\";size=\"+x[0].split(\"_-_\")[0].split(\"_\")[-1]+\";\"+x[0].split(\"_-_\")[1],x[1]),simera_seqs))\n",
    "\n",
    "print(simera_seqs_for_uchime[:2])\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "\n",
    "output_file = simera_fasta_path + \".\".join(simera_fasta_file.split(\".\")[:-1]) + \"_for_uchime.fa\"\n",
    "seq_records = []\n",
    "for header, seq in simera_seqs_for_uchime:\n",
    "    record = SeqRecord(seq, id=header, description=\"\")\n",
    "    seq_records.append(record)\n",
    "with open(output_file, 'w') as output_handle:\n",
    "    SeqIO.write(seq_records, output_handle, 'fasta-2line')\n",
    "print(f\"output_file: {output_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/20250325_182506/all_seqs_cutadapt_for_uchime.fa'"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Command executed successfully.\n",
      "STDOUT: \n",
      "STDERR: vsearch v2.29.2_linux_x86_64, 15.4GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/20250325_182506/all_seqs_cutadapt_for_uchime.fa 100%\n",
      "9948345 nt in 33066 seqs, min 1, max 927, avg 301\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Detecting chimeras 100%\n",
      "Found 10259 (31.0%) chimeras, 21703 (65.6%) non-chimeras,\n",
      "and 1104 (3.3%) borderline sequences in 33066 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "2093128 (1.2%) chimeras, 173234305 (98.6%) non-chimeras,\n",
      "and 327858 (0.2%) borderline sequences in 175655291 total sequences.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import subprocess\n",
    "out_chim_detect_dir = \"/\".join(output_file.split(\"/\")[:-1]) + \"/uchime_out/\"\n",
    "# Define the command and its arguments\n",
    "command = [\n",
    "    \"/home/mg/vsearch-2.29.2-linux-x86_64/bin/vsearch\",\n",
    "    \"--uchime_denovo\", output_file,\n",
    "    \"--chimeras\", out_chim_detect_dir+\"chimeras_uch_de.fa\",\n",
    "    \"--nonchimeras\", out_chim_detect_dir+\"nonchimeras_uch_de.fa\",\n",
    "    \"--fasta_score\"\n",
    "]\n",
    "\n",
    "# Run the command\n",
    "try:\n",
    "    result = subprocess.run(command, capture_output=True, text=True, check=True)\n",
    "    print(\"Command executed successfully.\")\n",
    "    print(\"STDOUT:\", result.stdout)\n",
    "    print(\"STDERR:\", result.stderr)\n",
    "except subprocess.CalledProcessError as e:\n",
    "    print(\"Error occurred while executing the command.\")\n",
    "    print(\"Return code:\", e.returncode)\n",
    "    print(\"STDOUT:\", e.stdout)\n",
    "    print(\"STDERR:\", e.stderr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chim_denovo 0.5780967960288809\n",
      "uchime_denovo 0.5349314265619506\n",
      "uchime_ref 0.4687110304954087\n",
      "uchime_ref all unite 0.534196099577604\n",
      "uchime_ref its2 unite 0.4527647126304921\n"
     ]
    }
   ],
   "source": [
    "out_chim_detect_dir = \"../my_data/all_simera_for_uchime_for_uchime_after_uchime/\"\n",
    "chimeras_uch_de = list(SeqIO.parse(out_chim_detect_dir+\"chimeras_chi_de.fa\", \"fasta\"))\n",
    "nonchimeras_uch_de = list(SeqIO.parse(out_chim_detect_dir+\"nonchimeras_chi_de.fa\", \"fasta\"))\n",
    "chimeras_uch_de_good = [i for i in chimeras_uch_de if \"chimera\" in i.description]\n",
    "nonchimeras_uch_de_good = [i for i in nonchimeras_uch_de if \"chimera\" not in i.description]\n",
    "#pprint(chimeras_uch_de_good[:2]);pprint(nonchimeras_uch_de_good[:2])\n",
    "acc = (len(nonchimeras_uch_de_good)+len(chimeras_uch_de_good))/(len(nonchimeras_uch_de)+len(chimeras_uch_de))\n",
    "print(\"chim_denovo\",acc)\n",
    "\n",
    "out_chim_detect_dir = \"../my_data/all_simera_for_uchime_for_uchime_after_uchime/\"\n",
    "chimeras_uch_de = list(SeqIO.parse(out_chim_detect_dir+\"chimeras_uch_de.fa\", \"fasta\"))\n",
    "nonchimeras_uch_de = list(SeqIO.parse(out_chim_detect_dir+\"nonchimeras_uch_de.fa\", \"fasta\"))\n",
    "chimeras_uch_de_good = [i for i in chimeras_uch_de if \"chimera\" in i.description]\n",
    "nonchimeras_uch_de_good = [i for i in nonchimeras_uch_de if \"chimera\" not in i.description]\n",
    "#pprint(chimeras_uch_de_good[:2]);pprint(nonchimeras_uch_de_good[:2])\n",
    "acc = (len(nonchimeras_uch_de_good)+len(chimeras_uch_de_good))/(len(nonchimeras_uch_de)+len(chimeras_uch_de))\n",
    "print(\"uchime_denovo\",acc)\n",
    "\n",
    "out_chim_detect_dir = \"../my_data/all_simera_for_uchime_for_uchime_after_uchime/\"\n",
    "chimeras_uch_de = list(SeqIO.parse(out_chim_detect_dir+\"chimeras_uch_ref_p_clust.fa\", \"fasta\"))\n",
    "nonchimeras_uch_de = list(SeqIO.parse(out_chim_detect_dir+\"nonchimeras_uch_ref_p_clust.fa\", \"fasta\"))\n",
    "chimeras_uch_de_good = [i for i in chimeras_uch_de if \"chimera\" in i.description]\n",
    "nonchimeras_uch_de_good = [i for i in nonchimeras_uch_de if \"chimera\" not in i.description]\n",
    "#pprint(chimeras_uch_de_good[:2]);pprint(nonchimeras_uch_de_good[:2])\n",
    "acc = (len(nonchimeras_uch_de_good)+len(chimeras_uch_de_good))/(len(nonchimeras_uch_de)+len(chimeras_uch_de))\n",
    "print(\"uchime_ref\",acc)\n",
    "\n",
    "out_chim_detect_dir = \"../my_data/all_simera_for_uchime_for_uchime_after_uchime/\"\n",
    "chimeras_uch_de = list(SeqIO.parse(out_chim_detect_dir+\"chimeras_uch_ref_unite_all.fa\", \"fasta\"))\n",
    "nonchimeras_uch_de = list(SeqIO.parse(out_chim_detect_dir+\"nonchimeras_uch_ref_unite_all.fa\", \"fasta\"))\n",
    "chimeras_uch_de_good = [i for i in chimeras_uch_de if \"chimera\" in i.description]\n",
    "nonchimeras_uch_de_good = [i for i in nonchimeras_uch_de if \"chimera\" not in i.description]\n",
    "#pprint(chimeras_uch_de_good[:2]);pprint(nonchimeras_uch_de_good[:2])\n",
    "acc = (len(nonchimeras_uch_de_good)+len(chimeras_uch_de_good))/(len(nonchimeras_uch_de)+len(chimeras_uch_de))\n",
    "print(\"uchime_ref all unite\",acc)\n",
    "\n",
    "out_chim_detect_dir = \"../my_data/all_simera_for_uchime_for_uchime_after_uchime/\"\n",
    "chimeras_uch_de = list(SeqIO.parse(out_chim_detect_dir+\"chimeras_uch_ref_unite_its2.fa\", \"fasta\"))\n",
    "nonchimeras_uch_de = list(SeqIO.parse(out_chim_detect_dir+\"nonchimeras_uch_ref_unite_its2.fa\", \"fasta\"))\n",
    "chimeras_uch_de_good = [i for i in chimeras_uch_de if \"chimera\" in i.description]\n",
    "nonchimeras_uch_de_good = [i for i in nonchimeras_uch_de if \"chimera\" not in i.description]\n",
    "#pprint(chimeras_uch_de_good[:2]);pprint(nonchimeras_uch_de_good[:2])\n",
    "acc = (len(nonchimeras_uch_de_good)+len(chimeras_uch_de_good))/(len(nonchimeras_uch_de)+len(chimeras_uch_de))\n",
    "print(\"uchime_ref its2 unite\",acc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Current PATH:\n",
      "1: /home/mg/miniconda3/envs/chim/bin\n",
      "2: /home/mg/.vscode-server/bin/ddc367ed5c8936efe395cffeec279b04ffd7db78/bin/remote-cli\n",
      "3: /home/mg/.cargo/bin\n",
      "4: /home/mg/miniconda3/envs/chim/bin\n",
      "5: /home/mg/miniconda3/condabin\n",
      "6: /usr/local/sbin\n",
      "7: /usr/local/bin\n",
      "8: /usr/sbin\n",
      "9: /usr/bin\n",
      "10: /sbin\n",
      "11: /bin\n",
      "12: /usr/games\n",
      "13: /usr/local/games\n",
      "14: /mnt/c/Program Files/Common Files/Oracle/Java/javapath\n",
      "15: /mnt/c/Program Files (x86)/Common Files/Oracle/Java/javapath\n",
      "16: /mnt/c/Windows/system32\n",
      "17: /mnt/c/Windows\n",
      "18: /mnt/c/Windows/System32/Wbem\n",
      "19: /mnt/c/Windows/System32/WindowsPowerShell/v1.0\n",
      "20: /mnt/c/Windows/System32/OpenSSH\n",
      "21: /mnt/c/Program Files/dotnet\n",
      "22: /mnt/c/Program Files/Mullvad VPN/resources\n",
      "23: /mnt/c/Program Files/MIT/Kerberos/bin\n",
      "24: /mnt/c/Users/hubmich/scoop/shims\n",
      "25: /mnt/c/Users/hubmich/AppData/Local/Programs/Python/Launcher\n",
      "26: /mnt/c/Users/hubmich/AppData/Local/Microsoft/WindowsApps\n",
      "27: /mnt/c/Users/hubmich/AppData/Local/Programs/Ollama\n",
      "28: /mnt/c/Users/hubmich/AppData/Local/Programs/Microsoft VS Code/bin\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import sys\n",
    "import shutil\n",
    "\n",
    "# Display the current PATH environment variable\n",
    "print(\"Current PATH:\")\n",
    "path_entries = os.environ.get('PATH', '').split(':')\n",
    "for i, path in enumerate(path_entries):\n",
    "    print(f\"{i+1}: {path}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### END"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## OLD versions:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "# old version\n",
    "import os\n",
    "import subprocess\n",
    "import re\n",
    "from collections import Counter\n",
    "import sys\n",
    "\n",
    "results = []  # List to store the content of hyperex_out.fa files\n",
    "warnings = []  # List to store the warnings\n",
    "errors = []  # List to store the errors\n",
    "hyperex_out_fa = f\"hyperex_temp_out.fa\"\n",
    "hyperex_log = f\"hyperex.log\"\n",
    "hyperex_gff = f\"hyperex_temp_out.gff\"\n",
    "\n",
    "# Clean up any existing temporary files\n",
    "if os.path.exists(hyperex_log): os.remove(hyperex_log)\n",
    "if os.path.exists(hyperex_gff): os.remove(hyperex_gff)\n",
    "if os.path.exists(hyperex_out_fa): os.remove(hyperex_out_fa)\n",
    "\n",
    "# Command template for hyperex\n",
    "command = [\"hyperex\", \"-f\", \"GTGARTCATCGARTCTTTG\", \"-r\", \"TCCTCCGCTTATTGATATGC\", \"-p\", \"hyperex_temp_out\"]\n",
    "\n",
    "lenka = len(fasta_seq)\n",
    "for i, record in enumerate(fasta_seq):\n",
    "    if i % 50 == 0: print(f\"{i}/{lenka}\", end=\"\\r\")\n",
    "    if i >0: command = [\"hyperex\", \"-f\", \"GTGARTCATCGARTCTTTG\", \"-r\", \"TCCTCCGCTTATTGATATGC\", \"-p\", \"hyperex_temp_out\",\"--force\"] \n",
    "    # Convert the record to FASTA format string\n",
    "    fasta_data = f\">{record.id}\\n{str(record.seq)}\\n\"\n",
    "    \n",
    "    # Run hyperex with the FASTA data passed via stdin\n",
    "    process = subprocess.Popen(command, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)\n",
    "    stdout, stderr = process.communicate(input=fasta_data)\n",
    "    \n",
    "    # Collect warnings and errors\n",
    "    warnings.extend(list(map(lambda x: x.replace(\"\\x1b[0m]\", \"\"), re.findall(\"WARN(.+)\", stdout))))\n",
    "    errors.extend(stderr)\n",
    "    \n",
    "    # Read results if the output file exists\n",
    "    if os.path.exists(hyperex_out_fa):\n",
    "        with open(hyperex_out_fa, \"r\") as fa_file:\n",
    "            results.append(fa_file.read())\n",
    "\n",
    "# Clean up any remaining temporary files\n",
    "if os.path.exists(hyperex_out_fa): os.remove(hyperex_out_fa)\n",
    "if os.path.exists(hyperex_log): os.remove(hyperex_log)\n",
    "if os.path.exists(hyperex_gff): os.remove(hyperex_gff)\n",
    "\n",
    "print()\n",
    "real_results = [i for i in results if len(i) > 0]\n",
    "print(\"number of real results:\", len(real_results))\n",
    "warning_counts = Counter(warnings)\n",
    "error_counts = Counter(errors)\n",
    "print(\"ERRORS:\", error_counts, \"WARNINGS:\", warning_counts, sep=\"\\n\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# old version 2\n",
    "import os\n",
    "import subprocess\n",
    "#from time import sleep\n",
    "#import sys\n",
    "import re\n",
    "from collections import Counter\n",
    "results = []  # List to store the content of hyperex_out.fa files\n",
    "warnings = []  # List to store the warnings\n",
    "errors = []  # List to store the errors\n",
    "hyperex_out_fa = f\"hyperex_temp_out.fa\"\n",
    "hyperex_log = f\"hyperex.log\"\n",
    "hyperex_gff = f\"hyperex_temp_out.gff\"\n",
    "temp_filename = f\"hyperex_temp_res.fasta\"\n",
    "if os.path.exists(hyperex_log): os.remove(hyperex_log)\n",
    "if os.path.exists(hyperex_gff): os.remove(hyperex_gff)\n",
    "if os.path.exists(temp_filename): os.remove(temp_filename)  # Delete the temporary input file\n",
    "\n",
    "with open(temp_filename, \"w\") as temp_fasta:\n",
    "    SeqIO.write(fasta_seq[0], temp_fasta, \"fasta\")\n",
    "command = [\"hyperex\",  \"-f\", \"GTGARTCATCGARTCTTTG\", \"-r\", \"TCCTCCGCTTATTGATATGC\",\"-p\",\"hyperex_temp_out\", temp_filename]\n",
    "result = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)\n",
    "warnings.extend(list(map(lambda x: x.replace(\"\\x1b[0m]\",\"\"), re.findall(\"WARN(.+)\",result.stdout))))\n",
    "errors.extend(result.stderr)\n",
    "if os.path.exists(hyperex_out_fa):\n",
    "    with open(hyperex_out_fa, \"r\") as fa_file:\n",
    "        results.append(fa_file.read())\n",
    "#print(result.stdout)\n",
    "#print(result.stderr)\n",
    "lenka = len(fasta_seq)\n",
    "command = [\"hyperex\",  \"-f\", \"GTGARTCATCGARTCTTTG\", \"-r\", \"TCCTCCGCTTATTGATATGC\",\"-p\",\"hyperex_temp_out\", temp_filename,\"--force\"]\n",
    "for i, record in enumerate(fasta_seq[1:]):\n",
    "    #temp_filename = f\"hyperex_temp_res.fasta\"\n",
    "    if i % 50 == 0: print(f\"{i}/{lenka}\",end=\"\\r\")\n",
    "    with open(temp_filename, \"w\") as temp_fasta:\n",
    "        SeqIO.write(record, temp_fasta, \"fasta\")\n",
    "    #if i == 0:command = [\"hyperex\",  \"-f\", \"GTGARTCATCGARTCTTTG\", \"-r\", \"TCCTCCGCTTATTGATATGC\",\"-p\",\"hyperex_temp_out\", temp_filename]\n",
    "    result = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)\n",
    "    warnings.extend(list(map(lambda x: x.replace(\"\\x1b[0m]\",\"\"), re.findall(\"WARN(.+)\",result.stdout))))\n",
    "    errors.extend(result.stderr)\n",
    "    #if os.path.exists(hyperex_out_fa):\n",
    "    with open(hyperex_out_fa, \"r\") as fa_file:\n",
    "        results.append(fa_file.read())\n",
    "\n",
    "if os.path.exists(hyperex_out_fa):os.remove(hyperex_out_fa)  \n",
    "if os.path.exists(hyperex_log): os.remove(hyperex_log)\n",
    "if os.path.exists(hyperex_gff): os.remove(hyperex_gff)\n",
    "if os.path.exists(temp_filename): os.remove(temp_filename)  # Delete the temporary input file\n",
    "\n",
    "print()\n",
    "real_results = [i for i in results if len(i) >0]\n",
    "print(\"number of real results:\",len(real_results))\n",
    "warning_counts = Counter(warnings)\n",
    "error_counts = Counter(errors)\n",
    "print(\"ERRORS:\",list(error_counts),\"WARNINGS:\",\"\\n\".join(list(warning_counts)),sep=\"\\n\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "1\n",
      "5\n",
      "2.23606797749979\n",
      "1.7099759466766968\n",
      "1.4953487812212205\n"
     ]
    }
   ],
   "source": [
    "from math import sqrt\n",
    "x = 5\n",
    "a = lambda x: 0\n",
    "b = lambda x: 1\n",
    "c = lambda x: x\n",
    "d = lambda x: sqrt(x)\n",
    "e = lambda x: x**(1/3)\n",
    "f = lambda x: sqrt(sqrt(x))\n",
    "g = lambda x: x**1.1\n",
    "h = lambda x: x**1.2\n",
    "i = lambda x: x**1.3\n",
    "j = lambda x: x**1.5\n",
    "k = lambda x: x**2\n",
    "\n",
    "\n",
    "print(a(x))\n",
    "print(b(x))\n",
    "print(c(x))\n",
    "print(d(x))\n",
    "print(e(x))\n",
    "print(f(x))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a =\"\"\"724243116\t\t602368116\n",
    "605418059\t\t603515119\n",
    "725531547\t\t607954746\n",
    "724 666 277\t\t602533799\n",
    "737363099\t\t736427302\n",
    "777049546\t\t776338316\n",
    "728604166\t\t\n",
    "732876652\t\t728839198\n",
    "776333462\t\t777333264\n",
    "776036182\t\t606061648\n",
    "724 666 277\t\t608 311 383\n",
    "725531547\t\t607954746\n",
    "731 624 508\t\t\"\"\".replace(\" \",\"\").split(\"\\n\")\n",
    "b= list(filter(lambda x: len(x)>0,\"\\t\".join(a).split(\"\\t\")))\n",
    "c=set(b)\n",
    "\",\".join(c)\n",
    "len(c)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "chim",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
