{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### In this notebook I will try to repair hyperex amplicon extraction tool\n",
    "\n",
    "hyperex keeps crushing when there is forward primer found upstream revese primer:\n",
    "\n",
    "thread 'main' panicked at /home/mg/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/hyperex-0.1.1/src/utils.rs:420:45: slice index starts at 256 but ends at 65 note: run with \\`RUST_BACKTRACE=1` environment variable to display a backtrace Aborted (core dumped)   \n",
    "\n",
    "I will  load the fasta seq with Biopython and then use hyperex on each read"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mg/ChimeraProject/scripts\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "print(os.getcwd())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[SeqRecord(seq=Seq('TCGAGTTAGGGTCTTCTCAGGCCCGACCTCCCAACCCCATGTTTACAAAACCTC...AAA'), id='Herpotrichiellaceae_sp|UDB03390183|SH0901949.09FU|reps|k__Fungi;p__Ascomycota;c__Eurotiomycetes;o__Chaetothyriales;f__Herpotrichiellaceae;g__unidentified;s__Herpotrichiellaceae_sp', name='Herpotrichiellaceae_sp|UDB03390183|SH0901949.09FU|reps|k__Fungi;p__Ascomycota;c__Eurotiomycetes;o__Chaetothyriales;f__Herpotrichiellaceae;g__unidentified;s__Herpotrichiellaceae_sp', description='Herpotrichiellaceae_sp|UDB03390183|SH0901949.09FU|reps|k__Fungi;p__Ascomycota;c__Eurotiomycetes;o__Chaetothyriales;f__Herpotrichiellaceae;g__unidentified;s__Herpotrichiellaceae_sp', dbxrefs=[]), SeqRecord(seq=Seq('ATCGAGTTAGGGTCTTCCCAGGCCCGACCTCCCAACCCTATGTTTACAAAACCT...AAA'), id='Cladophialophora_sp|UDB02869914|SH0901951.09FU|reps|k__Fungi;p__Ascomycota;c__Eurotiomycetes;o__Chaetothyriales;f__Herpotrichiellaceae;g__Cladophialophora;s__Cladophialophora_sp', name='Cladophialophora_sp|UDB02869914|SH0901951.09FU|reps|k__Fungi;p__Ascomycota;c__Eurotiomycetes;o__Chaetothyriales;f__Herpotrichiellaceae;g__Cladophialophora;s__Cladophialophora_sp', description='Cladophialophora_sp|UDB02869914|SH0901951.09FU|reps|k__Fungi;p__Ascomycota;c__Eurotiomycetes;o__Chaetothyriales;f__Herpotrichiellaceae;g__Cladophialophora;s__Cladophialophora_sp', dbxrefs=[]), SeqRecord(seq=Seq('AATGAGTTAGGGTCTTCTTGGCCCGACCTCCCAACCCTATGTTTACTAAACCTA...CAA'), id='Cladophialophora_sp|UDB02885146|SH0901952.09FU|reps|k__Fungi;p__Ascomycota;c__Eurotiomycetes;o__Chaetothyriales;f__Herpotrichiellaceae;g__Cladophialophora;s__Cladophialophora_sp', name='Cladophialophora_sp|UDB02885146|SH0901952.09FU|reps|k__Fungi;p__Ascomycota;c__Eurotiomycetes;o__Chaetothyriales;f__Herpotrichiellaceae;g__Cladophialophora;s__Cladophialophora_sp', description='Cladophialophora_sp|UDB02885146|SH0901952.09FU|reps|k__Fungi;p__Ascomycota;c__Eurotiomycetes;o__Chaetothyriales;f__Herpotrichiellaceae;g__Cladophialophora;s__Cladophialophora_sp', dbxrefs=[]), SeqRecord(seq=Seq('ATGAGTTAGGGTCCTCTGGGCCCGACCTCCCAACCCTATGTTTACTGAACCTAT...CAT'), id='Cladophialophora_sp|UDB02757411|SH0901953.09FU|reps|k__Fungi;p__Ascomycota;c__Eurotiomycetes;o__Chaetothyriales;f__Herpotrichiellaceae;g__Cladophialophora;s__Cladophialophora_sp', name='Cladophialophora_sp|UDB02757411|SH0901953.09FU|reps|k__Fungi;p__Ascomycota;c__Eurotiomycetes;o__Chaetothyriales;f__Herpotrichiellaceae;g__Cladophialophora;s__Cladophialophora_sp', description='Cladophialophora_sp|UDB02757411|SH0901953.09FU|reps|k__Fungi;p__Ascomycota;c__Eurotiomycetes;o__Chaetothyriales;f__Herpotrichiellaceae;g__Cladophialophora;s__Cladophialophora_sp', dbxrefs=[]), SeqRecord(seq=Seq('CCGAGTTAGGGTCTTCTCGGGCCCGACCTCCCAACCCCATGTTTACTCAATCTA...AAA'), id='Cladophialophora_sp|UDB02750185|SH0901955.09FU|reps|k__Fungi;p__Ascomycota;c__Eurotiomycetes;o__Chaetothyriales;f__Herpotrichiellaceae;g__Cladophialophora;s__Cladophialophora_sp', name='Cladophialophora_sp|UDB02750185|SH0901955.09FU|reps|k__Fungi;p__Ascomycota;c__Eurotiomycetes;o__Chaetothyriales;f__Herpotrichiellaceae;g__Cladophialophora;s__Cladophialophora_sp', description='Cladophialophora_sp|UDB02750185|SH0901955.09FU|reps|k__Fungi;p__Ascomycota;c__Eurotiomycetes;o__Chaetothyriales;f__Herpotrichiellaceae;g__Cladophialophora;s__Cladophialophora_sp', dbxrefs=[])]\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "fasta_seq = list(SeqIO.parse(\"../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/untrimmed_sequences/uchime_reference_dataset_untrimmed_16_10_2022.fasta\", \"fasta\"))\n",
    "#fasta_seq = fasta_seq[:300]\n",
    "print((fasta_seq[:5]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>homogenized primer names</th>\n",
       "      <th>primers sequences</th>\n",
       "      <th>target marker</th>\n",
       "      <th>Notes</th>\n",
       "      <th>better_names</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>18S-F/5.8S-R</td>\n",
       "      <td>GTAAAAGTCGTAACAAGGTTTC/GTTCAAAGAYTCGATGATTCAC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18S-F___5.8S-R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5.8S_Fun/ITS4_Fun</td>\n",
       "      <td>AACTTTYRRCAAYGGATCWCT/AGCCTCCGCTTATTGATATGCTTAART</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.8S_Fun___ITS4_Fun</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>58A2F/ITS4</td>\n",
       "      <td>ATCGATGAAGAACGCAG/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>58A2F___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>fITS7/ITS4</td>\n",
       "      <td>GTGARTCATCGAATCTTTG/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fITS7___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>fITS9/ITS4</td>\n",
       "      <td>GAACGCAGCRAAIIGYGA/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>fITS9=ITSf9=ITS9F</td>\n",
       "      <td>fITS9___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>FSeq/RSeq</td>\n",
       "      <td>ATGCCTGTTTGAGCGTC/CCTACCTGATTTGAGGTC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FSeq___RSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>gITS7/ITS4</td>\n",
       "      <td>GTGARTCATCGARTCTTTG/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gITS7___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>gITS7/ITS4ngs</td>\n",
       "      <td>GTGARTCATCGARTCTTTG/TCCTSCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gITS7___ITS4ngs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>gITS7ngs/ITS4ngsUni</td>\n",
       "      <td>GTGARTCATCRARTYTTTG/CCTSCSCTTANTDATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gITS7ngs___ITS4ngsUni</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ITS1/ITS2</td>\n",
       "      <td>TCCGTAGGTGAACCTGCGG/GCTGCGTTCTTCATCGATGC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1___ITS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ITS1/ITS4</td>\n",
       "      <td>TCCGTAGGTGAACCTGCGG/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITSboth</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ITS1/qITS2*</td>\n",
       "      <td>CTCCGTAGGTGAACCTGCGG/TTYGCTGYGTTCTTCATCG</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1___qITS2_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ITS1-30F/ITS1-217R</td>\n",
       "      <td>GTCCCTGCCCTTTGTACACA/TTTCGCTGCGTTCTTCATCG</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1-30F___ITS1-217R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ITS1F/58A2R</td>\n",
       "      <td>CTTGGTCATTTAGAGGAAGTAA/CTGCGTTCTTCATCGAT</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1F___58A2R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ITS1F/ITS2</td>\n",
       "      <td>CTTGGTCATTTAGAGGAAGTAA/GCTGCGTTCTTCATCGATGC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1F___ITS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ITS1F/ITS3</td>\n",
       "      <td>CTTGGTCATTTAGAGGAAGTAA/GCATCGATGAAGAACGCAGC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1F___ITS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ITS1F/ITS4</td>\n",
       "      <td>CTTGGTCATTTAGAGGAAGTAA/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITSboth</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1F___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ITS1F/LR6</td>\n",
       "      <td>CTTGGTCATTTAGAGGAAGTAA/CGCCAGTTCTGCTTACC</td>\n",
       "      <td>ITSboth</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1F___LR6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ITS1F_KYO1/ITS2_KYO1</td>\n",
       "      <td>CTHGGTCATTTAGAGGAASTAA/CTRYGTTCTTCATCGDT</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1F_KYO1___ITS2_KYO1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ITS1F_KYO1/ITS2_KYO2</td>\n",
       "      <td>CTHGGTCATTTAGAGGAASTAA/TTYRCTRCGTTCTTCATC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1F_KYO1___ITS2_KYO2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ITS1F_KYO2/ITS2_KYO2</td>\n",
       "      <td>TAGAGGAAGTAAAAGTCGTAA/TTYRCTRCGTTCTTCATC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1F_KYO2___ITS2_KYO2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ITS1FI2/5.8S</td>\n",
       "      <td>GAACCWGCGGARGGATCA/CGCTGCGTTCTTCATCG</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1FI2___5.8S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ITS1FI2/ITS2</td>\n",
       "      <td>GAACCWGCGGARGGATCA/GCTGCGTTCTTCATCGATGC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1FI2___ITS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ITS1Fngs/ITS2</td>\n",
       "      <td>GGTCATTTAGAGGAAGTAA/GCTGCGTTCTTCATCGATGC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1Fngs___ITS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ITS1ngs/ITS2</td>\n",
       "      <td>TCCGTAGGTGAACCTGC/GCTGCGTTCTTCATCGATGC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS1ngs___ITS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>ITS2F/ITS2R</td>\n",
       "      <td>GCATCGATGAAGAACGC/CCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>highly similar to ITS3/ITS4, and ITS2F/ITS2R s...</td>\n",
       "      <td>ITS2F___ITS2R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>ITS3/ITS4</td>\n",
       "      <td>GCATCGATGAAGAACGCAGC/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS3___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>ITS3_KYO2/ITS4</td>\n",
       "      <td>GATGAAGAACGYAGYRAA/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS3_KYO2___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>ITS3_KYO2/ITS4_KYO3</td>\n",
       "      <td>GATGAAGAACGYAGYRAA/CTBTTVCCKCTTCACTCG</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS3_KYO2___ITS4_KYO3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>ITS5/5.8S_fungi</td>\n",
       "      <td>GGAAGTAAAAGTCGTAACAAGG/CAAGAGATCCGTTGTTGAAAGTT</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS5___5.8S_fungi</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>ITS5/ITS2</td>\n",
       "      <td>GGAAGTAAAAGTCGTAACAAGG/GCTGCGTTCTTCATCGATGC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS5___ITS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>ITS5/ITS2_KYO2</td>\n",
       "      <td>GGAAGTAAAAGTCGTAACAAGG/TTYRCTRCGTTCTTCATC</td>\n",
       "      <td>ITS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS5___ITS2_KYO2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>ITS5/ITS4</td>\n",
       "      <td>GGAAGTAAAAGTCGTAACAAGG/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITSboth</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS5___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>ITS7o/ITS4</td>\n",
       "      <td>GTGAATCATCRAATYTTTG/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS7o___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>ITS86F/ITS4</td>\n",
       "      <td>GTGAATCATCGAATCTTTGAA/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS86F___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>ITS86F/ITS4-Tul</td>\n",
       "      <td>GTGAATCATCGAATCTTTGAA/CCGCCAGATTCACACATTGA</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS86F___ITS4-Tul</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>ITS9/ITS4</td>\n",
       "      <td>GCATTAGAAACTGCTCGTAATG/TCCTCCGCTTATTGATATGC</td>\n",
       "      <td>ITS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS9___ITS4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>ITS9MUNngs/ITS4ngsUni</td>\n",
       "      <td>TACACACCGCCCGTCG/CCTSCSCTTANTDATATGC</td>\n",
       "      <td>ITSboth</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS9MUNngs___ITS4ngsUni</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   homogenized primer names  \\\n",
       "0              18S-F/5.8S-R   \n",
       "1         5.8S_Fun/ITS4_Fun   \n",
       "2                58A2F/ITS4   \n",
       "3                fITS7/ITS4   \n",
       "4                fITS9/ITS4   \n",
       "5                 FSeq/RSeq   \n",
       "6                gITS7/ITS4   \n",
       "10            gITS7/ITS4ngs   \n",
       "11      gITS7ngs/ITS4ngsUni   \n",
       "12                ITS1/ITS2   \n",
       "13                ITS1/ITS4   \n",
       "14              ITS1/qITS2*   \n",
       "15       ITS1-30F/ITS1-217R   \n",
       "16              ITS1F/58A2R   \n",
       "17               ITS1F/ITS2   \n",
       "18               ITS1F/ITS3   \n",
       "19               ITS1F/ITS4   \n",
       "20                ITS1F/LR6   \n",
       "21     ITS1F_KYO1/ITS2_KYO1   \n",
       "22     ITS1F_KYO1/ITS2_KYO2   \n",
       "23     ITS1F_KYO2/ITS2_KYO2   \n",
       "25             ITS1FI2/5.8S   \n",
       "26             ITS1FI2/ITS2   \n",
       "27            ITS1Fngs/ITS2   \n",
       "28             ITS1ngs/ITS2   \n",
       "31              ITS2F/ITS2R   \n",
       "32                ITS3/ITS4   \n",
       "35           ITS3_KYO2/ITS4   \n",
       "36      ITS3_KYO2/ITS4_KYO3   \n",
       "47          ITS5/5.8S_fungi   \n",
       "48                ITS5/ITS2   \n",
       "49           ITS5/ITS2_KYO2   \n",
       "50                ITS5/ITS4   \n",
       "51               ITS7o/ITS4   \n",
       "52              ITS86F/ITS4   \n",
       "53          ITS86F/ITS4-Tul   \n",
       "54                ITS9/ITS4   \n",
       "55    ITS9MUNngs/ITS4ngsUni   \n",
       "\n",
       "                                    primers sequences target marker  \\\n",
       "0       GTAAAAGTCGTAACAAGGTTTC/GTTCAAAGAYTCGATGATTCAC          ITS1   \n",
       "1   AACTTTYRRCAAYGGATCWCT/AGCCTCCGCTTATTGATATGCTTAART          ITS2   \n",
       "2              ATCGATGAAGAACGCAG/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "3            GTGARTCATCGAATCTTTG/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "4             GAACGCAGCRAAIIGYGA/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "5                ATGCCTGTTTGAGCGTC/CCTACCTGATTTGAGGTC          ITS2   \n",
       "6            GTGARTCATCGARTCTTTG/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "10           GTGARTCATCGARTCTTTG/TCCTSCGCTTATTGATATGC          ITS2   \n",
       "11            GTGARTCATCRARTYTTTG/CCTSCSCTTANTDATATGC          ITS2   \n",
       "12           TCCGTAGGTGAACCTGCGG/GCTGCGTTCTTCATCGATGC          ITS1   \n",
       "13           TCCGTAGGTGAACCTGCGG/TCCTCCGCTTATTGATATGC       ITSboth   \n",
       "14           CTCCGTAGGTGAACCTGCGG/TTYGCTGYGTTCTTCATCG          ITS1   \n",
       "15          GTCCCTGCCCTTTGTACACA/TTTCGCTGCGTTCTTCATCG          ITS1   \n",
       "16           CTTGGTCATTTAGAGGAAGTAA/CTGCGTTCTTCATCGAT          ITS1   \n",
       "17        CTTGGTCATTTAGAGGAAGTAA/GCTGCGTTCTTCATCGATGC          ITS1   \n",
       "18        CTTGGTCATTTAGAGGAAGTAA/GCATCGATGAAGAACGCAGC          ITS1   \n",
       "19        CTTGGTCATTTAGAGGAAGTAA/TCCTCCGCTTATTGATATGC       ITSboth   \n",
       "20           CTTGGTCATTTAGAGGAAGTAA/CGCCAGTTCTGCTTACC       ITSboth   \n",
       "21           CTHGGTCATTTAGAGGAASTAA/CTRYGTTCTTCATCGDT          ITS1   \n",
       "22          CTHGGTCATTTAGAGGAASTAA/TTYRCTRCGTTCTTCATC          ITS1   \n",
       "23           TAGAGGAAGTAAAAGTCGTAA/TTYRCTRCGTTCTTCATC          ITS1   \n",
       "25               GAACCWGCGGARGGATCA/CGCTGCGTTCTTCATCG          ITS1   \n",
       "26            GAACCWGCGGARGGATCA/GCTGCGTTCTTCATCGATGC          ITS1   \n",
       "27           GGTCATTTAGAGGAAGTAA/GCTGCGTTCTTCATCGATGC          ITS1   \n",
       "28             TCCGTAGGTGAACCTGC/GCTGCGTTCTTCATCGATGC          ITS1   \n",
       "31              GCATCGATGAAGAACGC/CCTCCGCTTATTGATATGC          ITS2   \n",
       "32          GCATCGATGAAGAACGCAGC/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "35            GATGAAGAACGYAGYRAA/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "36              GATGAAGAACGYAGYRAA/CTBTTVCCKCTTCACTCG          ITS2   \n",
       "47     GGAAGTAAAAGTCGTAACAAGG/CAAGAGATCCGTTGTTGAAAGTT          ITS1   \n",
       "48        GGAAGTAAAAGTCGTAACAAGG/GCTGCGTTCTTCATCGATGC          ITS1   \n",
       "49          GGAAGTAAAAGTCGTAACAAGG/TTYRCTRCGTTCTTCATC          ITS1   \n",
       "50        GGAAGTAAAAGTCGTAACAAGG/TCCTCCGCTTATTGATATGC       ITSboth   \n",
       "51           GTGAATCATCRAATYTTTG/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "52         GTGAATCATCGAATCTTTGAA/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "53         GTGAATCATCGAATCTTTGAA/CCGCCAGATTCACACATTGA          ITS2   \n",
       "54        GCATTAGAAACTGCTCGTAATG/TCCTCCGCTTATTGATATGC          ITS2   \n",
       "55               TACACACCGCCCGTCG/CCTSCSCTTANTDATATGC       ITSboth   \n",
       "\n",
       "                                                Notes             better_names  \n",
       "0                                                 NaN           18S-F___5.8S-R  \n",
       "1                                                 NaN      5.8S_Fun___ITS4_Fun  \n",
       "2                                                 NaN             58A2F___ITS4  \n",
       "3                                                 NaN             fITS7___ITS4  \n",
       "4                                   fITS9=ITSf9=ITS9F             fITS9___ITS4  \n",
       "5                                                 NaN              FSeq___RSeq  \n",
       "6                                                 NaN             gITS7___ITS4  \n",
       "10                                                NaN          gITS7___ITS4ngs  \n",
       "11                                                NaN    gITS7ngs___ITS4ngsUni  \n",
       "12                                                NaN              ITS1___ITS2  \n",
       "13                                                NaN              ITS1___ITS4  \n",
       "14                                                NaN            ITS1___qITS2_  \n",
       "15                                                NaN     ITS1-30F___ITS1-217R  \n",
       "16                                                NaN            ITS1F___58A2R  \n",
       "17                                                NaN             ITS1F___ITS2  \n",
       "18                                                NaN             ITS1F___ITS3  \n",
       "19                                                NaN             ITS1F___ITS4  \n",
       "20                                                NaN              ITS1F___LR6  \n",
       "21                                                NaN   ITS1F_KYO1___ITS2_KYO1  \n",
       "22                                                NaN   ITS1F_KYO1___ITS2_KYO2  \n",
       "23                                                NaN   ITS1F_KYO2___ITS2_KYO2  \n",
       "25                                                NaN           ITS1FI2___5.8S  \n",
       "26                                                NaN           ITS1FI2___ITS2  \n",
       "27                                                NaN          ITS1Fngs___ITS2  \n",
       "28                                                NaN           ITS1ngs___ITS2  \n",
       "31  highly similar to ITS3/ITS4, and ITS2F/ITS2R s...            ITS2F___ITS2R  \n",
       "32                                                NaN              ITS3___ITS4  \n",
       "35                                                NaN         ITS3_KYO2___ITS4  \n",
       "36                                                NaN    ITS3_KYO2___ITS4_KYO3  \n",
       "47                                                NaN        ITS5___5.8S_fungi  \n",
       "48                                                NaN              ITS5___ITS2  \n",
       "49                                                NaN         ITS5___ITS2_KYO2  \n",
       "50                                                NaN              ITS5___ITS4  \n",
       "51                                                NaN             ITS7o___ITS4  \n",
       "52                                                NaN            ITS86F___ITS4  \n",
       "53                                                NaN        ITS86F___ITS4-Tul  \n",
       "54                                                NaN              ITS9___ITS4  \n",
       "55                                                NaN  ITS9MUNngs___ITS4ngsUni  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import re\n",
    "primers_df = pd.read_excel(\"../my_data/primers.xlsx\",header=1)\n",
    "primers_df[\"better_names\"] = primers_df[\"homogenized primer names\"].apply(lambda x: re.sub(r'[^a-zA-Z0-9\\-\\.\\_]', '_', str(x).replace(\"/\",\"___\")))\n",
    "primers_df_filtered = primers_df[~primers_df[\"primers sequences\"].str.contains(\"[\\\\(a-z\\\\+]\")]\n",
    "primers_df_filtered\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('GCATCGATGAAGAACGCAGC')"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "Seq(\"GCTGCGTTCTTCATCGATGC\").reverse_complement()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "from hyperex_repaired import hyperex_repaired"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TEST hyperex_repaired function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNINGS:\n",
      "25-times  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "1-times  Region  not found because primer TCCGTAGGTGAACCTGCGG was not found in the sequence\n",
      "1-times  Region  not found because primer GCTGCGTTCTTCATCGATGC was not found in the sequence\n",
      "number of real results: 22           ;  ERROR COUNT:  2;  SKIPPED READS:  3;  Execution time: 0.17003464698791504\n",
      "All  reads:\n",
      "seq_1, seq_2, seq_3_bad_miss_f, seq_4, seq_5, seq_6_bad_miss_r, seq_7, seq_8, seq_9_bad_rev_prim_first, seq_10, seq_11, seq_12, seq_13, seq_14, seq_15, seq_16, seq_17, seq_18, seq_19, seq_20, seq_21, seq_22, seq_23, seq_24, seq_25\n",
      "Good reads:\n",
      "seq_1, seq_2, seq_4, seq_5, seq_7, seq_8, seq_10, seq_11, seq_12, seq_13, seq_14, seq_15, seq_16, seq_17, seq_18, seq_19, seq_20, seq_21, seq_22, seq_23, seq_24, seq_25\n",
      "Bad reads:\n",
      "seq_6_bad_miss_r, seq_9_bad_rev_prim_first, seq_3_bad_miss_f\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_1', name='seq_1', description='seq_1 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_2', name='seq_2', description='seq_2 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_4', name='seq_4', description='seq_4 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_5', name='seq_5', description='seq_5 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_7', name='seq_7', description='seq_7 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_8', name='seq_8', description='seq_8 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_10', name='seq_10', description='seq_10 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_11', name='seq_11', description='seq_11 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_12', name='seq_12', description='seq_12 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_13', name='seq_13', description='seq_13 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_14', name='seq_14', description='seq_14 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_15', name='seq_15', description='seq_15 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_16', name='seq_16', description='seq_16 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_17', name='seq_17', description='seq_17 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_18', name='seq_18', description='seq_18 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_19', name='seq_19', description='seq_19 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_20', name='seq_20', description='seq_20 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_21', name='seq_21', description='seq_21 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_22', name='seq_22', description='seq_22 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_23', name='seq_23', description='seq_23 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_24', name='seq_24', description='seq_24 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTAGGTGAACCTGCGGAAAAAAAAAAAAAAAAAAAAAAAAAGCATCGATGA...AGC'), id='seq_25', name='seq_25', description='seq_25 forward=TCCGTAGGTGAACCTGCGG reverse=GCTGCGTTCTTCATCGATGC', dbxrefs=[])]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#test hyperex_repaired on test_hyperex_repaired.fa\n",
    "test_fasta_seq = list(SeqIO.parse(\"test_hyperex_repaired.fa\", \"fasta\"))\n",
    "test_res_seq = hyperex_repaired(test_fasta_seq,\"TCCGTAGGTGAACCTGCGG\",\"GCTGCGTTCTTCATCGATGC\")\n",
    "print(\"All  reads:\\n\",\", \".join([i.id for i in test_fasta_seq]),sep=\"\")\n",
    "print(\"Good reads:\\n\",\", \".join([i.id for i in test_res_seq]),sep=\"\")\n",
    "print(\"Bad reads:\\n\",\", \".join(set([i.id for i in test_fasta_seq])-set([i.id for i in test_res_seq])),sep=\"\")\n",
    "test_res_seq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract amplicons with hyperex repaired"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18S-F___5.8S-R\n",
      "20431/161335\r"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[14], line 11\u001b[0m\n\u001b[1;32m      9\u001b[0m rev \u001b[38;5;241m=\u001b[39m primer_seq\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m/\u001b[39m\u001b[38;5;124m\"\u001b[39m)[\u001b[38;5;241m1\u001b[39m]\u001b[38;5;241m.\u001b[39mstrip()\n\u001b[1;32m     10\u001b[0m \u001b[38;5;28mprint\u001b[39m(better_name)\n\u001b[0;32m---> 11\u001b[0m res_seq \u001b[38;5;241m=\u001b[39m hyperex_repaired(fasta_seq,forw,rev)\n\u001b[1;32m     12\u001b[0m output_file \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_hyperex/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mbetter_name\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m.fasta\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     13\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mopen\u001b[39m(output_file, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mw\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m handle:\n",
      "File \u001b[0;32m~/ChimeraProject/scripts/hyperex_repaired.py:51\u001b[0m, in \u001b[0;36mhyperex_repaired\u001b[0;34m(fasta_seq, forward_primer, reverse_primer)\u001b[0m\n\u001b[1;32m     49\u001b[0m fasta_data \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m.\u001b[39mjoin([\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m>\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mrecord\u001b[38;5;241m.\u001b[39mid\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mstr\u001b[39m(record\u001b[38;5;241m.\u001b[39mseq)\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;28;01mfor\u001b[39;00m record \u001b[38;5;129;01min\u001b[39;00m fasta_seq[start_index:]])\n\u001b[1;32m     50\u001b[0m process \u001b[38;5;241m=\u001b[39m subprocess\u001b[38;5;241m.\u001b[39mPopen(command, stdin\u001b[38;5;241m=\u001b[39msubprocess\u001b[38;5;241m.\u001b[39mPIPE, stdout\u001b[38;5;241m=\u001b[39msubprocess\u001b[38;5;241m.\u001b[39mPIPE, stderr\u001b[38;5;241m=\u001b[39msubprocess\u001b[38;5;241m.\u001b[39mPIPE, text\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m---> 51\u001b[0m stdout, stderr \u001b[38;5;241m=\u001b[39m process\u001b[38;5;241m.\u001b[39mcommunicate(\u001b[38;5;28minput\u001b[39m\u001b[38;5;241m=\u001b[39mfasta_data)\n\u001b[1;32m     52\u001b[0m warnings\u001b[38;5;241m.\u001b[39mextend(\u001b[38;5;28mlist\u001b[39m(\u001b[38;5;28mmap\u001b[39m(\u001b[38;5;28;01mlambda\u001b[39;00m x: x\u001b[38;5;241m.\u001b[39mreplace(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;130;01m\\x1b\u001b[39;00m\u001b[38;5;124m[0m]\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m), re\u001b[38;5;241m.\u001b[39mfindall(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mWARN(.+)\u001b[39m\u001b[38;5;124m\"\u001b[39m, stdout))))\n\u001b[1;32m     53\u001b[0m errors\u001b[38;5;241m.\u001b[39mappend(stderr)\n",
      "File \u001b[0;32m~/miniconda3/envs/chim/lib/python3.12/subprocess.py:1209\u001b[0m, in \u001b[0;36mPopen.communicate\u001b[0;34m(self, input, timeout)\u001b[0m\n\u001b[1;32m   1206\u001b[0m     endtime \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m   1208\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 1209\u001b[0m     stdout, stderr \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_communicate(\u001b[38;5;28minput\u001b[39m, endtime, timeout)\n\u001b[1;32m   1210\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyboardInterrupt\u001b[39;00m:\n\u001b[1;32m   1211\u001b[0m     \u001b[38;5;66;03m# https://bugs.python.org/issue25942\u001b[39;00m\n\u001b[1;32m   1212\u001b[0m     \u001b[38;5;66;03m# See the detailed comment in .wait().\u001b[39;00m\n\u001b[1;32m   1213\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "File \u001b[0;32m~/miniconda3/envs/chim/lib/python3.12/subprocess.py:2113\u001b[0m, in \u001b[0;36mPopen._communicate\u001b[0;34m(self, input, endtime, orig_timeout)\u001b[0m\n\u001b[1;32m   2106\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_check_timeout(endtime, orig_timeout,\n\u001b[1;32m   2107\u001b[0m                         stdout, stderr,\n\u001b[1;32m   2108\u001b[0m                         skip_check_and_raise\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[1;32m   2109\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mRuntimeError\u001b[39;00m(  \u001b[38;5;66;03m# Impossible :)\u001b[39;00m\n\u001b[1;32m   2110\u001b[0m         \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m_check_timeout(..., skip_check_and_raise=True) \u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[1;32m   2111\u001b[0m         \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mfailed to raise TimeoutExpired.\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m-> 2113\u001b[0m ready \u001b[38;5;241m=\u001b[39m selector\u001b[38;5;241m.\u001b[39mselect(timeout)\n\u001b[1;32m   2114\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_check_timeout(endtime, orig_timeout, stdout, stderr)\n\u001b[1;32m   2116\u001b[0m \u001b[38;5;66;03m# XXX Rewrite these to use non-blocking I/O on the file\u001b[39;00m\n\u001b[1;32m   2117\u001b[0m \u001b[38;5;66;03m# objects; they are no longer using C stdio!\u001b[39;00m\n",
      "File \u001b[0;32m~/miniconda3/envs/chim/lib/python3.12/selectors.py:415\u001b[0m, in \u001b[0;36m_PollLikeSelector.select\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    413\u001b[0m ready \u001b[38;5;241m=\u001b[39m []\n\u001b[1;32m    414\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 415\u001b[0m     fd_event_list \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_selector\u001b[38;5;241m.\u001b[39mpoll(timeout)\n\u001b[1;32m    416\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mInterruptedError\u001b[39;00m:\n\u001b[1;32m    417\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m ready\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "import sys\n",
    "\n",
    "lenky = {}\n",
    "seqs_list = []\n",
    "for n,row in primers_df_filtered.iterrows():\n",
    "    _,primer_seq,region,_,better_name, = list(row)\n",
    "    forw = primer_seq.split(\"/\")[0].strip()\n",
    "    rev = primer_seq.split(\"/\")[1].strip()\n",
    "    print(better_name)\n",
    "    res_seq = hyperex_repaired(fasta_seq,forw,rev)\n",
    "    output_file = f\"../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_hyperex/{better_name}.fasta\"\n",
    "    with open(output_file, \"w\") as handle:\n",
    "        SeqIO.write(res_seq, handle, \"fasta-2line\")\n",
    "    print(n)\n",
    "    lenky[better_name] = len(res_seq)\n",
    "    seqs_list.append(res_seq)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'seqs_list' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[27], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m ids \u001b[38;5;241m=\u001b[39m {}\n\u001b[0;32m----> 2\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m p,seq \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mzip\u001b[39m(primers_df_filtered[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mbetter_names\u001b[39m\u001b[38;5;124m\"\u001b[39m],seqs_list):\n\u001b[1;32m      3\u001b[0m     ids[p] \u001b[38;5;241m=\u001b[39m [i\u001b[38;5;241m.\u001b[39mid \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m seq]\n",
      "\u001b[0;31mNameError\u001b[0m: name 'seqs_list' is not defined"
     ]
    }
   ],
   "source": [
    "ids = {}\n",
    "for p,seq in zip(primers_df_filtered[\"better_names\"],seqs_list):\n",
    "    ids[p] = [i.id for i in seq]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'ids' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[12], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21;01mitertools\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mimport\u001b[39;00m chain\n\u001b[0;32m----> 2\u001b[0m ids_one_list \u001b[38;5;241m=\u001b[39m chain\u001b[38;5;241m.\u001b[39mfrom_iterable([v \u001b[38;5;28;01mfor\u001b[39;00m k, v \u001b[38;5;129;01min\u001b[39;00m \u001b[43mids\u001b[49m\u001b[38;5;241m.\u001b[39mitems()])\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21;01mcollections\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mimport\u001b[39;00m Counter\n\u001b[1;32m      4\u001b[0m counts \u001b[38;5;241m=\u001b[39m Counter(ids_one_list)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'ids' is not defined"
     ]
    }
   ],
   "source": [
    "from itertools import chain\n",
    "ids_one_list = chain.from_iterable([v for k, v in ids.items()])\n",
    "from collections import Counter\n",
    "counts = Counter(ids_one_list)\n",
    "len(counts),len(fasta_seq)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### use Simera on fastas from hyperex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "skipped = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "gITS7___ITS4\n",
      "simera!!! gITS7___ITS4 GTGARTCATCGARTCTTTG TCCTCCGCTTATTGATATGC\n",
      "\n",
      "========================================\n",
      "  Simera - simulates chimera formation\n",
      "  Use option -h for help\n",
      "========================================\n",
      "\n",
      "Generated 100000 potential chimeras...\n",
      "PCR Round 25...\n",
      "Done.\n",
      "Processed gITS7___ITS4:\n",
      "7 / 38\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n",
      "skipping....\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq\n",
    "from time import sleep\n",
    "from subprocess import run\n",
    "import time\n",
    "from subprocess import TimeoutExpired\n",
    "\n",
    "# Function to append \"_1\" to each read name in the FASTA file\n",
    "def append_suffix_to_fasta(input_file, output_file, suffix=\"_1\"):\n",
    "    records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "    modified_records = []\n",
    "    for record in records:\n",
    "        record.id = f\"{record.id}{suffix}\"\n",
    "        record.description = \"\"  # Clear description to avoid duplication\n",
    "        modified_records.append(record)\n",
    "    with open(output_file, \"w\") as handle:\n",
    "        SeqIO.write(modified_records, handle, \"fasta-2line\")\n",
    "\n",
    "# Iterate over each primer and its corresponding file\n",
    "for numero,row in enumerate(primers_df_filtered.iterrows()):\n",
    "    if numero < 0: continue\n",
    "    row = row[1]\n",
    "    primer_name = row[\"better_names\"]\n",
    "    primer_seq = row[\"primers sequences\"]\n",
    "    forward_primer = primer_seq.split(\"/\")[0].strip()\n",
    "    reverse_primer = primer_seq.split(\"/\")[1].strip()\n",
    "        # Check if the FASTA file is empty\n",
    "\n",
    "    if primer_name != \"gITS7___ITS4\": print(\"skipping....\") ; continue\n",
    "    print(primer_name)\n",
    "\n",
    "    input_file = f\"../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_hyperex/{primer_name}_sanitized.fasta\"\n",
    "    modified_file = f\"../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_hyperex/{primer_name}_abund.fasta\"\n",
    "    timestamp = time.strftime(\"%Y%m%d_%H%M%S\")\n",
    "    output_dir = f\"../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/{primer_name}/{timestamp}/\"\n",
    "    records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "    if len(records) < 10:  # If no records are found, continue to the next iteration\n",
    "        print(f\"Skipping {primer_name} as the file is empty.\")\n",
    "        continue\n",
    "    # Append \"_1\" to each read name\n",
    "    append_suffix_to_fasta(input_file,modified_file)\n",
    "    \n",
    "    # Create output directory if it doesn't exist\n",
    "    os.makedirs(output_dir, exist_ok=True)\n",
    "    print(\"simera!!!\",primer_name,forward_primer,reverse_primer)\n",
    "    # Run Simera\n",
    "    command = [\n",
    "        \"../Simera/Simera\",\n",
    "        \"-i\", modified_file,\n",
    "        \"-o\", output_dir,\n",
    "        \"-n\", \"25\",\n",
    "        \"-s\", \"10000\",\n",
    "        \"-l\", \"0.00005\",\n",
    "        \"-c\", \"100000\", # too large values for big fastas (6+Mb) will freeze Simera sometimes ??? dunno why, it fries my pc tho\n",
    "        \"-f\", forward_primer, # the limit is 20000 sequences in one fasta file                      \n",
    "        \"-r\", reverse_primer,\n",
    "        \"-x\"\n",
    "    ] # Simera is stuck when there is too little seqs in the fasta\n",
    "    #result = run(command, capture_output=True, text=True)\n",
    "    try:\n",
    "        result = run(command, capture_output=False, text=False, timeout=240)  # Timeout after 300 seconds\n",
    "    except TimeoutExpired:\n",
    "        print(f\"Simera timed out for {primer_name}. Skipping...\")\n",
    "        skipped.append((numero,primer_name))\n",
    "        continue  # Skip to the next iteration\n",
    " \n",
    "    # Print the result for debugging\n",
    "    print(f\"Processed {primer_name}:\")\n",
    "    #print(result.stdout)\n",
    "    #print(result.stderr)\n",
    "    print(numero+1,\"/\",len(primers_df_filtered))\n",
    "    sleep(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "Seq(\"GTTCAAAGAYTCGATGATTCAC\").reverse_complement()\n",
    "from random import choice\n",
    "ss = \"\".join([choice(\"ACGT\") for i in range(370)])\n",
    "ss\n",
    "#len(\"GTAAAAGTCGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTATTGAAATAAACTTAGGGGTTGTTGCTGGCCTTTCAAGGGCAATTTTGTGCACGCCTCTATCTATTTCCCTTTAACACCCCCATTGTGCATCTTGTGTAGGTTAACGTAAAGTTAATCTATGTTTTTATTTTTATACCTTTGTTTTTGAATGGGAATGTAATTTTTTAAACCCTCAGTAAGCTCTCATTTTGGAGCTGAAAAGGAAATAAATATGTGTACAACTTTCAACAACGGATCTCTTGGCTCTCGCATCGATGAAGAACGCAGCGAAATGCGATAAGTAATGTGAATTGCAGAATTCAGTGAATCATCGAATCTTTGAAC\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cut primers with cutadapt - cut the Simera output - I should do it at the end"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250322_134555/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250322_134555/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   6,141\n",
      "Reads with adapters:                     6,139 (100.0%)\n",
      "Reads written (passing filters):         6,141 (100.0%)\n",
      "\n",
      "Total basepairs processed:     2,097,043 bp\n",
      "Total written (filtered):      1,961,985 bp (93.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 6139 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t6139\t0.0\t2\t6138 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250322_134855/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250322_134855/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   5,984\n",
      "Reads with adapters:                     5,975 (99.8%)\n",
      "Reads written (passing filters):         5,984 (100.0%)\n",
      "\n",
      "Total basepairs processed:     2,044,849 bp\n",
      "Total written (filtered):      1,913,399 bp (93.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 5975 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t5975\t0.0\t2\t5974 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250322_134956/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250322_134956/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   6,090\n",
      "Reads with adapters:                     6,083 (99.9%)\n",
      "Reads written (passing filters):         6,090 (100.0%)\n",
      "\n",
      "Total basepairs processed:     2,075,486 bp\n",
      "Total written (filtered):      1,941,660 bp (93.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 6083 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t6083\t0.0\t2\t6081 1 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250322_135112/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250322_135112/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   5,720\n",
      "Reads with adapters:                     5,720 (100.0%)\n",
      "Reads written (passing filters):         5,720 (100.0%)\n",
      "\n",
      "Total basepairs processed:     1,950,388 bp\n",
      "Total written (filtered):      1,824,548 bp (93.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 5720 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t5720\t0.0\t2\t5720\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250322_135138/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250322_135138/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   5,993\n",
      "Reads with adapters:                     5,985 (99.9%)\n",
      "Reads written (passing filters):         5,993 (100.0%)\n",
      "\n",
      "Total basepairs processed:     2,047,333 bp\n",
      "Total written (filtered):      1,915,662 bp (93.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 5985 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t5984\t0.0\t2\t5983 1\n",
      "23\t1\t0.0\t2\t0 0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250322_135706/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250322_135706/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   5,991\n",
      "Reads with adapters:                     5,988 (99.9%)\n",
      "Reads written (passing filters):         5,991 (100.0%)\n",
      "\n",
      "Total basepairs processed:     2,046,022 bp\n",
      "Total written (filtered):      1,914,286 bp (93.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 5988 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t5988\t0.0\t2\t5988\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250322_135826/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250322_135826/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   5,957\n",
      "Reads with adapters:                     5,952 (99.9%)\n",
      "Reads written (passing filters):         5,957 (100.0%)\n",
      "\n",
      "Total basepairs processed:     2,031,233 bp\n",
      "Total written (filtered):      1,900,289 bp (93.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 5952 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t5952\t0.0\t2\t5952\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250325_144335/all_seqs.fa does not exist. Skipping...\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250325_144508/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250325_144508/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   8,200\n",
      "Reads with adapters:                     8,189 (99.9%)\n",
      "Reads written (passing filters):         8,200 (100.0%)\n",
      "\n",
      "Total basepairs processed:     2,797,923 bp\n",
      "Total written (filtered):      2,617,764 bp (93.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 8189 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "22\t8188\t0.0\t2\t8184 3 1\n",
      "23\t1\t0.0\t2\t0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250325_144741/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250325_144741/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  22,944\n",
      "Reads with adapters:                    22,926 (99.9%)\n",
      "Reads written (passing filters):        22,944 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,911,838 bp\n",
      "Total written (filtered):      7,407,467 bp (93.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 22926 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "21\t3\t0.0\t2\t0 3\n",
      "22\t22921\t0.0\t2\t22913 8\n",
      "23\t2\t0.0\t2\t0 2\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 18S-F___5.8S-R:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTAAAAGTCGTAACAAGGTTTC -a GTTCAAAGAYTCGATGATTCAC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250325_145337/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/20250325_145337/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  23,227\n",
      "Reads with adapters:                    23,214 (99.9%)\n",
      "Reads written (passing filters):        23,227 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,999,271 bp\n",
      "Total written (filtered):      7,488,560 bp (93.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTAAAAGTCGTAACAAGGTTTC; Type: regular 5'; Length: 22; Trimmed: 23214 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "21\t3\t0.0\t2\t0 3\n",
      "22\t23205\t0.0\t2\t23197 7 1\n",
      "23\t6\t0.0\t2\t0 6\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GTTCAAAGAYTCGATGATTCAC; Type: regular 3'; Length: 22; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/abund.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/all_seqs.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/breaks.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/chimeras.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/good.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/info.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/parents.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/samp_abund.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/samp_all_seqs.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/samp_breaks.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/samp_chimeras.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/samp_good.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/samp_parents.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/samp_summary.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/18S-F___5.8S-R/summary.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/5.8S_Fun___ITS4_Fun/abund.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/5.8S_Fun___ITS4_Fun/all_seqs.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/5.8S_Fun___ITS4_Fun/breaks.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/5.8S_Fun___ITS4_Fun/chimeras.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/5.8S_Fun___ITS4_Fun/good.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/5.8S_Fun___ITS4_Fun/info.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/5.8S_Fun___ITS4_Fun/parents.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/5.8S_Fun___ITS4_Fun/samp_abund.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/5.8S_Fun___ITS4_Fun/samp_all_seqs.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/5.8S_Fun___ITS4_Fun/samp_breaks.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/5.8S_Fun___ITS4_Fun/samp_chimeras.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/5.8S_Fun___ITS4_Fun/samp_good.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/5.8S_Fun___ITS4_Fun/samp_parents.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/5.8S_Fun___ITS4_Fun/samp_summary.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/5.8S_Fun___ITS4_Fun/summary.txt/all_seqs.fa does not exist. Skipping...\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250322_134556/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250322_134556/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  10,611\n",
      "Reads with adapters:                    10,599 (99.9%)\n",
      "Reads written (passing filters):        10,611 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,927,488 bp\n",
      "Total written (filtered):      3,747,306 bp (95.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 10599 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "16\t1\t0.0\t1\t0 1\n",
      "17\t10598\t0.0\t1\t10597 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250322_134856/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250322_134856/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  10,220\n",
      "Reads with adapters:                    10,216 (100.0%)\n",
      "Reads written (passing filters):        10,220 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,783,951 bp\n",
      "Total written (filtered):      3,610,279 bp (95.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 10216 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t10216\t0.0\t1\t10215 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250322_134957/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250322_134957/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  10,382\n",
      "Reads with adapters:                    10,374 (99.9%)\n",
      "Reads written (passing filters):        10,382 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,838,317 bp\n",
      "Total written (filtered):      3,661,959 bp (95.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 10374 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t10374\t0.0\t1\t10374\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250322_135113/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250322_135113/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   9,856\n",
      "Reads with adapters:                     9,856 (100.0%)\n",
      "Reads written (passing filters):         9,856 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,642,169 bp\n",
      "Total written (filtered):      3,474,617 bp (95.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 9856 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t9856\t0.0\t1\t9856\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250322_135139/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250322_135139/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  10,256\n",
      "Reads with adapters:                    10,250 (99.9%)\n",
      "Reads written (passing filters):        10,256 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,793,204 bp\n",
      "Total written (filtered):      3,618,954 bp (95.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 10250 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t10250\t0.0\t1\t10249 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250322_135707/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250322_135707/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  10,255\n",
      "Reads with adapters:                    10,251 (100.0%)\n",
      "Reads written (passing filters):        10,255 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,789,821 bp\n",
      "Total written (filtered):      3,615,554 bp (95.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 10251 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t10251\t0.0\t1\t10251\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250322_135827/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250322_135827/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  10,258\n",
      "Reads with adapters:                    10,250 (99.9%)\n",
      "Reads written (passing filters):        10,258 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,791,139 bp\n",
      "Total written (filtered):      3,616,889 bp (95.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 10250 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t10250\t0.0\t1\t10250\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed 58A2F___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATCGATGAAGAACGCAG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250325_144514/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250325_144514/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  15,386\n",
      "Reads with adapters:                    15,358 (99.8%)\n",
      "Reads written (passing filters):        15,386 (100.0%)\n",
      "\n",
      "Total basepairs processed:     5,755,572 bp\n",
      "Total written (filtered):      5,494,486 bp (95.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCGATGAAGAACGCAG; Type: regular 5'; Length: 17; Trimmed: 15358 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t15358\t0.0\t1\t15357 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250325_144846/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/20250325_145457/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/abund.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/all_seqs.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/breaks.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/chimeras.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/good.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/info.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/parents.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/samp_abund.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/samp_all_seqs.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/samp_breaks.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/samp_chimeras.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/samp_good.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/samp_parents.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/samp_summary.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/58A2F___ITS4/summary.txt/all_seqs.fa does not exist. Skipping...\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250322_134559/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250322_134559/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   9,922\n",
      "Reads with adapters:                     9,906 (99.8%)\n",
      "Reads written (passing filters):         9,922 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,177,191 bp\n",
      "Total written (filtered):      2,988,978 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 9906 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t1\t0.0\t1\t0 1\n",
      "19\t9905\t0.0\t1\t9903 2\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250322_134858/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250322_134858/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   9,596\n",
      "Reads with adapters:                     9,582 (99.9%)\n",
      "Reads written (passing filters):         9,596 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,073,660 bp\n",
      "Total written (filtered):      2,891,602 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 9582 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t9582\t0.0\t1\t9581 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250322_134959/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250322_134959/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   9,755\n",
      "Reads with adapters:                     9,737 (99.8%)\n",
      "Reads written (passing filters):         9,755 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,121,479 bp\n",
      "Total written (filtered):      2,936,476 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 9737 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t9737\t0.0\t1\t9737\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250322_135114/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250322_135114/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   9,186\n",
      "Reads with adapters:                     9,186 (100.0%)\n",
      "Reads written (passing filters):         9,186 (100.0%)\n",
      "\n",
      "Total basepairs processed:     2,938,435 bp\n",
      "Total written (filtered):      2,763,901 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 9186 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t9186\t0.0\t1\t9186\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250322_135141/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250322_135141/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   9,583\n",
      "Reads with adapters:                     9,569 (99.9%)\n",
      "Reads written (passing filters):         9,583 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,067,781 bp\n",
      "Total written (filtered):      2,885,970 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 9569 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t9569\t0.0\t1\t9568 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250322_135709/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250322_135709/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   9,602\n",
      "Reads with adapters:                     9,582 (99.8%)\n",
      "Reads written (passing filters):         9,602 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,072,290 bp\n",
      "Total written (filtered):      2,890,233 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 9582 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t1\t0.0\t1\t0 1\n",
      "19\t9581\t0.0\t1\t9579 2\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250322_135829/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250322_135829/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   9,588\n",
      "Reads with adapters:                     9,574 (99.9%)\n",
      "Reads written (passing filters):         9,588 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,068,143 bp\n",
      "Total written (filtered):      2,886,237 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 9574 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t9574\t0.0\t1\t9571 3\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250325_144524/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250325_144524/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  13,692\n",
      "Reads with adapters:                    13,632 (99.6%)\n",
      "Reads written (passing filters):        13,692 (100.0%)\n",
      "\n",
      "Total basepairs processed:     4,404,052 bp\n",
      "Total written (filtered):      4,145,045 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 13632 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t1\t0.0\t1\t0 1\n",
      "19\t13631\t0.0\t1\t13627 4\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed fITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGAATCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250325_145857/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/20250325_145857/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  31,515\n",
      "Reads with adapters:                    31,422 (99.7%)\n",
      "Reads written (passing filters):        31,515 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,091,459 bp\n",
      "Total written (filtered):      9,494,441 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGAATCTTTG; Type: regular 5'; Length: 19; Trimmed: 31422 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t31422\t0.0\t1\t31412 10\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/abund.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/all_seqs.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/breaks.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/chimeras.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/good.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/info.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/parents.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/samp_abund.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/samp_all_seqs.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/samp_breaks.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/samp_chimeras.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/samp_good.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/samp_parents.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/samp_summary.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/fITS7___ITS4/summary.txt/all_seqs.fa does not exist. Skipping...\n",
      "Processed FSeq___RSeq:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATGCCTGTTTGAGCGTC -a CCTACCTGATTTGAGGTC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250322_134601/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250322_134601/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   1,037\n",
      "Reads with adapters:                     1,028 (99.1%)\n",
      "Reads written (passing filters):         1,037 (100.0%)\n",
      "\n",
      "Total basepairs processed:       234,624 bp\n",
      "Total written (filtered):        217,148 bp (92.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATGCCTGTTTGAGCGTC; Type: regular 5'; Length: 17; Trimmed: 1028 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t1028\t0.0\t1\t1028\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACCTGATTTGAGGTC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed FSeq___RSeq:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATGCCTGTTTGAGCGTC -a CCTACCTGATTTGAGGTC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250322_134900/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250322_134900/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     894\n",
      "Reads with adapters:                       886 (99.1%)\n",
      "Reads written (passing filters):           894 (100.0%)\n",
      "\n",
      "Total basepairs processed:       200,597 bp\n",
      "Total written (filtered):        185,535 bp (92.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATGCCTGTTTGAGCGTC; Type: regular 5'; Length: 17; Trimmed: 886 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t886\t0.0\t1\t886\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACCTGATTTGAGGTC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed FSeq___RSeq:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATGCCTGTTTGAGCGTC -a CCTACCTGATTTGAGGTC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250322_135000/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250322_135000/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   1,001\n",
      "Reads with adapters:                       991 (99.0%)\n",
      "Reads written (passing filters):         1,001 (100.0%)\n",
      "\n",
      "Total basepairs processed:       228,414 bp\n",
      "Total written (filtered):        211,568 bp (92.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATGCCTGTTTGAGCGTC; Type: regular 5'; Length: 17; Trimmed: 991 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "16\t1\t0.0\t1\t0 1\n",
      "17\t990\t0.0\t1\t987 3\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACCTGATTTGAGGTC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed FSeq___RSeq:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATGCCTGTTTGAGCGTC -a CCTACCTGATTTGAGGTC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250322_135115/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250322_135115/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     591\n",
      "Reads with adapters:                       591 (100.0%)\n",
      "Reads written (passing filters):           591 (100.0%)\n",
      "\n",
      "Total basepairs processed:       132,018 bp\n",
      "Total written (filtered):        121,971 bp (92.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATGCCTGTTTGAGCGTC; Type: regular 5'; Length: 17; Trimmed: 591 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t591\t0.0\t1\t591\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACCTGATTTGAGGTC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed FSeq___RSeq:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATGCCTGTTTGAGCGTC -a CCTACCTGATTTGAGGTC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250322_135142/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250322_135142/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     878\n",
      "Reads with adapters:                       872 (99.3%)\n",
      "Reads written (passing filters):           878 (100.0%)\n",
      "\n",
      "Total basepairs processed:       197,463 bp\n",
      "Total written (filtered):        182,639 bp (92.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATGCCTGTTTGAGCGTC; Type: regular 5'; Length: 17; Trimmed: 872 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t872\t0.0\t1\t871 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACCTGATTTGAGGTC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed FSeq___RSeq:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATGCCTGTTTGAGCGTC -a CCTACCTGATTTGAGGTC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250322_135710/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250322_135710/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     882\n",
      "Reads with adapters:                       872 (98.9%)\n",
      "Reads written (passing filters):           882 (100.0%)\n",
      "\n",
      "Total basepairs processed:       197,935 bp\n",
      "Total written (filtered):        183,111 bp (92.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATGCCTGTTTGAGCGTC; Type: regular 5'; Length: 17; Trimmed: 872 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t872\t0.0\t1\t871 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACCTGATTTGAGGTC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed FSeq___RSeq:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATGCCTGTTTGAGCGTC -a CCTACCTGATTTGAGGTC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250322_135830/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250322_135830/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     868\n",
      "Reads with adapters:                       860 (99.1%)\n",
      "Reads written (passing filters):           868 (100.0%)\n",
      "\n",
      "Total basepairs processed:       197,158 bp\n",
      "Total written (filtered):        182,538 bp (92.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATGCCTGTTTGAGCGTC; Type: regular 5'; Length: 17; Trimmed: 860 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t860\t0.0\t1\t860\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACCTGATTTGAGGTC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed FSeq___RSeq:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATGCCTGTTTGAGCGTC -a CCTACCTGATTTGAGGTC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250325_144533/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250325_144533/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   2,468\n",
      "Reads with adapters:                     2,455 (99.5%)\n",
      "Reads written (passing filters):         2,468 (100.0%)\n",
      "\n",
      "Total basepairs processed:       566,193 bp\n",
      "Total written (filtered):        524,457 bp (92.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATGCCTGTTTGAGCGTC; Type: regular 5'; Length: 17; Trimmed: 2455 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t2454\t0.0\t1\t2453 1\n",
      "18\t1\t0.0\t1\t0 1\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACCTGATTTGAGGTC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed FSeq___RSeq:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g ATGCCTGTTTGAGCGTC -a CCTACCTGATTTGAGGTC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250325_150047/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/20250325_150047/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   7,634\n",
      "Reads with adapters:                     7,617 (99.8%)\n",
      "Reads written (passing filters):         7,634 (100.0%)\n",
      "\n",
      "Total basepairs processed:     1,750,166 bp\n",
      "Total written (filtered):      1,620,677 bp (92.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATGCCTGTTTGAGCGTC; Type: regular 5'; Length: 17; Trimmed: 7617 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-17 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t7617\t0.0\t1\t7614 3\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACCTGATTTGAGGTC; Type: regular 3'; Length: 18; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/abund.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/all_seqs.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/breaks.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/chimeras.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/good.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/info.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/parents.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/samp_abund.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/samp_all_seqs.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/samp_breaks.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/samp_chimeras.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/samp_good.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/samp_parents.fa/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/samp_summary.txt/all_seqs.fa does not exist. Skipping...\n",
      "Input file ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/FSeq___RSeq/summary.txt/all_seqs.fa does not exist. Skipping...\n",
      "Processed gITS7___ITS4:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTCCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/20250325_182506/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4/20250325_182506/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  33,778\n",
      "Reads with adapters:                    33,675 (99.7%)\n",
      "Reads written (passing filters):        33,778 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,807,480 bp\n",
      "Total written (filtered):     10,167,672 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 33675 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1\t527.8\t0\t1\n",
      "18\t1\t0.0\t1\t0 1\n",
      "19\t33673\t0.0\t1\t33660 13\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTCCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4ngs:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTSCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/20250322_134604/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/20250322_134604/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  10,619\n",
      "Reads with adapters:                    10,592 (99.7%)\n",
      "Reads written (passing filters):        10,619 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,388,660 bp\n",
      "Total written (filtered):      3,187,412 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 10592 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t10592\t0.0\t1\t10587 5\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTSCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n",
      "Processed gITS7___ITS4ngs:\n",
      "This is cutadapt 5.0 with Python 3.12.2\n",
      "Command line parameters: -g GTGARTCATCGARTCTTTG -a TCCTSCGCTTATTGATATGC -o ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/20250322_134903/all_seqs_cutadapt.fa ../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/gITS7___ITS4ngs/20250322_134903/all_seqs_sanitized.fa\n",
      "Processing single-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  10,204\n",
      "Reads with adapters:                    10,191 (99.9%)\n",
      "Reads written (passing filters):        10,204 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,261,756 bp\n",
      "Total written (filtered):      3,068,127 bp (94.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGARTCATCGARTCTTTG; Type: regular 5'; Length: 19; Trimmed: 10191 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t10191\t0.0\t1\t10188 3\n",
      "\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: TCCTSCGCTTATTGATATGC; Type: regular 3'; Length: 20; Trimmed: 0 times\n",
      "\n",
      "\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[14], line 34\u001b[0m\n\u001b[1;32m     32\u001b[0m \u001b[38;5;66;03m# Sanitize the input FASTA file\u001b[39;00m\n\u001b[1;32m     33\u001b[0m records \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(SeqIO\u001b[38;5;241m.\u001b[39mparse(input_file, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mfasta\u001b[39m\u001b[38;5;124m\"\u001b[39m))\n\u001b[0;32m---> 34\u001b[0m sanitized_records \u001b[38;5;241m=\u001b[39m sanitize_fasta_records(records)\n\u001b[1;32m     35\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mopen\u001b[39m(sanitized_file, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mw\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m handle:\n\u001b[1;32m     36\u001b[0m     SeqIO\u001b[38;5;241m.\u001b[39mwrite(sanitized_records, handle, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mfasta\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "Cell \u001b[0;32mIn[14], line 10\u001b[0m, in \u001b[0;36msanitize_fasta_records\u001b[0;34m(records)\u001b[0m\n\u001b[1;32m      8\u001b[0m sanitized_records \u001b[38;5;241m=\u001b[39m []\n\u001b[1;32m      9\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m record \u001b[38;5;129;01min\u001b[39;00m records:\n\u001b[0;32m---> 10\u001b[0m     sanitized_id \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;241m.\u001b[39mjoin(c \u001b[38;5;28;01mif\u001b[39;00m c\u001b[38;5;241m.\u001b[39misascii() \u001b[38;5;129;01mor\u001b[39;00m c \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m-_.\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m_\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;28;01mfor\u001b[39;00m c \u001b[38;5;129;01min\u001b[39;00m record\u001b[38;5;241m.\u001b[39mid)\n\u001b[1;32m     11\u001b[0m     sanitized_description \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;241m.\u001b[39mjoin(c \u001b[38;5;28;01mif\u001b[39;00m c\u001b[38;5;241m.\u001b[39misascii() \u001b[38;5;129;01mor\u001b[39;00m c \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m-_.\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m_\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;28;01mfor\u001b[39;00m c \u001b[38;5;129;01min\u001b[39;00m record\u001b[38;5;241m.\u001b[39mdescription)\n\u001b[1;32m     12\u001b[0m     sanitized_records\u001b[38;5;241m.\u001b[39mappend(SeqRecord(Seq(\u001b[38;5;28mstr\u001b[39m(record\u001b[38;5;241m.\u001b[39mseq)), \u001b[38;5;28mid\u001b[39m\u001b[38;5;241m=\u001b[39msanitized_id, description\u001b[38;5;241m=\u001b[39msanitized_description))\n",
      "Cell \u001b[0;32mIn[14], line 10\u001b[0m, in \u001b[0;36m<genexpr>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m      8\u001b[0m sanitized_records \u001b[38;5;241m=\u001b[39m []\n\u001b[1;32m      9\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m record \u001b[38;5;129;01min\u001b[39;00m records:\n\u001b[0;32m---> 10\u001b[0m     sanitized_id \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;241m.\u001b[39mjoin(c \u001b[38;5;28;01mif\u001b[39;00m c\u001b[38;5;241m.\u001b[39misascii() \u001b[38;5;129;01mor\u001b[39;00m c \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m-_.\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m_\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;28;01mfor\u001b[39;00m c \u001b[38;5;129;01min\u001b[39;00m record\u001b[38;5;241m.\u001b[39mid)\n\u001b[1;32m     11\u001b[0m     sanitized_description \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;241m.\u001b[39mjoin(c \u001b[38;5;28;01mif\u001b[39;00m c\u001b[38;5;241m.\u001b[39misascii() \u001b[38;5;129;01mor\u001b[39;00m c \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m-_.\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m_\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;28;01mfor\u001b[39;00m c \u001b[38;5;129;01min\u001b[39;00m record\u001b[38;5;241m.\u001b[39mdescription)\n\u001b[1;32m     12\u001b[0m     sanitized_records\u001b[38;5;241m.\u001b[39mappend(SeqRecord(Seq(\u001b[38;5;28mstr\u001b[39m(record\u001b[38;5;241m.\u001b[39mseq)), \u001b[38;5;28mid\u001b[39m\u001b[38;5;241m=\u001b[39msanitized_id, description\u001b[38;5;241m=\u001b[39msanitized_description))\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq\n",
    "import os\n",
    "from subprocess import run\n",
    "\n",
    "# Function to sanitize sequence IDs and descriptions\n",
    "def sanitize_fasta_records(records):\n",
    "    sanitized_records = []\n",
    "    for record in records:\n",
    "        sanitized_id = ''.join(c if c.isascii() or c in \"-_.\" else \"_\" for c in record.id)\n",
    "        sanitized_description = ''.join(c if c.isascii() or c in \"-_.\" else \"_\" for c in record.description)\n",
    "        sanitized_records.append(SeqRecord(Seq(str(record.seq)), id=sanitized_id, description=sanitized_description))\n",
    "    return sanitized_records\n",
    "\n",
    "# Iterate over each primer and its corresponding file\n",
    "for _, row in primers_df_filtered.iterrows():\n",
    "    primer_name = row[\"better_names\"]\n",
    "    primer_seq = row[\"primers sequences\"]\n",
    "    #if primer_name != \"gITS7___ITS4\": continue #########!!!!!!!!!!!\n",
    "\n",
    "    simera_dir = f\"../datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/amplicons_from_primers_simera/{primer_name}/\"\n",
    "    for time_dir in os.listdir(simera_dir):\n",
    "        time_dir = simera_dir + time_dir\n",
    "        \n",
    "        input_file = time_dir + \"/all_seqs.fa\"\n",
    "        sanitized_file = time_dir + \"/all_seqs_sanitized.fa\"\n",
    "        output_file = time_dir + \"/all_seqs_cutadapt.fa\"\n",
    "        \n",
    "        if not os.path.exists(input_file):\n",
    "            print(f\"Input file {input_file} does not exist. Skipping...\")\n",
    "            continue\n",
    "        # Sanitize the input FASTA file\n",
    "        records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "        sanitized_records = sanitize_fasta_records(records)\n",
    "        with open(sanitized_file, \"w\") as handle:\n",
    "            SeqIO.write(sanitized_records, handle, \"fasta\")\n",
    "        \n",
    "        # Run cutadapt to cut primers\n",
    "        command = [\n",
    "            \"cutadapt\",\n",
    "            \"-g\", primer_seq.split(\"/\")[0].strip(),  # Forward primer\n",
    "            \"-a\", primer_seq.split(\"/\")[1].strip(),  # Reverse primer\n",
    "            \"-o\", output_file,\n",
    "            sanitized_file\n",
    "        ]\n",
    "        result = run(command, capture_output=True, text=True)\n",
    "        \n",
    "        # Print the result for debugging\n",
    "        print(f\"Processed {primer_name}:\")\n",
    "        print(result.stdout)\n",
    "        print(result.stderr)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### END"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## OLD versions:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "# old version\n",
    "import os\n",
    "import subprocess\n",
    "import re\n",
    "from collections import Counter\n",
    "import sys\n",
    "\n",
    "results = []  # List to store the content of hyperex_out.fa files\n",
    "warnings = []  # List to store the warnings\n",
    "errors = []  # List to store the errors\n",
    "hyperex_out_fa = f\"hyperex_temp_out.fa\"\n",
    "hyperex_log = f\"hyperex.log\"\n",
    "hyperex_gff = f\"hyperex_temp_out.gff\"\n",
    "\n",
    "# Clean up any existing temporary files\n",
    "if os.path.exists(hyperex_log): os.remove(hyperex_log)\n",
    "if os.path.exists(hyperex_gff): os.remove(hyperex_gff)\n",
    "if os.path.exists(hyperex_out_fa): os.remove(hyperex_out_fa)\n",
    "\n",
    "# Command template for hyperex\n",
    "command = [\"hyperex\", \"-f\", \"GTGARTCATCGARTCTTTG\", \"-r\", \"TCCTCCGCTTATTGATATGC\", \"-p\", \"hyperex_temp_out\"]\n",
    "\n",
    "lenka = len(fasta_seq)\n",
    "for i, record in enumerate(fasta_seq):\n",
    "    if i % 50 == 0: print(f\"{i}/{lenka}\", end=\"\\r\")\n",
    "    if i >0: command = [\"hyperex\", \"-f\", \"GTGARTCATCGARTCTTTG\", \"-r\", \"TCCTCCGCTTATTGATATGC\", \"-p\", \"hyperex_temp_out\",\"--force\"] \n",
    "    # Convert the record to FASTA format string\n",
    "    fasta_data = f\">{record.id}\\n{str(record.seq)}\\n\"\n",
    "    \n",
    "    # Run hyperex with the FASTA data passed via stdin\n",
    "    process = subprocess.Popen(command, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)\n",
    "    stdout, stderr = process.communicate(input=fasta_data)\n",
    "    \n",
    "    # Collect warnings and errors\n",
    "    warnings.extend(list(map(lambda x: x.replace(\"\\x1b[0m]\", \"\"), re.findall(\"WARN(.+)\", stdout))))\n",
    "    errors.extend(stderr)\n",
    "    \n",
    "    # Read results if the output file exists\n",
    "    if os.path.exists(hyperex_out_fa):\n",
    "        with open(hyperex_out_fa, \"r\") as fa_file:\n",
    "            results.append(fa_file.read())\n",
    "\n",
    "# Clean up any remaining temporary files\n",
    "if os.path.exists(hyperex_out_fa): os.remove(hyperex_out_fa)\n",
    "if os.path.exists(hyperex_log): os.remove(hyperex_log)\n",
    "if os.path.exists(hyperex_gff): os.remove(hyperex_gff)\n",
    "\n",
    "print()\n",
    "real_results = [i for i in results if len(i) > 0]\n",
    "print(\"number of real results:\", len(real_results))\n",
    "warning_counts = Counter(warnings)\n",
    "error_counts = Counter(errors)\n",
    "print(\"ERRORS:\", error_counts, \"WARNINGS:\", warning_counts, sep=\"\\n\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# old version 2\n",
    "import os\n",
    "import subprocess\n",
    "#from time import sleep\n",
    "#import sys\n",
    "import re\n",
    "from collections import Counter\n",
    "results = []  # List to store the content of hyperex_out.fa files\n",
    "warnings = []  # List to store the warnings\n",
    "errors = []  # List to store the errors\n",
    "hyperex_out_fa = f\"hyperex_temp_out.fa\"\n",
    "hyperex_log = f\"hyperex.log\"\n",
    "hyperex_gff = f\"hyperex_temp_out.gff\"\n",
    "temp_filename = f\"hyperex_temp_res.fasta\"\n",
    "if os.path.exists(hyperex_log): os.remove(hyperex_log)\n",
    "if os.path.exists(hyperex_gff): os.remove(hyperex_gff)\n",
    "if os.path.exists(temp_filename): os.remove(temp_filename)  # Delete the temporary input file\n",
    "\n",
    "with open(temp_filename, \"w\") as temp_fasta:\n",
    "    SeqIO.write(fasta_seq[0], temp_fasta, \"fasta\")\n",
    "command = [\"hyperex\",  \"-f\", \"GTGARTCATCGARTCTTTG\", \"-r\", \"TCCTCCGCTTATTGATATGC\",\"-p\",\"hyperex_temp_out\", temp_filename]\n",
    "result = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)\n",
    "warnings.extend(list(map(lambda x: x.replace(\"\\x1b[0m]\",\"\"), re.findall(\"WARN(.+)\",result.stdout))))\n",
    "errors.extend(result.stderr)\n",
    "if os.path.exists(hyperex_out_fa):\n",
    "    with open(hyperex_out_fa, \"r\") as fa_file:\n",
    "        results.append(fa_file.read())\n",
    "#print(result.stdout)\n",
    "#print(result.stderr)\n",
    "lenka = len(fasta_seq)\n",
    "command = [\"hyperex\",  \"-f\", \"GTGARTCATCGARTCTTTG\", \"-r\", \"TCCTCCGCTTATTGATATGC\",\"-p\",\"hyperex_temp_out\", temp_filename,\"--force\"]\n",
    "for i, record in enumerate(fasta_seq[1:]):\n",
    "    #temp_filename = f\"hyperex_temp_res.fasta\"\n",
    "    if i % 50 == 0: print(f\"{i}/{lenka}\",end=\"\\r\")\n",
    "    with open(temp_filename, \"w\") as temp_fasta:\n",
    "        SeqIO.write(record, temp_fasta, \"fasta\")\n",
    "    #if i == 0:command = [\"hyperex\",  \"-f\", \"GTGARTCATCGARTCTTTG\", \"-r\", \"TCCTCCGCTTATTGATATGC\",\"-p\",\"hyperex_temp_out\", temp_filename]\n",
    "    result = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)\n",
    "    warnings.extend(list(map(lambda x: x.replace(\"\\x1b[0m]\",\"\"), re.findall(\"WARN(.+)\",result.stdout))))\n",
    "    errors.extend(result.stderr)\n",
    "    #if os.path.exists(hyperex_out_fa):\n",
    "    with open(hyperex_out_fa, \"r\") as fa_file:\n",
    "        results.append(fa_file.read())\n",
    "\n",
    "if os.path.exists(hyperex_out_fa):os.remove(hyperex_out_fa)  \n",
    "if os.path.exists(hyperex_log): os.remove(hyperex_log)\n",
    "if os.path.exists(hyperex_gff): os.remove(hyperex_gff)\n",
    "if os.path.exists(temp_filename): os.remove(temp_filename)  # Delete the temporary input file\n",
    "\n",
    "print()\n",
    "real_results = [i for i in results if len(i) >0]\n",
    "print(\"number of real results:\",len(real_results))\n",
    "warning_counts = Counter(warnings)\n",
    "error_counts = Counter(errors)\n",
    "print(\"ERRORS:\",list(error_counts),\"WARNINGS:\",\"\\n\".join(list(warning_counts)),sep=\"\\n\")\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "chim",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
