{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### In this notebook I will try to repair hyperex amplicon extraction tool\n",
    "\n",
    "hyperex keeps crushing when there is forward primer found upstream revese primer:\n",
    "\n",
    "thread 'main' panicked at /home/mg/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/hyperex-0.1.1/src/utils.rs:420:45:                                                                               slice index starts at 256 but ends at 65                                                                                                                                                                 note: run with \\`RUST_BACKTRACE=1` environment variable to display a backtrace                                                                                                                            Aborted (core dumped)   \n",
    "\n",
    "I will  load the fasta seq with Biopython and then use hyperex on each read"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "161335"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "fasta_seq = list(SeqIO.parse(\"./datasets/uchime_reference_dataset_16_10_2022/2022_10_26_chimera_reference_release/untrimmed_sequences/uchime_reference_dataset_untrimmed_16_10_2022.fasta\", \"fasta\"))\n",
    "#fasta_seq = fasta_seq[:300]\n",
    "len(fasta_seq)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of real results: 9772           \n",
      "WARNINGS:\n",
      "159223 x  Sequence length is less than 1500 bp. We may not be able to find some regions\n",
      "120436 x  Region  not found because primer TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "1295 x  Region  not found because primer GTGARTCATCGARTCTTTG was not found in the sequence\n",
      "29744 x  Region  not found because primers GTGARTCATCGARTCTTTG, TCCTCCGCTTATTGATATGC was not found in the sequence\n",
      "\n",
      "ERROR COUNT:\n",
      "13\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "import re\n",
    "from collections import Counter\n",
    "from Bio import SeqIO\n",
    "\n",
    "# Initialize variables\n",
    "results = []  # List to store the content of hyperex_out.fa files\n",
    "warnings = []  # List to store the warnings\n",
    "errors = []  # List to store the errors\n",
    "hyperex_out_fa = \"hyperex_temp_out.fa\"\n",
    "hyperex_log = \"hyperex.log\"\n",
    "hyperex_gff = \"hyperex_temp_out.gff\"\n",
    "hyperex_infile = \"infile.fa\"\n",
    "# Clean up any existing temporary files\n",
    "if os.path.exists(hyperex_log): os.remove(hyperex_log)\n",
    "if os.path.exists(hyperex_gff): os.remove(hyperex_gff)\n",
    "if os.path.exists(hyperex_out_fa): os.remove(hyperex_out_fa)\n",
    "if os.path.exists(hyperex_infile): os.remove(hyperex_infile)\n",
    "\n",
    "# Command template for hyperex\n",
    "\n",
    "# Function to count occurrences of the substring in the log file\n",
    "def count_log_occurrences(log_file, substring):\n",
    "    count = 0\n",
    "    with open(log_file, \"r\") as log:\n",
    "        for line in log:\n",
    "            if substring in line:\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Process the fasta sequences\n",
    "lenka = len(fasta_seq)\n",
    "start_index = 0\n",
    "\n",
    "command = [\"hyperex\", \"-f\", \"GTGARTCATCGARTCTTTG\", \"-r\", \"TCCTCCGCTTATTGATATGC\", \"-p\", \"hyperex_temp_out\"]\n",
    "\n",
    "cc = 0\n",
    "while start_index < lenka:\n",
    "    print(f\"{start_index}/{lenka}, good results: {len(results)}\", end=\"\\r\")\n",
    "    fasta_data = \"\".join([f\">{record.id}\\n{str(record.seq)}\\n\" for record in fasta_seq[start_index:]])\n",
    "    process = subprocess.Popen(command, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)\n",
    "    stdout, stderr = process.communicate(input=fasta_data)\n",
    "    warnings.extend(list(map(lambda x: x.replace(\"\\x1b[0m]\", \"\"), re.findall(\"WARN(.+)\", stdout))))\n",
    "    errors.append(stderr)\n",
    "    #print(stderr)\n",
    "    # Read results if the output file exists\n",
    "    if os.path.exists(hyperex_out_fa):\n",
    "        results.extend(list(SeqIO.parse(hyperex_out_fa,\"fasta\")))\n",
    "    \n",
    "    # Check for errors and adjust the start index\n",
    "    if stderr:\n",
    "        if os.path.exists(hyperex_log):\n",
    "            log_count = count_log_occurrences(hyperex_log, \"[hyperex::utils][INFO] Sequence type is DNA\")\n",
    "            start_index += log_count\n",
    "            #print(stderr)\n",
    "        if len(stderr) == 0:\n",
    "            print(\"\\nno more errors\")\n",
    "            break\n",
    "    else:break\n",
    "    if os.path.exists(hyperex_out_fa): os.remove(hyperex_out_fa)\n",
    "    if os.path.exists(hyperex_log): os.remove(hyperex_log)\n",
    "    if os.path.exists(hyperex_gff): os.remove(hyperex_gff)\n",
    "    if os.path.exists(hyperex_infile): os.remove(hyperex_infile)\n",
    "\n",
    "\n",
    "# Clean up any remaining temporary files\n",
    "if os.path.exists(hyperex_out_fa): os.remove(hyperex_out_fa)\n",
    "if os.path.exists(hyperex_log): os.remove(hyperex_log)\n",
    "if os.path.exists(hyperex_gff): os.remove(hyperex_gff)\n",
    "if os.path.exists(hyperex_infile): os.remove(hyperex_infile)\n",
    "\n",
    "# Print results\n",
    "real_results = [i for i in results if len(i) > 0]\n",
    "print(\"number of real results:\", len(real_results),\" \"*10)\n",
    "warning_counts = Counter(warnings)\n",
    "error_counts = Counter(errors)\n",
    "print(\"WARNINGS:\")\n",
    "for k,v in warning_counts.items():\n",
    "    print(v,\"x\",k)\n",
    "print(\"\\nERROR COUNT:\",len(errors), sep=\"\\n\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of features: 0\n",
      "Seq('')\n",
      "ID: Naumovozyma_baii|JX087438|SH1313431.09FU|reps|k__Fungi;p__Ascomycota;c__Saccharomycetes;o__Saccharomycetales;f__Saccharomycetaceae;g__Naumovozyma;s__Naumovozyma_baii\n",
      "Name: Naumovozyma_baii|JX087438|SH1313431.09FU|reps|k__Fungi;p__Ascomycota;c__Saccharomycetes;o__Saccharomycetales;f__Saccharomycetaceae;g__Naumovozyma;s__Naumovozyma_baii\n",
      "Description: Naumovozyma_baii|JX087438|SH1313431.09FU|reps|k__Fungi;p__Ascomycota;c__Saccharomycetes;o__Saccharomycetales;f__Saccharomycetaceae;g__Naumovozyma;s__Naumovozyma_baii forward=GTGARTCATCGARTCTTTG reverse=TCCTCCGCTTATTGATATGC\n",
      "Number of features: 0\n",
      "Seq('')\n",
      "Number of features: 0\n",
      "Seq('')\n",
      "ID: Pezoloma_ericae|UDB012594|SH1081240.09FU|reps|k__Fungi;p__Ascomycota;c__Leotiomycetes;o__Helotiales;f__Leotiaceae;g__Pezoloma;s__Pezoloma_ericae\n",
      "Name: Pezoloma_ericae|UDB012594|SH1081240.09FU|reps|k__Fungi;p__Ascomycota;c__Leotiomycetes;o__Helotiales;f__Leotiaceae;g__Pezoloma;s__Pezoloma_ericae\n",
      "Description: Pezoloma_ericae|UDB012594|SH1081240.09FU|reps|k__Fungi;p__Ascomycota;c__Leotiomycetes;o__Helotiales;f__Leotiaceae;g__Pezoloma;s__Pezoloma_ericae\n",
      "Number of features: 0\n",
      "Seq('')\n",
      "Number of features: 0\n",
      "Seq('')\n",
      "ID: Mycena_sp|KF800669|SH1145679.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Agaricales;f__Mycenaceae;g__Mycena;s__Mycena_sp\n",
      "Name: Mycena_sp|KF800669|SH1145679.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Agaricales;f__Mycenaceae;g__Mycena;s__Mycena_sp\n",
      "Description: Mycena_sp|KF800669|SH1145679.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Agaricales;f__Mycenaceae;g__Mycena;s__Mycena_sp\n",
      "Number of features: 0\n",
      "Seq('')\n",
      "ID: Galerina_sp|KX574498|SH1162902.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Agaricales;f__Strophariaceae;g__Galerina;s__Galerina_sp\n",
      "Name: Galerina_sp|KX574498|SH1162902.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Agaricales;f__Strophariaceae;g__Galerina;s__Galerina_sp\n",
      "Description: Galerina_sp|KX574498|SH1162902.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Agaricales;f__Strophariaceae;g__Galerina;s__Galerina_sp forward=GTGARTCATCGARTCTTTG reverse=TCCTCCGCTTATTGATATGC\n",
      "Number of features: 0\n",
      "Seq('')\n",
      "ID: Thelephoraceae_sp|EU057086|SH1196715.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Thelephorales;f__Thelephoraceae;g__unidentified;s__Thelephoraceae_sp\n",
      "Name: Thelephoraceae_sp|EU057086|SH1196715.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Thelephorales;f__Thelephoraceae;g__unidentified;s__Thelephoraceae_sp\n",
      "Description: Thelephoraceae_sp|EU057086|SH1196715.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Thelephorales;f__Thelephoraceae;g__unidentified;s__Thelephoraceae_sp forward=GTGARTCATCGARTCTTTG reverse=TCCTCCGCTTATTGATATGC\n",
      "Number of features: 0\n",
      "Seq('')\n",
      "ID: Micropsalliota_sp|UDB033704|SH1253519.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Agaricales;f__Agaricaceae;g__Micropsalliota;s__Micropsalliota_sp\n",
      "Name: Micropsalliota_sp|UDB033704|SH1253519.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Agaricales;f__Agaricaceae;g__Micropsalliota;s__Micropsalliota_sp\n",
      "Description: Micropsalliota_sp|UDB033704|SH1253519.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Agaricales;f__Agaricaceae;g__Micropsalliota;s__Micropsalliota_sp forward=GTGARTCATCGARTCTTTG reverse=TCCTCCGCTTATTGATATGC\n",
      "Number of features: 0\n",
      "Seq('')\n",
      "ID: Rachicladosporium_iridis|EU167586|SH0944768.09FU|refs|k__Fungi;p__Ascomycota;c__Dothideomycetes;o__Capnodiales;f__Cladosporiaceae;g__Rachicladosporium;s__Rachicladosporium_iridis\n",
      "Name: Rachicladosporium_iridis|EU167586|SH0944768.09FU|refs|k__Fungi;p__Ascomycota;c__Dothideomycetes;o__Capnodiales;f__Cladosporiaceae;g__Rachicladosporium;s__Rachicladosporium_iridis\n",
      "Description: Rachicladosporium_iridis|EU167586|SH0944768.09FU|refs|k__Fungi;p__Ascomycota;c__Dothideomycetes;o__Capnodiales;f__Cladosporiaceae;g__Rachicladosporium;s__Rachicladosporium_iridis forward=GTGARTCATCGARTCTTTG reverse=TCCTCCGCTTATTGATATGC\n",
      "Number of features: 0\n",
      "Seq('')\n",
      "ID: Hymenochaetales_sp|FJ179463|SH1136185.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Hymenochaetales;f__unidentified;g__unidentified;s__Hymenochaetales_sp\n",
      "Name: Hymenochaetales_sp|FJ179463|SH1136185.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Hymenochaetales;f__unidentified;g__unidentified;s__Hymenochaetales_sp\n",
      "Description: Hymenochaetales_sp|FJ179463|SH1136185.09FU|reps|k__Fungi;p__Basidiomycota;c__Agaricomycetes;o__Hymenochaetales;f__unidentified;g__unidentified;s__Hymenochaetales_sp forward=GTGARTCATCGARTCTTTG reverse=TCCTCCGCTTATTGATATGC\n",
      "Number of features: 0\n",
      "Seq('')\n"
     ]
    }
   ],
   "source": [
    "results[0]\n",
    "len(results)\n",
    "results[:10]\n",
    "for i in results:\n",
    "    if len(i) <1: print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0/300\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "250/300\n",
      "number of real results: 15\n",
      "ERRORS:\n",
      "Counter()\n",
      "WARNINGS:\n",
      "[' Sequence length is less than 1500 bp. We may not be able to find some regions', ' Region  not found because primer TCCTCCGCTTATTGATATGC was not found in the sequence', ' Overwriting hyperex_temp_out.fa and hyperex_temp_out.gff files', ' Region  not found because primer GTGARTCATCGARTCTTTG was not found in the sequence', ' Region  not found because primers GTGARTCATCGARTCTTTG, TCCTCCGCTTATTGATATGC was not found in the sequence']\n"
     ]
    }
   ],
   "source": [
    "# old version\n",
    "import os\n",
    "import subprocess\n",
    "import re\n",
    "from collections import Counter\n",
    "import sys\n",
    "\n",
    "results = []  # List to store the content of hyperex_out.fa files\n",
    "warnings = []  # List to store the warnings\n",
    "errors = []  # List to store the errors\n",
    "hyperex_out_fa = f\"hyperex_temp_out.fa\"\n",
    "hyperex_log = f\"hyperex.log\"\n",
    "hyperex_gff = f\"hyperex_temp_out.gff\"\n",
    "\n",
    "# Clean up any existing temporary files\n",
    "if os.path.exists(hyperex_log): os.remove(hyperex_log)\n",
    "if os.path.exists(hyperex_gff): os.remove(hyperex_gff)\n",
    "if os.path.exists(hyperex_out_fa): os.remove(hyperex_out_fa)\n",
    "\n",
    "# Command template for hyperex\n",
    "command = [\"hyperex\", \"-f\", \"GTGARTCATCGARTCTTTG\", \"-r\", \"TCCTCCGCTTATTGATATGC\", \"-p\", \"hyperex_temp_out\"]\n",
    "\n",
    "lenka = len(fasta_seq)\n",
    "for i, record in enumerate(fasta_seq):\n",
    "    if i % 50 == 0: print(f\"{i}/{lenka}\", end=\"\\r\")\n",
    "    if i >0: command = [\"hyperex\", \"-f\", \"GTGARTCATCGARTCTTTG\", \"-r\", \"TCCTCCGCTTATTGATATGC\", \"-p\", \"hyperex_temp_out\",\"--force\"] \n",
    "    # Convert the record to FASTA format string\n",
    "    fasta_data = f\">{record.id}\\n{str(record.seq)}\\n\"\n",
    "    \n",
    "    # Run hyperex with the FASTA data passed via stdin\n",
    "    process = subprocess.Popen(command, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)\n",
    "    stdout, stderr = process.communicate(input=fasta_data)\n",
    "    \n",
    "    # Collect warnings and errors\n",
    "    warnings.extend(list(map(lambda x: x.replace(\"\\x1b[0m]\", \"\"), re.findall(\"WARN(.+)\", stdout))))\n",
    "    errors.extend(stderr)\n",
    "    \n",
    "    # Read results if the output file exists\n",
    "    if os.path.exists(hyperex_out_fa):\n",
    "        with open(hyperex_out_fa, \"r\") as fa_file:\n",
    "            results.append(fa_file.read())\n",
    "\n",
    "# Clean up any remaining temporary files\n",
    "if os.path.exists(hyperex_out_fa): os.remove(hyperex_out_fa)\n",
    "if os.path.exists(hyperex_log): os.remove(hyperex_log)\n",
    "if os.path.exists(hyperex_gff): os.remove(hyperex_gff)\n",
    "\n",
    "print()\n",
    "real_results = [i for i in results if len(i) > 0]\n",
    "print(\"number of real results:\", len(real_results))\n",
    "warning_counts = Counter(warnings)\n",
    "error_counts = Counter(errors)\n",
    "print(\"ERRORS:\", error_counts, \"WARNINGS:\", warning_counts, sep=\"\\n\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# old version 2\n",
    "import os\n",
    "import subprocess\n",
    "#from time import sleep\n",
    "#import sys\n",
    "import re\n",
    "from collections import Counter\n",
    "results = []  # List to store the content of hyperex_out.fa files\n",
    "warnings = []  # List to store the warnings\n",
    "errors = []  # List to store the errors\n",
    "hyperex_out_fa = f\"hyperex_temp_out.fa\"\n",
    "hyperex_log = f\"hyperex.log\"\n",
    "hyperex_gff = f\"hyperex_temp_out.gff\"\n",
    "temp_filename = f\"hyperex_temp_res.fasta\"\n",
    "if os.path.exists(hyperex_log): os.remove(hyperex_log)\n",
    "if os.path.exists(hyperex_gff): os.remove(hyperex_gff)\n",
    "if os.path.exists(temp_filename): os.remove(temp_filename)  # Delete the temporary input file\n",
    "\n",
    "with open(temp_filename, \"w\") as temp_fasta:\n",
    "    SeqIO.write(fasta_seq[0], temp_fasta, \"fasta\")\n",
    "command = [\"hyperex\",  \"-f\", \"GTGARTCATCGARTCTTTG\", \"-r\", \"TCCTCCGCTTATTGATATGC\",\"-p\",\"hyperex_temp_out\", temp_filename]\n",
    "result = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)\n",
    "warnings.extend(list(map(lambda x: x.replace(\"\\x1b[0m]\",\"\"), re.findall(\"WARN(.+)\",result.stdout))))\n",
    "errors.extend(result.stderr)\n",
    "if os.path.exists(hyperex_out_fa):\n",
    "    with open(hyperex_out_fa, \"r\") as fa_file:\n",
    "        results.append(fa_file.read())\n",
    "#print(result.stdout)\n",
    "#print(result.stderr)\n",
    "lenka = len(fasta_seq)\n",
    "command = [\"hyperex\",  \"-f\", \"GTGARTCATCGARTCTTTG\", \"-r\", \"TCCTCCGCTTATTGATATGC\",\"-p\",\"hyperex_temp_out\", temp_filename,\"--force\"]\n",
    "for i, record in enumerate(fasta_seq[1:]):\n",
    "    #temp_filename = f\"hyperex_temp_res.fasta\"\n",
    "    if i % 50 == 0: print(f\"{i}/{lenka}\",end=\"\\r\")\n",
    "    with open(temp_filename, \"w\") as temp_fasta:\n",
    "        SeqIO.write(record, temp_fasta, \"fasta\")\n",
    "    #if i == 0:command = [\"hyperex\",  \"-f\", \"GTGARTCATCGARTCTTTG\", \"-r\", \"TCCTCCGCTTATTGATATGC\",\"-p\",\"hyperex_temp_out\", temp_filename]\n",
    "    result = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)\n",
    "    warnings.extend(list(map(lambda x: x.replace(\"\\x1b[0m]\",\"\"), re.findall(\"WARN(.+)\",result.stdout))))\n",
    "    errors.extend(result.stderr)\n",
    "    #if os.path.exists(hyperex_out_fa):\n",
    "    with open(hyperex_out_fa, \"r\") as fa_file:\n",
    "        results.append(fa_file.read())\n",
    "\n",
    "if os.path.exists(hyperex_out_fa):os.remove(hyperex_out_fa)  \n",
    "if os.path.exists(hyperex_log): os.remove(hyperex_log)\n",
    "if os.path.exists(hyperex_gff): os.remove(hyperex_gff)\n",
    "if os.path.exists(temp_filename): os.remove(temp_filename)  # Delete the temporary input file\n",
    "\n",
    "print()\n",
    "real_results = [i for i in results if len(i) >0]\n",
    "print(\"number of real results:\",len(real_results))\n",
    "warning_counts = Counter(warnings)\n",
    "error_counts = Counter(errors)\n",
    "print(\"ERRORS:\",list(error_counts),\"WARNINGS:\",\"\\n\".join(list(warning_counts)),sep=\"\\n\")\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "chim",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
